{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from pprint import pprint\n",
    "import pandas as pd\n",
    "pd.set_option('display.max_colwidth', -1)\n",
    "\n",
    "DATA_DIR = '/Users/singhn4/Projects/kids_first/data/Schiffman'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Helper functions\n",
    "def dropna_rows_cols(df_func):\n",
    "    \"\"\"\n",
    "    Decorator to drop rows and cols w all nan values\n",
    "    \"\"\"\n",
    "\n",
    "    def wrapper(*args, **kwargs):\n",
    "        df = df_func(*args, **kwargs)\n",
    "\n",
    "        # None or empty df\n",
    "        try:\n",
    "            if df.empty:\n",
    "                return df\n",
    "        except AttributeError:\n",
    "            return df\n",
    "\n",
    "        # Rows\n",
    "        df.dropna(how=\"all\", inplace=True)\n",
    "        # Cols\n",
    "        df.dropna(how=\"all\", axis=1, inplace=True)\n",
    "        return df\n",
    "\n",
    "    return wrapper\n",
    "\n",
    "def reformat_column_names(df_func):\n",
    "    \"\"\"\n",
    "    Decorator to reformat DataFrame column names.\n",
    "\n",
    "    Replace all column names having whitespace with underscore\n",
    "    and make lowercase\n",
    "    \"\"\"\n",
    "\n",
    "    def wrapper(*args, **kwargs):\n",
    "        df = df_func(*args, **kwargs)\n",
    "        # None or empty df\n",
    "        try:\n",
    "            if df.empty:\n",
    "                return df\n",
    "        except AttributeError:\n",
    "            return df\n",
    "        df.columns = map((lambda x: x.replace(\" \", \"_\").lower()),\n",
    "                         df.columns)\n",
    "        return df\n",
    "\n",
    "    return wrapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create study\n",
    "study = {\n",
    "    'data_access_authority': 'dbGaP',\n",
    "    'study_id': 'phs001194',\n",
    "    'study_version': 'v1.p2',\n",
    "    'study_name': 'Expanded Ewing sarcoma cohort for tumor genomics and association with DNA repair deficiences, clinical presentation, and outcome',\n",
    "    'attribution': 'https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetAcknowledgementStatement.cgi?study_id=phs001194.v1.p2'\n",
    "}\n",
    "study_df = pd.DataFrame([study])\n",
    "study_df.to_csv(os.path.join(DATA_DIR, 'study.txt'))\n",
    "\n",
    "# Create investigator\n",
    "invest = {\n",
    "    'investigator_name': 'Joshua Schiffman',\n",
    "    'institution': 'University of Utah'\n",
    "}\n",
    "inv_df = pd.DataFrame([invest])\n",
    "inv_df.to_csv(os.path.join(DATA_DIR, 'investigator.txt'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read study files data\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_study_file_data(filepaths=None):\n",
    "    \"\"\"\n",
    "    Read in raw study files\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filepaths = os.listdir(DATA_DIR)\n",
    "\n",
    "    study_files = [{\"study_file_name\": f}\n",
    "                   for f in filepaths]\n",
    "    return pd.DataFrame(study_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_study_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read study data\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR,\n",
    "                                'study.txt')\n",
    "    df = pd.read_csv(filepath)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_investigator_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read investigator data\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR,\n",
    "                                'investigator.txt')\n",
    "    df = pd.read_csv(filepath)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def cols_to_lower(df):\n",
    "    df.columns = map((lambda x: x.replace(\" \", \"_\").lower()), df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_participant_data(filepath=None):\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR, 'Schiffman_X01 Sample List.xlsx')\n",
    "    # Participants\n",
    "    df = pd.read_excel(filepath)\n",
    "    cols_to_lower(df)\n",
    "    # Extract participant columns\n",
    "    participant_df = df[['individual_name', 'ewing_trio_number', 'relationship_to_proband']]\n",
    "\n",
    "    # Create is_proband col\n",
    "    def func(row): return row.relationship_to_proband == 'Self/Case'\n",
    "    participant_df['relationship_to_proband'] = participant_df.apply(func, axis=1)\n",
    "\n",
    "    # Create family_id column\n",
    "    participant_df.rename(columns={'ewing_trio_number': 'family_id'})\n",
    "    \n",
    "    return participant_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Participants\n",
    "filepath = os.path.join(DATA_DIR, 'Schiffman_X01 Sample List.xlsx')\n",
    "all_data_df = pd.read_excel(filepath)\n",
    "cols_to_lower(all_data_df)\n",
    "\n",
    "def create_diagnosis_df(df):\n",
    "    # Create diagnosis df\n",
    "    df.loc[(df['morphology'] == \"9260/3: Ewing's sarcoma\\r\\nEwing's tumor\"),'morphology']= 'Ewing Sarcoma'    \n",
    "    def func(row): return \"_\".join(['diagnosis', str(row.name)])\n",
    "    df['diagnosis_id'] = df.apply(func, axis=1)\n",
    "    df = df[['individual_name', 'age_at_diagnosis_(days)', 'morphology']]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sequencing experiment\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_genomic_data(filepath=None):\n",
    "    if not filepath:\n",
    "        df = pd.read_excel(os.path.join(DATA_DIR, 'Schiffman_EwingSarcoma_QC_vs_Phenotype.xlsx'))\n",
    "        del df['instrument_data_ids']\n",
    "    return df\n",
    "\n",
    "def create_seq_exp_data(df):\n",
    "    df = df[['build_id', 'mean_insert_size', 'pf_reads', 'phenotype_sheet_sample_name']]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>unique</th>\n",
       "      <th>top</th>\n",
       "      <th>freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>c</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>H_UM-Schiffman-467-SS-470</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>prefix</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>H_UM-Schiffman-1181</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>coded</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>SS-522</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phenotype_sheet_sample_name</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>SS-522</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>individual_name</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>Schiffman-905</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>build_id</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>db2c90e8acdc4974bfe9b2cc44199e17</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bam_path</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>/gscmnt/gc13034/prod-builder/build_merged_alignments/merged-alignment-blade17-3-1.gsc.wustl.edu-prod-builder-3431-5059d31e024f4c94a71bf581ac401ea8/5059d31e024f4c94a71bf581ac401ea8.bam</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>topography</th>\n",
       "      <td>454</td>\n",
       "      <td>33</td>\n",
       "      <td>C40.2: Long bones of lower limb and associated joints</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>morphology</th>\n",
       "      <td>452</td>\n",
       "      <td>11</td>\n",
       "      <td>9260/3: Ewing's sarcoma\\n\\nEwing's tumor</td>\n",
       "      <td>368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>notes_from_washu</th>\n",
       "      <td>1107</td>\n",
       "      <td>6</td>\n",
       "      <td>Sequenced</td>\n",
       "      <td>974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>qc_comment</th>\n",
       "      <td>1112</td>\n",
       "      <td>5</td>\n",
       "      <td>pass</td>\n",
       "      <td>918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relationship_to_proband</th>\n",
       "      <td>1112</td>\n",
       "      <td>4</td>\n",
       "      <td>Self/Case</td>\n",
       "      <td>452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>full_trio/singleton/dropped</th>\n",
       "      <td>1112</td>\n",
       "      <td>3</td>\n",
       "      <td>Full Trio</td>\n",
       "      <td>988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_type</th>\n",
       "      <td>1112</td>\n",
       "      <td>3</td>\n",
       "      <td>Saliva</td>\n",
       "      <td>1071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gender</th>\n",
       "      <td>1112</td>\n",
       "      <td>2</td>\n",
       "      <td>Male</td>\n",
       "      <td>571</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            count unique  \\\n",
       "c                            1112  1112    \n",
       "prefix                       1112  1112    \n",
       "coded                        1112  1112    \n",
       "phenotype_sheet_sample_name  1112  1112    \n",
       "individual_name              1112  1112    \n",
       "build_id                     1112  1112    \n",
       "bam_path                     1112  1112    \n",
       "topography                   454   33      \n",
       "morphology                   452   11      \n",
       "notes_from_washu             1107  6       \n",
       "qc_comment                   1112  5       \n",
       "relationship_to_proband      1112  4       \n",
       "full_trio/singleton/dropped  1112  3       \n",
       "sample_type                  1112  3       \n",
       "gender                       1112  2       \n",
       "\n",
       "                                                                                                                                                                                                                 top  \\\n",
       "c                            H_UM-Schiffman-467-SS-470                                                                                                                                                                 \n",
       "prefix                       H_UM-Schiffman-1181                                                                                                                                                                       \n",
       "coded                        SS-522                                                                                                                                                                                    \n",
       "phenotype_sheet_sample_name  SS-522                                                                                                                                                                                    \n",
       "individual_name              Schiffman-905                                                                                                                                                                             \n",
       "build_id                     db2c90e8acdc4974bfe9b2cc44199e17                                                                                                                                                          \n",
       "bam_path                     /gscmnt/gc13034/prod-builder/build_merged_alignments/merged-alignment-blade17-3-1.gsc.wustl.edu-prod-builder-3431-5059d31e024f4c94a71bf581ac401ea8/5059d31e024f4c94a71bf581ac401ea8.bam   \n",
       "topography                   C40.2: Long bones of lower limb and associated joints                                                                                                                                     \n",
       "morphology                   9260/3: Ewing's sarcoma\\n\\nEwing's tumor                                                                                                                                                  \n",
       "notes_from_washu             Sequenced                                                                                                                                                                                 \n",
       "qc_comment                   pass                                                                                                                                                                                      \n",
       "relationship_to_proband      Self/Case                                                                                                                                                                                 \n",
       "full_trio/singleton/dropped  Full Trio                                                                                                                                                                                 \n",
       "sample_type                  Saliva                                                                                                                                                                                    \n",
       "gender                       Male                                                                                                                                                                                      \n",
       "\n",
       "                             freq  \n",
       "c                            1     \n",
       "prefix                       1     \n",
       "coded                        1     \n",
       "phenotype_sheet_sample_name  1     \n",
       "individual_name              1     \n",
       "build_id                     1     \n",
       "bam_path                     1     \n",
       "topography                   86    \n",
       "morphology                   368   \n",
       "notes_from_washu             974   \n",
       "qc_comment                   918   \n",
       "relationship_to_proband      452   \n",
       "full_trio/singleton/dropped  988   \n",
       "sample_type                  1071  \n",
       "gender                       571   "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Genomic files\n",
    "import json \n",
    "df = read_genomic_data()\n",
    "df.describe(include=['O']).T.sort_values('unique', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create genomic file df\n",
    "def create_genomic_file_df(df):\n",
    "    df = df[['build_id', 'bam_path']]\n",
    "    df['file_name'] = df['bam_path'].apply(lambda p: os.path.basename(p))\n",
    "\n",
    "    def get_ext(fp):\n",
    "        filename = os.path.basename(fp)\n",
    "        parts = filename.split('.')\n",
    "        if len(parts) > 2:\n",
    "            ext = '.'.join(parts[1:])\n",
    "        else:\n",
    "            ext = parts[-1]\n",
    "        return ext\n",
    "\n",
    "    with open(os.path.join(DATA_DIR, 'genomic_file_uuid.json'), 'r') as json_file:\n",
    "        uuid_dict = json.load(json_file)\n",
    "\n",
    "    gf_dicts = []\n",
    "    for k, v in uuid_dict.items():\n",
    "        file_info = {\n",
    "            'uuid': v['did'],\n",
    "            'md5sum': v['hashes']['md5'],\n",
    "            'file_url': v['urls'][0],\n",
    "            'data_type': 'submitted aligned reads',\n",
    "            'file_format': get_ext(v['urls'][0]),\n",
    "            'file_name': os.path.basename(v['urls'][0])\n",
    "        }\n",
    "        gf_dicts.append(file_info)\n",
    "\n",
    "    gf_df = pd.DataFrame(gf_dicts)\n",
    "\n",
    "    genomic_file_df = pd.merge(df, gf_df, on='file_name')\n",
    "    \n",
    "    return genomic_file_df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Explore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['All_Schiffman_EwingSarcoma_QC_vs_Phenotype.xlsx',\n",
      " 'Copy of Pilot Score Sheet 2017.xlsx',\n",
      " 'EwingSarcoma_QC.1112samples.csv',\n",
      " 'genomic_file_uuid.json',\n",
      " 'investigator.txt',\n",
      " 'Schiffman_EwingSarcoma_QC_vs_Phenotype.csv',\n",
      " 'Schiffman_EwingSarcoma_QC_vs_Phenotype.txt',\n",
      " 'Schiffman_EwingSarcoma_QC_vs_Phenotype.xlsx',\n",
      " 'Schiffman_EwingSarcoma_QC_vs_Phenotype_trios_annotated.txt',\n",
      " 'Schiffman_X01 Sample List.xlsx',\n",
      " 'study.txt',\n",
      " '~$All_Schiffman_EwingSarcoma_QC_vs_Phenotype.xlsx']\n"
     ]
    }
   ],
   "source": [
    "pprint([f for f in os.listdir(DATA_DIR)])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Sample List (Schiffman_X01 Sample List.xlsx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1207, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>individual_name</th>\n",
       "      <th>ewing_trio_number</th>\n",
       "      <th>relationship_to_proband</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_diagnosis_(days)</th>\n",
       "      <th>age_at_enrollment_(days)</th>\n",
       "      <th>topography</th>\n",
       "      <th>morphology</th>\n",
       "      <th>full_trio/singleton/dropped</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>notes_from_washu</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SS-1</td>\n",
       "      <td>Schiffman-1</td>\n",
       "      <td>189</td>\n",
       "      <td>Father</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12678.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Full Trio</td>\n",
       "      <td>Saliva</td>\n",
       "      <td>Sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SS-2</td>\n",
       "      <td>Schiffman-2</td>\n",
       "      <td>189</td>\n",
       "      <td>Mother</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12919.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Full Trio</td>\n",
       "      <td>Saliva</td>\n",
       "      <td>Sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SS-3</td>\n",
       "      <td>Schiffman-3</td>\n",
       "      <td>189</td>\n",
       "      <td>Self/Case</td>\n",
       "      <td>Male</td>\n",
       "      <td>3761.0</td>\n",
       "      <td>4424.0</td>\n",
       "      <td>C40.2: Long bones of lower limb and associated joints</td>\n",
       "      <td>9260/3: Ewing's sarcoma\\r\\nEwing's tumor</td>\n",
       "      <td>Full Trio</td>\n",
       "      <td>Saliva</td>\n",
       "      <td>Sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SS-4</td>\n",
       "      <td>Schiffman-4</td>\n",
       "      <td>684</td>\n",
       "      <td>Mother</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16554.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Full Trio</td>\n",
       "      <td>Saliva</td>\n",
       "      <td>Sequenced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SS-5</td>\n",
       "      <td>Schiffman-5</td>\n",
       "      <td>684</td>\n",
       "      <td>Father</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16613.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Full Trio</td>\n",
       "      <td>Saliva</td>\n",
       "      <td>Sequenced</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_name individual_name ewing_trio_number relationship_to_proband  \\\n",
       "0  SS-1        Schiffman-1     189               Father                   \n",
       "1  SS-2        Schiffman-2     189               Mother                   \n",
       "2  SS-3        Schiffman-3     189               Self/Case                \n",
       "3  SS-4        Schiffman-4     684               Mother                   \n",
       "4  SS-5        Schiffman-5     684               Father                   \n",
       "\n",
       "   gender  age_at_diagnosis_(days)  age_at_enrollment_(days)  \\\n",
       "0  Male   NaN                       12678.0                    \n",
       "1  Female NaN                       12919.0                    \n",
       "2  Male    3761.0                   4424.0                     \n",
       "3  Female NaN                       16554.0                    \n",
       "4  Male   NaN                       16613.0                    \n",
       "\n",
       "                                              topography  \\\n",
       "0  NaN                                                     \n",
       "1  NaN                                                     \n",
       "2  C40.2: Long bones of lower limb and associated joints   \n",
       "3  NaN                                                     \n",
       "4  NaN                                                     \n",
       "\n",
       "                                 morphology full_trio/singleton/dropped  \\\n",
       "0  NaN                                       Full Trio                    \n",
       "1  NaN                                       Full Trio                    \n",
       "2  9260/3: Ewing's sarcoma\\r\\nEwing's tumor  Full Trio                    \n",
       "3  NaN                                       Full Trio                    \n",
       "4  NaN                                       Full Trio                    \n",
       "\n",
       "  sample_type notes_from_washu  \n",
       "0  Saliva      Sequenced        \n",
       "1  Saliva      Sequenced        \n",
       "2  Saliva      Sequenced        \n",
       "3  Saliva      Sequenced        \n",
       "4  Saliva      Sequenced        "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_list_df = pd.read_excel(os.path.join(DATA_DIR, 'Schiffman_X01 Sample List.xlsx'))\n",
    "cols_to_lower(sample_list_df)\n",
    "print(sample_list_df.shape)\n",
    "sample_list_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>unique</th>\n",
       "      <th>top</th>\n",
       "      <th>freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <td>1207</td>\n",
       "      <td>1207</td>\n",
       "      <td>SS-522</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>individual_name</th>\n",
       "      <td>1207</td>\n",
       "      <td>1203</td>\n",
       "      <td>Schiffman-449</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ewing_trio_number</th>\n",
       "      <td>1207</td>\n",
       "      <td>485</td>\n",
       "      <td>681</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>topography</th>\n",
       "      <td>381</td>\n",
       "      <td>32</td>\n",
       "      <td>C40.2: Long bones of lower limb and associated joints</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>morphology</th>\n",
       "      <td>380</td>\n",
       "      <td>9</td>\n",
       "      <td>9260/3: Ewing's sarcoma\\r\\nEwing's tumor</td>\n",
       "      <td>371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>notes_from_washu</th>\n",
       "      <td>1200</td>\n",
       "      <td>8</td>\n",
       "      <td>Sequenced</td>\n",
       "      <td>974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>full_trio/singleton/dropped</th>\n",
       "      <td>1207</td>\n",
       "      <td>6</td>\n",
       "      <td>Full Trio</td>\n",
       "      <td>989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relationship_to_proband</th>\n",
       "      <td>1207</td>\n",
       "      <td>4</td>\n",
       "      <td>Self/Case</td>\n",
       "      <td>487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_type</th>\n",
       "      <td>1207</td>\n",
       "      <td>3</td>\n",
       "      <td>Saliva</td>\n",
       "      <td>1153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gender</th>\n",
       "      <td>1207</td>\n",
       "      <td>2</td>\n",
       "      <td>Male</td>\n",
       "      <td>618</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            count unique  \\\n",
       "sample_name                  1207  1207    \n",
       "individual_name              1207  1203    \n",
       "ewing_trio_number            1207  485     \n",
       "topography                   381   32      \n",
       "morphology                   380   9       \n",
       "notes_from_washu             1200  8       \n",
       "full_trio/singleton/dropped  1207  6       \n",
       "relationship_to_proband      1207  4       \n",
       "sample_type                  1207  3       \n",
       "gender                       1207  2       \n",
       "\n",
       "                                                                               top  \\\n",
       "sample_name                  SS-522                                                  \n",
       "individual_name              Schiffman-449                                           \n",
       "ewing_trio_number            681                                                     \n",
       "topography                   C40.2: Long bones of lower limb and associated joints   \n",
       "morphology                   9260/3: Ewing's sarcoma\\r\\nEwing's tumor                \n",
       "notes_from_washu             Sequenced                                               \n",
       "full_trio/singleton/dropped  Full Trio                                               \n",
       "relationship_to_proband      Self/Case                                               \n",
       "sample_type                  Saliva                                                  \n",
       "gender                       Male                                                    \n",
       "\n",
       "                             freq  \n",
       "sample_name                  1     \n",
       "individual_name              2     \n",
       "ewing_trio_number            6     \n",
       "topography                   86    \n",
       "morphology                   371   \n",
       "notes_from_washu             974   \n",
       "full_trio/singleton/dropped  989   \n",
       "relationship_to_proband      487   \n",
       "sample_type                  1153  \n",
       "gender                       618   "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_list_df.describe(include=['O']).T.sort_values('unique', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Morphologies\n",
      "[nan,\n",
      " \"9260/3: Ewing's sarcoma\\r\\nEwing's tumor\",\n",
      " '170.9 - Ewing Sarcoma & 171.0 - Rhabdomyosarcoma',\n",
      " 'Ewing sarcoma with EWSR1 (22q12) rearrangement',\n",
      " 'Most consistent with Ewing (EWS-FLI negative)',\n",
      " '9364/3: Neuroectodermal Tumor, NOS',\n",
      " '9364/3: Peripheral neuroectodermal tumor\\r\\nNeuroectodermal tumor, NOS',\n",
      " 'Ewing Sarcoma/PNET',\n",
      " '171.8 - Malignant Neoplasm of other specified sites of connective and other '\n",
      " 'soft tissue. Undifferentiated sarcoma of the left thigh',\n",
      " '170.9: Malignant neoplasm of bon and articular cartilage, Ewing Sarcoma']\n",
      "\n",
      "Topographies\n",
      "[nan,\n",
      " 'C40.2: Long bones of lower limb and associated joints',\n",
      " 'C41.2: Vertebral column',\n",
      " 'C41.4: Pelvic bones, sacrum, coccyx and associated joints',\n",
      " 'C40.0: Long bones of upper limb, scapula and associated joints',\n",
      " 'C41.3: Rib, sternum, clavicle and associated joints',\n",
      " 'C76.5: Lower limb, NOS',\n",
      " 'C40.3: Short bones of lower limb and associated joints',\n",
      " 'C41.1: Mandible',\n",
      " 'C76.3: Pelvis, NOS',\n",
      " 'C40.9: Bone of limb, NOS',\n",
      " 'C41.9: Bone, NOS',\n",
      " 'C41.0: Bones of skull and face and associated joints',\n",
      " 'C76.1: Thorax, NOS',\n",
      " 'Unknown',\n",
      " 'C40.1: Short bones of upper limb and associated joints',\n",
      " 'Soft tissue of the left pelvis',\n",
      " 'C76.4: Upper limb, NOS',\n",
      " 'Mediastinum',\n",
      " 'Left Posterior flank',\n",
      " 'C76.7: Other ill-defined sites',\n",
      " 'C76.2: Abdomen, NOS',\n",
      " 'Left Upper Back',\n",
      " 'Left Posterior Thigh',\n",
      " 'C71.9: Primitive neuroectodermal tumor',\n",
      " 'Right pumonary lesion',\n",
      " 'Ewing Sarcoma of the Right Foot',\n",
      " 'Anterior left middle cranial fossa',\n",
      " 'Left thigh',\n",
      " 'C76.0: Head, face or neck, NOS',\n",
      " 'Right Flank',\n",
      " 'Soft tissue of left forearm',\n",
      " 'Ewing sarcoma of the Kidney']\n"
     ]
    }
   ],
   "source": [
    "# Unique morphologies\n",
    "print('Morphologies')\n",
    "pprint(sample_list_df['morphology'].unique().tolist())\n",
    "print('\\nTopographies')\n",
    "# Unique topographies\n",
    "pprint(sample_list_df['topography'].unique().tolist())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### QC Phenotype (Schiffman_EwingSarcoma_QC_vs_Phenotype.xlsx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>c</th>\n",
       "      <th>prefix</th>\n",
       "      <th>coded</th>\n",
       "      <th>phenotype_sheet_sample_name</th>\n",
       "      <th>individual_name</th>\n",
       "      <th>ewing_trio_number</th>\n",
       "      <th>relationship_to_proband</th>\n",
       "      <th>relationship_coded</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_diagnosis_(days)</th>\n",
       "      <th>...</th>\n",
       "      <th>reads_mapped_in_interchromosomal_pairs_percentage</th>\n",
       "      <th>reads_mapped_as_singleton_percentage</th>\n",
       "      <th>pct_10x</th>\n",
       "      <th>pct_20x</th>\n",
       "      <th>discordant_rate</th>\n",
       "      <th>qc_comment</th>\n",
       "      <th>pass/fail</th>\n",
       "      <th>proband_pass</th>\n",
       "      <th>mother_pass</th>\n",
       "      <th>father_pass</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H_UM-Schiffman-856-SS-859</td>\n",
       "      <td>H_UM-Schiffman-856</td>\n",
       "      <td>SS-859</td>\n",
       "      <td>SS-859</td>\n",
       "      <td>Schiffman-856</td>\n",
       "      <td>1</td>\n",
       "      <td>Self/Case</td>\n",
       "      <td>1</td>\n",
       "      <td>Female</td>\n",
       "      <td>5377.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018480</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.988625</td>\n",
       "      <td>0.924837</td>\n",
       "      <td>2.448235</td>\n",
       "      <td>pass</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>H_UM-Schiffman-857-SS-860</td>\n",
       "      <td>H_UM-Schiffman-857</td>\n",
       "      <td>SS-860</td>\n",
       "      <td>SS-860</td>\n",
       "      <td>Schiffman-857</td>\n",
       "      <td>1</td>\n",
       "      <td>Mother</td>\n",
       "      <td>2</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.021641</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.989257</td>\n",
       "      <td>0.959560</td>\n",
       "      <td>2.708281</td>\n",
       "      <td>pass</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>H_UM-Schiffman-858-SS-861</td>\n",
       "      <td>H_UM-Schiffman-858</td>\n",
       "      <td>SS-861</td>\n",
       "      <td>SS-861</td>\n",
       "      <td>Schiffman-858</td>\n",
       "      <td>1</td>\n",
       "      <td>Father</td>\n",
       "      <td>3</td>\n",
       "      <td>Male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.021698</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.992999</td>\n",
       "      <td>0.923224</td>\n",
       "      <td>2.803853</td>\n",
       "      <td>pass</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>H_UM-Schiffman-827-SS-830</td>\n",
       "      <td>H_UM-Schiffman-827</td>\n",
       "      <td>SS-830</td>\n",
       "      <td>SS-830</td>\n",
       "      <td>Schiffman-827</td>\n",
       "      <td>2</td>\n",
       "      <td>Self/Case</td>\n",
       "      <td>1</td>\n",
       "      <td>Female</td>\n",
       "      <td>5455.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018074</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.988363</td>\n",
       "      <td>0.907885</td>\n",
       "      <td>2.401644</td>\n",
       "      <td>pass</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>H_UM-Schiffman-826-SS-829</td>\n",
       "      <td>H_UM-Schiffman-826</td>\n",
       "      <td>SS-829</td>\n",
       "      <td>SS-829</td>\n",
       "      <td>Schiffman-826</td>\n",
       "      <td>2</td>\n",
       "      <td>Mother</td>\n",
       "      <td>2</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018384</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.988576</td>\n",
       "      <td>0.927557</td>\n",
       "      <td>2.597264</td>\n",
       "      <td>pass</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           c              prefix   coded  \\\n",
       "0  H_UM-Schiffman-856-SS-859  H_UM-Schiffman-856  SS-859   \n",
       "1  H_UM-Schiffman-857-SS-860  H_UM-Schiffman-857  SS-860   \n",
       "2  H_UM-Schiffman-858-SS-861  H_UM-Schiffman-858  SS-861   \n",
       "3  H_UM-Schiffman-827-SS-830  H_UM-Schiffman-827  SS-830   \n",
       "4  H_UM-Schiffman-826-SS-829  H_UM-Schiffman-826  SS-829   \n",
       "\n",
       "  phenotype_sheet_sample_name individual_name  ewing_trio_number  \\\n",
       "0  SS-859                      Schiffman-856   1                   \n",
       "1  SS-860                      Schiffman-857   1                   \n",
       "2  SS-861                      Schiffman-858   1                   \n",
       "3  SS-830                      Schiffman-827   2                   \n",
       "4  SS-829                      Schiffman-826   2                   \n",
       "\n",
       "  relationship_to_proband  relationship_coded  gender  \\\n",
       "0  Self/Case               1                   Female   \n",
       "1  Mother                  2                   Female   \n",
       "2  Father                  3                   Male     \n",
       "3  Self/Case               1                   Female   \n",
       "4  Mother                  2                   Female   \n",
       "\n",
       "   age_at_diagnosis_(days)     ...       \\\n",
       "0  5377.0                      ...        \n",
       "1 NaN                          ...        \n",
       "2 NaN                          ...        \n",
       "3  5455.0                      ...        \n",
       "4 NaN                          ...        \n",
       "\n",
       "   reads_mapped_in_interchromosomal_pairs_percentage  \\\n",
       "0  0.018480                                            \n",
       "1  0.021641                                            \n",
       "2  0.021698                                            \n",
       "3  0.018074                                            \n",
       "4  0.018384                                            \n",
       "\n",
       "  reads_mapped_as_singleton_percentage   pct_10x   pct_20x discordant_rate  \\\n",
       "0  0.17                                 0.988625  0.924837  2.448235         \n",
       "1  0.17                                 0.989257  0.959560  2.708281         \n",
       "2  0.16                                 0.992999  0.923224  2.803853         \n",
       "3  0.21                                 0.988363  0.907885  2.401644         \n",
       "4  0.25                                 0.988576  0.927557  2.597264         \n",
       "\n",
       "  qc_comment pass/fail  proband_pass mother_pass  father_pass  \n",
       "0  pass       0         1             0           0            \n",
       "1  pass       0         0             1           0            \n",
       "2  pass       0         0             0           1            \n",
       "3  pass       0         1             0           0            \n",
       "4  pass       0         0             1           0            \n",
       "\n",
       "[5 rows x 43 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "qc_phenotype_df = pd.read_excel(os.path.join(DATA_DIR, 'Schiffman_EwingSarcoma_QC_vs_Phenotype.xlsx'))\n",
    "del qc_phenotype_df['instrument_data_ids']\n",
    "cols_to_lower(qc_phenotype_df)\n",
    "qc_phenotype_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>unique</th>\n",
       "      <th>top</th>\n",
       "      <th>freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>c</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>H_UM-Schiffman-467-SS-470</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>prefix</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>H_UM-Schiffman-1181</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>coded</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>SS-522</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phenotype_sheet_sample_name</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>SS-522</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>individual_name</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>Schiffman-905</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>build_id</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>db2c90e8acdc4974bfe9b2cc44199e17</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bam_path</th>\n",
       "      <td>1112</td>\n",
       "      <td>1112</td>\n",
       "      <td>/gscmnt/gc13034/prod-builder/build_merged_alignments/merged-alignment-blade17-3-1.gsc.wustl.edu-prod-builder-3431-5059d31e024f4c94a71bf581ac401ea8/5059d31e024f4c94a71bf581ac401ea8.bam</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>topography</th>\n",
       "      <td>454</td>\n",
       "      <td>33</td>\n",
       "      <td>C40.2: Long bones of lower limb and associated joints</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>morphology</th>\n",
       "      <td>452</td>\n",
       "      <td>11</td>\n",
       "      <td>9260/3: Ewing's sarcoma\\n\\nEwing's tumor</td>\n",
       "      <td>368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>notes_from_washu</th>\n",
       "      <td>1107</td>\n",
       "      <td>6</td>\n",
       "      <td>Sequenced</td>\n",
       "      <td>974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>qc_comment</th>\n",
       "      <td>1112</td>\n",
       "      <td>5</td>\n",
       "      <td>pass</td>\n",
       "      <td>918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relationship_to_proband</th>\n",
       "      <td>1112</td>\n",
       "      <td>4</td>\n",
       "      <td>Self/Case</td>\n",
       "      <td>452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>full_trio/singleton/dropped</th>\n",
       "      <td>1112</td>\n",
       "      <td>3</td>\n",
       "      <td>Full Trio</td>\n",
       "      <td>988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_type</th>\n",
       "      <td>1112</td>\n",
       "      <td>3</td>\n",
       "      <td>Saliva</td>\n",
       "      <td>1071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gender</th>\n",
       "      <td>1112</td>\n",
       "      <td>2</td>\n",
       "      <td>Male</td>\n",
       "      <td>571</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            count unique  \\\n",
       "c                            1112  1112    \n",
       "prefix                       1112  1112    \n",
       "coded                        1112  1112    \n",
       "phenotype_sheet_sample_name  1112  1112    \n",
       "individual_name              1112  1112    \n",
       "build_id                     1112  1112    \n",
       "bam_path                     1112  1112    \n",
       "topography                   454   33      \n",
       "morphology                   452   11      \n",
       "notes_from_washu             1107  6       \n",
       "qc_comment                   1112  5       \n",
       "relationship_to_proband      1112  4       \n",
       "full_trio/singleton/dropped  1112  3       \n",
       "sample_type                  1112  3       \n",
       "gender                       1112  2       \n",
       "\n",
       "                                                                                                                                                                                                                 top  \\\n",
       "c                            H_UM-Schiffman-467-SS-470                                                                                                                                                                 \n",
       "prefix                       H_UM-Schiffman-1181                                                                                                                                                                       \n",
       "coded                        SS-522                                                                                                                                                                                    \n",
       "phenotype_sheet_sample_name  SS-522                                                                                                                                                                                    \n",
       "individual_name              Schiffman-905                                                                                                                                                                             \n",
       "build_id                     db2c90e8acdc4974bfe9b2cc44199e17                                                                                                                                                          \n",
       "bam_path                     /gscmnt/gc13034/prod-builder/build_merged_alignments/merged-alignment-blade17-3-1.gsc.wustl.edu-prod-builder-3431-5059d31e024f4c94a71bf581ac401ea8/5059d31e024f4c94a71bf581ac401ea8.bam   \n",
       "topography                   C40.2: Long bones of lower limb and associated joints                                                                                                                                     \n",
       "morphology                   9260/3: Ewing's sarcoma\\n\\nEwing's tumor                                                                                                                                                  \n",
       "notes_from_washu             Sequenced                                                                                                                                                                                 \n",
       "qc_comment                   pass                                                                                                                                                                                      \n",
       "relationship_to_proband      Self/Case                                                                                                                                                                                 \n",
       "full_trio/singleton/dropped  Full Trio                                                                                                                                                                                 \n",
       "sample_type                  Saliva                                                                                                                                                                                    \n",
       "gender                       Male                                                                                                                                                                                      \n",
       "\n",
       "                             freq  \n",
       "c                            1     \n",
       "prefix                       1     \n",
       "coded                        1     \n",
       "phenotype_sheet_sample_name  1     \n",
       "individual_name              1     \n",
       "build_id                     1     \n",
       "bam_path                     1     \n",
       "topography                   86    \n",
       "morphology                   368   \n",
       "notes_from_washu             974   \n",
       "qc_comment                   918   \n",
       "relationship_to_proband      452   \n",
       "full_trio/singleton/dropped  988   \n",
       "sample_type                  1071  \n",
       "gender                       571   "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "qc_phenotype_df.describe(include=['O']).T.sort_values('unique', ascending=False)\n",
    "# qc_phenotype_df.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Trio Summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "trio_df = pd.read_excel(os.path.join(DATA_DIR, 'Schiffman_EwingSarcoma_QC_vs_Phenotype.xlsx'), sheet_name=1)\n",
    "# trio_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extract"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unnamed:_0</th>\n",
       "      <th>attribution</th>\n",
       "      <th>data_access_authority</th>\n",
       "      <th>study_id</th>\n",
       "      <th>study_name</th>\n",
       "      <th>study_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetAcknowledgementStatement.cgi?study_id=phs001194.v1.p2</td>\n",
       "      <td>dbGaP</td>\n",
       "      <td>phs001194</td>\n",
       "      <td>Expanded Ewing sarcoma cohort for tumor genomics and association with DNA repair deficiences, clinical presentation, and outcome</td>\n",
       "      <td>v1.p2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   unnamed:_0  \\\n",
       "0  0            \n",
       "\n",
       "                                                                                                  attribution  \\\n",
       "0  https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetAcknowledgementStatement.cgi?study_id=phs001194.v1.p2   \n",
       "\n",
       "  data_access_authority   study_id  \\\n",
       "0  dbGaP                 phs001194   \n",
       "\n",
       "                                                                                                                         study_name  \\\n",
       "0  Expanded Ewing sarcoma cohort for tumor genomics and association with DNA repair deficiences, clinical presentation, and outcome   \n",
       "\n",
       "  study_version  \n",
       "0  v1.p2         "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Study\n",
    "study_df = read_study_data()\n",
    "study_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unnamed:_0</th>\n",
       "      <th>institution</th>\n",
       "      <th>investigator_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>University of Utah</td>\n",
       "      <td>Joshua Schiffman</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   unnamed:_0         institution investigator_name\n",
       "0  0           University of Utah  Joshua Schiffman"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Investigator\n",
    "investigator_df = read_investigator_data()\n",
    "investigator_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>All_Schiffman_EwingSarcoma_QC_vs_Phenotype.xlsx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Copy of Pilot Score Sheet 2017.xlsx</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EwingSarcoma_QC.1112samples.csv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>genomic_file_uuid.json</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>investigator.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   study_file_name\n",
       "0  All_Schiffman_EwingSarcoma_QC_vs_Phenotype.xlsx\n",
       "1  Copy of Pilot Score Sheet 2017.xlsx            \n",
       "2  EwingSarcoma_QC.1112samples.csv                \n",
       "3  genomic_file_uuid.json                         \n",
       "4  investigator.txt                               "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Study files\n",
    "study_file_df = read_study_file_data()\n",
    "study_file_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/singhn4/Projects/kids_first/kf-api-dataservice/venv/lib/python3.6/site-packages/ipykernel_launcher.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individual_name</th>\n",
       "      <th>ewing_trio_number</th>\n",
       "      <th>relationship_to_proband</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Schiffman-1</td>\n",
       "      <td>189</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Schiffman-2</td>\n",
       "      <td>189</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Schiffman-3</td>\n",
       "      <td>189</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Schiffman-4</td>\n",
       "      <td>684</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Schiffman-5</td>\n",
       "      <td>684</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  individual_name ewing_trio_number  relationship_to_proband\n",
       "0  Schiffman-1     189               False                  \n",
       "1  Schiffman-2     189               False                  \n",
       "2  Schiffman-3     189               True                   \n",
       "3  Schiffman-4     684               False                  \n",
       "4  Schiffman-5     684               False                  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Participants\n",
    "participant_df = read_participant_data()\n",
    "participant_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individual_name</th>\n",
       "      <th>age_at_diagnosis_(days)</th>\n",
       "      <th>morphology</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Schiffman-1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Schiffman-2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Schiffman-3</td>\n",
       "      <td>3761</td>\n",
       "      <td>Ewing Sarcoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Schiffman-4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Schiffman-5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  individual_name age_at_diagnosis_(days)     morphology\n",
       "0  Schiffman-1     None                    None         \n",
       "1  Schiffman-2     None                    None         \n",
       "2  Schiffman-3     3761                    Ewing Sarcoma\n",
       "3  Schiffman-4     None                    None         \n",
       "4  Schiffman-5     None                    None         "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Diagnosis\n",
    "diagnosis_df = create_diagnosis_df(all_data_df)\n",
    "diagnosis_df.head()\n",
    "diagnosis_df = diagnosis_df.where((pd.notnull(diagnosis_df)), None)\n",
    "diagnosis_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>individual_name</th>\n",
       "      <th>age_at_diagnosis_(days)</th>\n",
       "      <th>morphology</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>hpo_id</th>\n",
       "      <th>observed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Schiffman-1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Ewing's Sarcoma</td>\n",
       "      <td>HP:0012254</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Schiffman-2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Ewing's Sarcoma</td>\n",
       "      <td>HP:0012254</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Schiffman-3</td>\n",
       "      <td>3761</td>\n",
       "      <td>Ewing Sarcoma</td>\n",
       "      <td>Ewing's Sarcoma</td>\n",
       "      <td>HP:0012254</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Schiffman-4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Ewing's Sarcoma</td>\n",
       "      <td>HP:0012254</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Schiffman-5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Ewing's Sarcoma</td>\n",
       "      <td>HP:0012254</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  individual_name age_at_diagnosis_(days)     morphology        phenotype  \\\n",
       "0  Schiffman-1     None                    None           Ewing's Sarcoma   \n",
       "1  Schiffman-2     None                    None           Ewing's Sarcoma   \n",
       "2  Schiffman-3     3761                    Ewing Sarcoma  Ewing's Sarcoma   \n",
       "3  Schiffman-4     None                    None           Ewing's Sarcoma   \n",
       "4  Schiffman-5     None                    None           Ewing's Sarcoma   \n",
       "\n",
       "       hpo_id observed  \n",
       "0  HP:0012254  yes      \n",
       "1  HP:0012254  yes      \n",
       "2  HP:0012254  yes      \n",
       "3  HP:0012254  yes      \n",
       "4  HP:0012254  yes      "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Phenotypes\n",
    "def create_phenotype_df(df):\n",
    "    df['phenotype'] = \"Ewing's Sarcoma\"\n",
    "    df['hpo_id'] = \"HP:0012254\"\n",
    "    df['observed'] = \"yes\"\n",
    "    return df\n",
    "phenotype_df = create_phenotype_df(diagnosis_df)\n",
    "phenotype_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/singhn4/Projects/kids_first/kf-api-dataservice/venv/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>unique</th>\n",
       "      <th>top</th>\n",
       "      <th>freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>individual_name</th>\n",
       "      <td>1203</td>\n",
       "      <td>1203</td>\n",
       "      <td>Schiffman-905</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ewing_trio_number</th>\n",
       "      <td>1203</td>\n",
       "      <td>485</td>\n",
       "      <td>Control Trio 4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relationship_to_proband</th>\n",
       "      <td>1203</td>\n",
       "      <td>4</td>\n",
       "      <td>Self/Case</td>\n",
       "      <td>485</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        count unique             top freq\n",
       "individual_name          1203  1203   Schiffman-905   1  \n",
       "ewing_trio_number        1203  485    Control Trio 4  3  \n",
       "relationship_to_proband  1203  4      Self/Case       485"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Family relationships\n",
    "df = all_data_df[['individual_name', 'relationship_to_proband', 'ewing_trio_number']]\n",
    "df.drop_duplicates('individual_name', inplace=True)\n",
    "df['ewing_trio_number'] = df['ewing_trio_number'].astype(str)\n",
    "fams = []\n",
    "# For each group\n",
    "for name, g in df.groupby('ewing_trio_number'):\n",
    "    fam = {}\n",
    "    # For each individual in family group\n",
    "    for idx, row in g.iterrows():\n",
    "        rel = row['relationship_to_proband']\n",
    "        if rel == 'Self/Case':\n",
    "            col = 'relative_id'\n",
    "        else:\n",
    "            col = 'participant_id'\n",
    "        fam[col] = row['individual_name']\n",
    "        fam['participant_to_relative_relation'] = rel\n",
    "    if rel != 'Self/Case':    \n",
    "        fams.append(fam)\n",
    "df = pd.DataFrame(fams)\n",
    "def func(row): return \"_\".join(['rel', str(row.name)])\n",
    "df['rel_id'] = df.apply(func, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Father</th>\n",
       "      <th>Mother</th>\n",
       "      <th>Self/Case</th>\n",
       "      <th>Sister</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Schiffman-1</td>\n",
       "      <td>Schiffman-2</td>\n",
       "      <td>Schiffman-3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Schiffman-5</td>\n",
       "      <td>Schiffman-4</td>\n",
       "      <td>Schiffman-6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Schiffman-8</td>\n",
       "      <td>Schiffman-7</td>\n",
       "      <td>Schiffman-9</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-10</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Schiffman-12</td>\n",
       "      <td>Schiffman-11</td>\n",
       "      <td>Schiffman-13</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-14</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Schiffman-16</td>\n",
       "      <td>Schiffman-15</td>\n",
       "      <td>Schiffman-17</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Schiffman-20</td>\n",
       "      <td>Schiffman-19</td>\n",
       "      <td>Schiffman-18</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Schiffman-21</td>\n",
       "      <td>Schiffman-22</td>\n",
       "      <td>Schiffman-23</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Schiffman-24</td>\n",
       "      <td>Schiffman-26</td>\n",
       "      <td>Schiffman-25</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-27</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Schiffman-28</td>\n",
       "      <td>Schiffman-29</td>\n",
       "      <td>Schiffman-30</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Schiffman-32</td>\n",
       "      <td>Schiffman-31</td>\n",
       "      <td>Schiffman-33</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Schiffman-35</td>\n",
       "      <td>Schiffman-34</td>\n",
       "      <td>Schiffman-36</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Schiffman-38</td>\n",
       "      <td>Schiffman-37</td>\n",
       "      <td>Schiffman-39</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-40</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Schiffman-973</td>\n",
       "      <td>Schiffman-974</td>\n",
       "      <td>Schiffman-41</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Schiffman-43</td>\n",
       "      <td>Schiffman-42</td>\n",
       "      <td>Schiffman-44</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Schiffman-45</td>\n",
       "      <td>Schiffman-46</td>\n",
       "      <td>Schiffman-47</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Schiffman-48</td>\n",
       "      <td>Schiffman-49</td>\n",
       "      <td>Schiffman-50</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Schiffman-53</td>\n",
       "      <td>Schiffman-52</td>\n",
       "      <td>Schiffman-51</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Schiffman-54</td>\n",
       "      <td>Schiffman-55</td>\n",
       "      <td>Schiffman-56</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Schiffman-59</td>\n",
       "      <td>Schiffman-58</td>\n",
       "      <td>Schiffman-57</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-60</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-61</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Schiffman-64</td>\n",
       "      <td>Schiffman-63</td>\n",
       "      <td>Schiffman-62</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-65</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Schiffman-68</td>\n",
       "      <td>Schiffman-66</td>\n",
       "      <td>Schiffman-67</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Schiffman-69</td>\n",
       "      <td>Schiffman-70</td>\n",
       "      <td>Schiffman-71</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Schiffman-73</td>\n",
       "      <td>Schiffman-74</td>\n",
       "      <td>Schiffman-72</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>455</th>\n",
       "      <td>Schiffman-1124</td>\n",
       "      <td>Schiffman-1123</td>\n",
       "      <td>Schiffman-1122</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>Schiffman-1125</td>\n",
       "      <td>Schiffman-1126</td>\n",
       "      <td>Schiffman-1127</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>Schiffman-1130</td>\n",
       "      <td>Schiffman-1128</td>\n",
       "      <td>Schiffman-1129</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>458</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-1131</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>459</th>\n",
       "      <td>Schiffman-1133</td>\n",
       "      <td>Schiffman-1132</td>\n",
       "      <td>Schiffman-1134</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-1135</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>461</th>\n",
       "      <td>Schiffman-1137</td>\n",
       "      <td>Schiffman-1136</td>\n",
       "      <td>Schiffman-1138</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>462</th>\n",
       "      <td>Schiffman-1140</td>\n",
       "      <td>Schiffman-1141</td>\n",
       "      <td>Schiffman-1139</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-1142</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464</th>\n",
       "      <td>Schiffman-1143</td>\n",
       "      <td>Schiffman-1144</td>\n",
       "      <td>Schiffman-1145</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>465</th>\n",
       "      <td>Schiffman-1146</td>\n",
       "      <td>Schiffman-1147</td>\n",
       "      <td>Schiffman-1148</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>466</th>\n",
       "      <td>Schiffman-1150</td>\n",
       "      <td>Schiffman-1151</td>\n",
       "      <td>Schiffman-1149</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>467</th>\n",
       "      <td>Schiffman-1153</td>\n",
       "      <td>Schiffman-1152</td>\n",
       "      <td>Schiffman-1154</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>468</th>\n",
       "      <td>Schiffman-1156</td>\n",
       "      <td>Schiffman-1157</td>\n",
       "      <td>Schiffman-1155</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>469</th>\n",
       "      <td>Schiffman-1158</td>\n",
       "      <td>Schiffman-1160</td>\n",
       "      <td>Schiffman-1159</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>Schiffman-1163</td>\n",
       "      <td>Schiffman-1161</td>\n",
       "      <td>Schiffman-1162</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>Schiffman-1166</td>\n",
       "      <td>Schiffman-1165</td>\n",
       "      <td>Schiffman-1164</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>Schiffman-1168</td>\n",
       "      <td>Schiffman-1167</td>\n",
       "      <td>Schiffman-1169</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>Schiffman-1171</td>\n",
       "      <td>Schiffman-1170</td>\n",
       "      <td>Schiffman-1172</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>Schiffman-1173</td>\n",
       "      <td>Schiffman-1174</td>\n",
       "      <td>Schiffman-1175</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>Schiffman-1176</td>\n",
       "      <td>Schiffman-1178</td>\n",
       "      <td>Schiffman-1177</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Schiffman-1179</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>477</th>\n",
       "      <td>Schiffman-1180</td>\n",
       "      <td>Schiffman-1181</td>\n",
       "      <td>Schiffman-1182</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>478</th>\n",
       "      <td>Schiffman-1184</td>\n",
       "      <td>Schiffman-1185</td>\n",
       "      <td>Schiffman-1183</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479</th>\n",
       "      <td>Schiffman-1186</td>\n",
       "      <td>Schiffman-1187</td>\n",
       "      <td>Schiffman-1188</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>480</th>\n",
       "      <td>Schiffman-1190</td>\n",
       "      <td>Schiffman-1189</td>\n",
       "      <td>Schiffman-1191</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>481</th>\n",
       "      <td>Schiffman-1194</td>\n",
       "      <td>Schiffman-1193</td>\n",
       "      <td>Schiffman-1192</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>482</th>\n",
       "      <td>Schiffman-1196</td>\n",
       "      <td>Schiffman-1195</td>\n",
       "      <td>Schiffman-1197</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>483</th>\n",
       "      <td>Schiffman-1199</td>\n",
       "      <td>Schiffman-1198</td>\n",
       "      <td>Schiffman-1200</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>Schiffman-1203</td>\n",
       "      <td>Schiffman-1202</td>\n",
       "      <td>Schiffman-1201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>485 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Father          Mother       Self/Case Sister\n",
       "0    Schiffman-1     Schiffman-2     Schiffman-3     NaN  \n",
       "1    Schiffman-5     Schiffman-4     Schiffman-6     NaN  \n",
       "2    Schiffman-8     Schiffman-7     Schiffman-9     NaN  \n",
       "3    NaN             NaN             Schiffman-10    NaN  \n",
       "4    Schiffman-12    Schiffman-11    Schiffman-13    NaN  \n",
       "5    NaN             NaN             Schiffman-14    NaN  \n",
       "6    Schiffman-16    Schiffman-15    Schiffman-17    NaN  \n",
       "7    Schiffman-20    Schiffman-19    Schiffman-18    NaN  \n",
       "8    Schiffman-21    Schiffman-22    Schiffman-23    NaN  \n",
       "9    Schiffman-24    Schiffman-26    Schiffman-25    NaN  \n",
       "10   NaN             NaN             Schiffman-27    NaN  \n",
       "11   Schiffman-28    Schiffman-29    Schiffman-30    NaN  \n",
       "12   Schiffman-32    Schiffman-31    Schiffman-33    NaN  \n",
       "13   Schiffman-35    Schiffman-34    Schiffman-36    NaN  \n",
       "14   Schiffman-38    Schiffman-37    Schiffman-39    NaN  \n",
       "15   NaN             NaN             Schiffman-40    NaN  \n",
       "16   Schiffman-973   Schiffman-974   Schiffman-41    NaN  \n",
       "17   Schiffman-43    Schiffman-42    Schiffman-44    NaN  \n",
       "18   Schiffman-45    Schiffman-46    Schiffman-47    NaN  \n",
       "19   Schiffman-48    Schiffman-49    Schiffman-50    NaN  \n",
       "20   Schiffman-53    Schiffman-52    Schiffman-51    NaN  \n",
       "21   Schiffman-54    Schiffman-55    Schiffman-56    NaN  \n",
       "22   Schiffman-59    Schiffman-58    Schiffman-57    NaN  \n",
       "23   NaN             NaN             Schiffman-60    NaN  \n",
       "24   NaN             NaN             Schiffman-61    NaN  \n",
       "25   Schiffman-64    Schiffman-63    Schiffman-62    NaN  \n",
       "26   NaN             NaN             Schiffman-65    NaN  \n",
       "27   Schiffman-68    Schiffman-66    Schiffman-67    NaN  \n",
       "28   Schiffman-69    Schiffman-70    Schiffman-71    NaN  \n",
       "29   Schiffman-73    Schiffman-74    Schiffman-72    NaN  \n",
       "..            ...             ...             ...    ...  \n",
       "455  Schiffman-1124  Schiffman-1123  Schiffman-1122  NaN  \n",
       "456  Schiffman-1125  Schiffman-1126  Schiffman-1127  NaN  \n",
       "457  Schiffman-1130  Schiffman-1128  Schiffman-1129  NaN  \n",
       "458  NaN             NaN             Schiffman-1131  NaN  \n",
       "459  Schiffman-1133  Schiffman-1132  Schiffman-1134  NaN  \n",
       "460  NaN             NaN             Schiffman-1135  NaN  \n",
       "461  Schiffman-1137  Schiffman-1136  Schiffman-1138  NaN  \n",
       "462  Schiffman-1140  Schiffman-1141  Schiffman-1139  NaN  \n",
       "463  NaN             NaN             Schiffman-1142  NaN  \n",
       "464  Schiffman-1143  Schiffman-1144  Schiffman-1145  NaN  \n",
       "465  Schiffman-1146  Schiffman-1147  Schiffman-1148  NaN  \n",
       "466  Schiffman-1150  Schiffman-1151  Schiffman-1149  NaN  \n",
       "467  Schiffman-1153  Schiffman-1152  Schiffman-1154  NaN  \n",
       "468  Schiffman-1156  Schiffman-1157  Schiffman-1155  NaN  \n",
       "469  Schiffman-1158  Schiffman-1160  Schiffman-1159  NaN  \n",
       "470  Schiffman-1163  Schiffman-1161  Schiffman-1162  NaN  \n",
       "471  Schiffman-1166  Schiffman-1165  Schiffman-1164  NaN  \n",
       "472  Schiffman-1168  Schiffman-1167  Schiffman-1169  NaN  \n",
       "473  Schiffman-1171  Schiffman-1170  Schiffman-1172  NaN  \n",
       "474  Schiffman-1173  Schiffman-1174  Schiffman-1175  NaN  \n",
       "475  Schiffman-1176  Schiffman-1178  Schiffman-1177  NaN  \n",
       "476  NaN             NaN             Schiffman-1179  NaN  \n",
       "477  Schiffman-1180  Schiffman-1181  Schiffman-1182  NaN  \n",
       "478  Schiffman-1184  Schiffman-1185  Schiffman-1183  NaN  \n",
       "479  Schiffman-1186  Schiffman-1187  Schiffman-1188  NaN  \n",
       "480  Schiffman-1190  Schiffman-1189  Schiffman-1191  NaN  \n",
       "481  Schiffman-1194  Schiffman-1193  Schiffman-1192  NaN  \n",
       "482  Schiffman-1196  Schiffman-1195  Schiffman-1197  NaN  \n",
       "483  Schiffman-1199  Schiffman-1198  Schiffman-1200  NaN  \n",
       "484  Schiffman-1203  Schiffman-1202  Schiffman-1201  NaN  \n",
       "\n",
       "[485 rows x 4 columns]"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create family relationships\n",
    "df = all_data_df[['individual_name', 'relationship_to_proband', 'ewing_trio_number']]\n",
    "family_dict = {}\n",
    "for idx, row in df.iterrows():\n",
    "    fam_id = row['ewing_trio_number']\n",
    "    if fam_id not in family_dict:\n",
    "        family_dict[fam_id] = {}\n",
    "    family_dict[fam_id][row['relationship_to_proband']] = row['individual_name']\n",
    "df = pd.DataFrame(list(family_dict.values()))\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
