{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "from dataservice.util.data_import.utils import (\n",
    "    dropna_rows_cols,\n",
    "    reformat_column_names,\n",
    "    read_json, \n",
    "    write_json\n",
    ")\n",
    "\n",
    "DATA_DIR = '/Users/singhn4/Projects/kids_first/data/Seidman_2015'\n",
    "DBGAP_DIR = os.path.join(DATA_DIR, 'dbgap')\n",
    "ALIQUOT_SHIP_DIR = os.path.join(DATA_DIR, 'manifests', 'shipping')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extraction - Methods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_study_file_data(filepaths=None):\n",
    "    \"\"\"\n",
    "    Read in raw study files\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filepaths = os.listdir(DBGAP_DIR)\n",
    "\n",
    "    study_files = [{\"study_file_name\": f}\n",
    "                   for f in filepaths if 'dbGaP' in f]\n",
    "    return pd.DataFrame(study_files)\n",
    "\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_study_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read study data\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR,\n",
    "                                'study.txt')\n",
    "    df = pd.read_csv(filepath)\n",
    "\n",
    "    return df\n",
    "\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_investigator_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read investigator data\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR,\n",
    "                                'investigator.txt')\n",
    "    df = pd.read_csv(filepath)\n",
    "\n",
    "    return df\n",
    "\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_family_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read family data for all participants\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DBGAP_DIR,\n",
    "                                '7a_dbGaP_PedigreeDS.txt')\n",
    "    df = pd.read_csv(filepath,\n",
    "                     delimiter='\\t',\n",
    "                     dtype={'SUBJID': str})\n",
    "    # Subset of columns\n",
    "    df.drop(['SEX'], axis=1, inplace=True)\n",
    "\n",
    "    # Add proband column\n",
    "    def func(row): return bool(row['MOTHER'] and row['FATHER'])\n",
    "    df['is_proband'] = df.apply(func, axis=1)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_phenotype_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read phenotype data\n",
    "    \"\"\"\n",
    "    # Read in cached phenotypes or create if they don't exist\n",
    "    hpo_fp = os.path.join(DATA_DIR, 'phenotype_hpo_mapping.txt')\n",
    "    if os.path.exists(hpo_fp):\n",
    "        return pd.read_csv(hpo_fp,dtype={'SUBJID': str})\n",
    "        \n",
    "    filepath = os.path.join(\n",
    "    DBGAP_DIR,\n",
    "    '3a_dbGaP_SubjectPhenotypes_ExtracardiacFindingsDS.txt')\n",
    "\n",
    "    # Read csv\n",
    "    df = pd.read_csv(filepath,\n",
    "                     delimiter='\\t',\n",
    "                     dtype={'SUBJID': str})\n",
    "\n",
    "    # Convert age years to days\n",
    "    df['LATEST_EXAM_AGE'] = df[\"LATEST_EXAM_AGE\"].apply(\n",
    "        lambda x: int(x) * 365)\n",
    "    age_at_event_days = df[['LATEST_EXAM_AGE', 'SUBJID']]\n",
    "\n",
    "    # Select string based phenotypes\n",
    "    df = df.select_dtypes(include='object')\n",
    "\n",
    "    # Make all values lower case\n",
    "    for col in df.columns.tolist():\n",
    "        df[col] = df[col].apply(lambda x: str(x).lower())\n",
    "\n",
    "    # Reshape to build the phenotypes df\n",
    "    cols = df.columns.tolist()[2:]\n",
    "    phenotype_cols = [col for col in cols if not col.startswith('OTHER')]\n",
    "    phenotype_df = pd.melt(df, id_vars='SUBJID', value_vars=phenotype_cols,\n",
    "                           var_name='phenotype', value_name='orig_observed')\n",
    "\n",
    "    # Remove unkonwns\n",
    "    unknown_values = ['none', 'unknown', 'no/not checked', 'not applicable', 'absent']\n",
    "    phenotype_df = phenotype_df[phenotype_df['orig_observed'].apply(lambda x: x not in unknown_values)]\n",
    "\n",
    "    # Add HPOs\n",
    "    from dataservice.util.data_import.etl.hpo import mapper\n",
    "    hpo_mapper = mapper.HPOMapper(DATA_DIR)\n",
    "    phenotype_df = hpo_mapper.add_hpo_id_col(phenotype_df)\n",
    "\n",
    "    # Map to positive/negative\n",
    "    def func(row): \n",
    "        return 'negative' if row['orig_observed'] == 'no' else 'positive'\n",
    "    phenotype_df['observed'] = phenotype_df.apply(func, axis=1)\n",
    "\n",
    "    # Merge back in age at event in days\n",
    "    phenotype_df = pd.merge(phenotype_df, age_at_event_days, on='SUBJID')\n",
    "\n",
    "    # Add unique col\n",
    "    def func(row): return \"_\".join(['phenotype', str(row.name)])\n",
    "    phenotype_df['phenotype_id'] = phenotype_df.apply(func, axis=1)\n",
    "    \n",
    "    # Write to file\n",
    "    phenotype_df.to_csv(hpo_fp, index=False)\n",
    "    \n",
    "    return phenotype_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>orig_observed</th>\n",
       "      <th>hpo_id</th>\n",
       "      <th>observed</th>\n",
       "      <th>latest_exam_age</th>\n",
       "      <th>phenotype_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>21296</td>\n",
       "      <td>HAND_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>21296</td>\n",
       "      <td>HAND_POLYDACTYLY</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0001161</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>21296</td>\n",
       "      <td>FEET_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>21296</td>\n",
       "      <td>CLUB_FOOT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>9653</td>\n",
       "      <td>DERMATOLOGICAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>9653</td>\n",
       "      <td>ENDOCRINOLOGIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>9653</td>\n",
       "      <td>ENDOCRINOLOGIC_ABNORMALITIES</td>\n",
       "      <td>congenital hypoglycemia</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>21474</td>\n",
       "      <td>EAR_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>4015</td>\n",
       "      <td>phenotype_56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>21474</td>\n",
       "      <td>HEARING_LOSS</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>4015</td>\n",
       "      <td>phenotype_57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>21474</td>\n",
       "      <td>AIRWAY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>4015</td>\n",
       "      <td>phenotype_62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>21474</td>\n",
       "      <td>TRACHEOESOPHAGEAL_FISTULA</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0002575</td>\n",
       "      <td>positive</td>\n",
       "      <td>4015</td>\n",
       "      <td>phenotype_63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>22358</td>\n",
       "      <td>DYSMORPHIC_FACIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>22358</td>\n",
       "      <td>EAR_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>22358</td>\n",
       "      <td>NECK_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>22358</td>\n",
       "      <td>MOUTH_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>22358</td>\n",
       "      <td>MICROGNATHIA</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0000347</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>22358</td>\n",
       "      <td>PALATE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>22358</td>\n",
       "      <td>CLEFT_LIP</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0410030</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>22358</td>\n",
       "      <td>RENAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>22358</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>22358</td>\n",
       "      <td>GOLDENHAR</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>7397</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>2920</td>\n",
       "      <td>phenotype_120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>7397</td>\n",
       "      <td>ASPLENIA</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0001746</td>\n",
       "      <td>positive</td>\n",
       "      <td>2920</td>\n",
       "      <td>phenotype_121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>7780</td>\n",
       "      <td>MOUTH_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>1460</td>\n",
       "      <td>phenotype_140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>7780</td>\n",
       "      <td>GENERAL_SKELETAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>1460</td>\n",
       "      <td>phenotype_149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>7780</td>\n",
       "      <td>CONGENITAL_SCOLIOSIS</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>1460</td>\n",
       "      <td>phenotype_150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>7780</td>\n",
       "      <td>FEET_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>1460</td>\n",
       "      <td>phenotype_152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>25102</td>\n",
       "      <td>DYSMORPHIC_FACIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>2920</td>\n",
       "      <td>phenotype_160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>25102</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>2920</td>\n",
       "      <td>phenotype_172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>25102</td>\n",
       "      <td>MALROTATION</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>2920</td>\n",
       "      <td>phenotype_173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7722</th>\n",
       "      <td>6478</td>\n",
       "      <td>PULMONARY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7723</th>\n",
       "      <td>6478</td>\n",
       "      <td>LUNG_HYPOPLASIA</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7734</th>\n",
       "      <td>6478</td>\n",
       "      <td>THYMUS_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7735</th>\n",
       "      <td>6478</td>\n",
       "      <td>THYMUS_ABNORMALITY</td>\n",
       "      <td>other</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7739</th>\n",
       "      <td>17249</td>\n",
       "      <td>DYSMORPHIC_FACIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7739</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7749</th>\n",
       "      <td>17249</td>\n",
       "      <td>PULMONARY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7750</th>\n",
       "      <td>17249</td>\n",
       "      <td>BILATERAL_LEFT_SIDEDNESS</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7751</th>\n",
       "      <td>17249</td>\n",
       "      <td>HEPATIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7751</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7752</th>\n",
       "      <td>17249</td>\n",
       "      <td>LEFT_SIDED_OR_MIDLINE_LIVER</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7753</th>\n",
       "      <td>17249</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7754</th>\n",
       "      <td>17249</td>\n",
       "      <td>POLYSPLENIA</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0001748</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7769</th>\n",
       "      <td>10168</td>\n",
       "      <td>EYE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>3285</td>\n",
       "      <td>phenotype_7769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7770</th>\n",
       "      <td>10168</td>\n",
       "      <td>ESOTROPIA</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0000565</td>\n",
       "      <td>positive</td>\n",
       "      <td>3285</td>\n",
       "      <td>phenotype_7770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7771</th>\n",
       "      <td>10168</td>\n",
       "      <td>STRABISMUS</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0000486</td>\n",
       "      <td>positive</td>\n",
       "      <td>3285</td>\n",
       "      <td>phenotype_7771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7772</th>\n",
       "      <td>10168</td>\n",
       "      <td>EAR_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>3285</td>\n",
       "      <td>phenotype_7772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7783</th>\n",
       "      <td>10168</td>\n",
       "      <td>GENITOURINARY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>3285</td>\n",
       "      <td>phenotype_7783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7784</th>\n",
       "      <td>10168</td>\n",
       "      <td>HYPOSPADIAS</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0000047</td>\n",
       "      <td>positive</td>\n",
       "      <td>3285</td>\n",
       "      <td>phenotype_7784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7785</th>\n",
       "      <td>10168</td>\n",
       "      <td>GENERAL_SKELETAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>3285</td>\n",
       "      <td>phenotype_7785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7788</th>\n",
       "      <td>10168</td>\n",
       "      <td>NEUROLOGICAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>3285</td>\n",
       "      <td>phenotype_7788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7807</th>\n",
       "      <td>21538</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7808</th>\n",
       "      <td>21538</td>\n",
       "      <td>POLYSPLENIA</td>\n",
       "      <td>yes</td>\n",
       "      <td>HP:0001748</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7823</th>\n",
       "      <td>19656</td>\n",
       "      <td>EYE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>5475</td>\n",
       "      <td>phenotype_7823</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7846</th>\n",
       "      <td>14477</td>\n",
       "      <td>DYSMORPHIC_FACIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7849</th>\n",
       "      <td>14477</td>\n",
       "      <td>EAR_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7850</th>\n",
       "      <td>14477</td>\n",
       "      <td>LOW_SET_EARS</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7851</th>\n",
       "      <td>14477</td>\n",
       "      <td>NOSE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7864</th>\n",
       "      <td>14477</td>\n",
       "      <td>FEET_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>0</td>\n",
       "      <td>phenotype_7864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7890</th>\n",
       "      <td>9631</td>\n",
       "      <td>NEUROLOGICAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>2555</td>\n",
       "      <td>phenotype_7890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7892</th>\n",
       "      <td>9631</td>\n",
       "      <td>ENDOCRINOLOGIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>yes</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>2555</td>\n",
       "      <td>phenotype_7892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7893</th>\n",
       "      <td>9631</td>\n",
       "      <td>ENDOCRINOLOGIC_ABNORMALITIES</td>\n",
       "      <td>hypoglycemia</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>2555</td>\n",
       "      <td>phenotype_7893</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1170 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     subjid                               phenotype            orig_observed  \\\n",
       "16    21296              HAND_ABNORMALITIES_PRESENT                      yes   \n",
       "17    21296                        HAND_POLYDACTYLY                      yes   \n",
       "18    21296              FEET_ABNORMALITIES_PRESENT                      yes   \n",
       "19    21296                               CLUB_FOOT                      yes   \n",
       "46     9653    DERMATOLOGICAL_ABNORMALITIES_PRESENT                      yes   \n",
       "47     9653    ENDOCRINOLOGIC_ABNORMALITIES_PRESENT                      yes   \n",
       "48     9653            ENDOCRINOLOGIC_ABNORMALITIES  congenital hypoglycemia   \n",
       "56    21474               EAR_ABNORMALITIES_PRESENT                      yes   \n",
       "57    21474                            HEARING_LOSS                      yes   \n",
       "62    21474            AIRWAY_ABNORMALITIES_PRESENT                      yes   \n",
       "63    21474               TRACHEOESOPHAGEAL_FISTULA                      yes   \n",
       "80    22358               DYSMORPHIC_FACIES_PRESENT                      yes   \n",
       "83    22358               EAR_ABNORMALITIES_PRESENT                      yes   \n",
       "85    22358              NECK_ABNORMALITIES_PRESENT                      yes   \n",
       "86    22358             MOUTH_ABNORMALITIES_PRESENT                      yes   \n",
       "87    22358                            MICROGNATHIA                      yes   \n",
       "88    22358            PALATE_ABNORMALITIES_PRESENT                      yes   \n",
       "89    22358                               CLEFT_LIP                      yes   \n",
       "95    22358             RENAL_ABNORMALITIES_PRESENT                      yes   \n",
       "105   22358               GENETIC_SYNDROMES_PRESENT                      yes   \n",
       "106   22358                               GOLDENHAR                      yes   \n",
       "120    7397         ABDOMINAL_ABNORMALITIES_PRESENT                      yes   \n",
       "121    7397                                ASPLENIA                      yes   \n",
       "140    7780             MOUTH_ABNORMALITIES_PRESENT                      yes   \n",
       "149    7780  GENERAL_SKELETAL_ABNORMALITIES_PRESENT                      yes   \n",
       "150    7780                    CONGENITAL_SCOLIOSIS                      yes   \n",
       "152    7780              FEET_ABNORMALITIES_PRESENT                      yes   \n",
       "160   25102               DYSMORPHIC_FACIES_PRESENT                      yes   \n",
       "172   25102         ABDOMINAL_ABNORMALITIES_PRESENT                      yes   \n",
       "173   25102                             MALROTATION                      yes   \n",
       "...     ...                                     ...                      ...   \n",
       "7722   6478         PULMONARY_ABNORMALITIES_PRESENT                      yes   \n",
       "7723   6478                         LUNG_HYPOPLASIA                      yes   \n",
       "7734   6478            THYMUS_ABNORMALITIES_PRESENT                      yes   \n",
       "7735   6478                      THYMUS_ABNORMALITY                    other   \n",
       "7739  17249               DYSMORPHIC_FACIES_PRESENT                      yes   \n",
       "7749  17249         PULMONARY_ABNORMALITIES_PRESENT                      yes   \n",
       "7750  17249                BILATERAL_LEFT_SIDEDNESS                      yes   \n",
       "7751  17249           HEPATIC_ABNORMALITIES_PRESENT                      yes   \n",
       "7752  17249             LEFT_SIDED_OR_MIDLINE_LIVER                      yes   \n",
       "7753  17249         ABDOMINAL_ABNORMALITIES_PRESENT                      yes   \n",
       "7754  17249                             POLYSPLENIA                      yes   \n",
       "7769  10168               EYE_ABNORMALITIES_PRESENT                      yes   \n",
       "7770  10168                               ESOTROPIA                      yes   \n",
       "7771  10168                              STRABISMUS                      yes   \n",
       "7772  10168               EAR_ABNORMALITIES_PRESENT                      yes   \n",
       "7783  10168     GENITOURINARY_ABNORMALITIES_PRESENT                      yes   \n",
       "7784  10168                             HYPOSPADIAS                      yes   \n",
       "7785  10168  GENERAL_SKELETAL_ABNORMALITIES_PRESENT                      yes   \n",
       "7788  10168      NEUROLOGICAL_ABNORMALITIES_PRESENT                      yes   \n",
       "7807  21538         ABDOMINAL_ABNORMALITIES_PRESENT                      yes   \n",
       "7808  21538                             POLYSPLENIA                      yes   \n",
       "7823  19656               EYE_ABNORMALITIES_PRESENT                      yes   \n",
       "7846  14477               DYSMORPHIC_FACIES_PRESENT                      yes   \n",
       "7849  14477               EAR_ABNORMALITIES_PRESENT                      yes   \n",
       "7850  14477                            LOW_SET_EARS                      yes   \n",
       "7851  14477              NOSE_ABNORMALITIES_PRESENT                      yes   \n",
       "7864  14477              FEET_ABNORMALITIES_PRESENT                      yes   \n",
       "7890   9631      NEUROLOGICAL_ABNORMALITIES_PRESENT                      yes   \n",
       "7892   9631    ENDOCRINOLOGIC_ABNORMALITIES_PRESENT                      yes   \n",
       "7893   9631            ENDOCRINOLOGIC_ABNORMALITIES             hypoglycemia   \n",
       "\n",
       "          hpo_id  observed  latest_exam_age    phenotype_id  \n",
       "16          None  positive                0    phenotype_16  \n",
       "17    HP:0001161  positive                0    phenotype_17  \n",
       "18          None  positive                0    phenotype_18  \n",
       "19          None  positive                0    phenotype_19  \n",
       "46          None  positive                0    phenotype_46  \n",
       "47          None  positive                0    phenotype_47  \n",
       "48          None  positive                0    phenotype_48  \n",
       "56          None  positive             4015    phenotype_56  \n",
       "57          None  positive             4015    phenotype_57  \n",
       "62          None  positive             4015    phenotype_62  \n",
       "63    HP:0002575  positive             4015    phenotype_63  \n",
       "80          None  positive                0    phenotype_80  \n",
       "83          None  positive                0    phenotype_83  \n",
       "85          None  positive                0    phenotype_85  \n",
       "86          None  positive                0    phenotype_86  \n",
       "87    HP:0000347  positive                0    phenotype_87  \n",
       "88          None  positive                0    phenotype_88  \n",
       "89    HP:0410030  positive                0    phenotype_89  \n",
       "95          None  positive                0    phenotype_95  \n",
       "105         None  positive                0   phenotype_105  \n",
       "106         None  positive                0   phenotype_106  \n",
       "120         None  positive             2920   phenotype_120  \n",
       "121   HP:0001746  positive             2920   phenotype_121  \n",
       "140         None  positive             1460   phenotype_140  \n",
       "149         None  positive             1460   phenotype_149  \n",
       "150         None  positive             1460   phenotype_150  \n",
       "152         None  positive             1460   phenotype_152  \n",
       "160         None  positive             2920   phenotype_160  \n",
       "172         None  positive             2920   phenotype_172  \n",
       "173         None  positive             2920   phenotype_173  \n",
       "...          ...       ...              ...             ...  \n",
       "7722        None  positive                0  phenotype_7722  \n",
       "7723        None  positive                0  phenotype_7723  \n",
       "7734        None  positive                0  phenotype_7734  \n",
       "7735        None  positive                0  phenotype_7735  \n",
       "7739        None  positive                0  phenotype_7739  \n",
       "7749        None  positive                0  phenotype_7749  \n",
       "7750        None  positive                0  phenotype_7750  \n",
       "7751        None  positive                0  phenotype_7751  \n",
       "7752        None  positive                0  phenotype_7752  \n",
       "7753        None  positive                0  phenotype_7753  \n",
       "7754  HP:0001748  positive                0  phenotype_7754  \n",
       "7769        None  positive             3285  phenotype_7769  \n",
       "7770  HP:0000565  positive             3285  phenotype_7770  \n",
       "7771  HP:0000486  positive             3285  phenotype_7771  \n",
       "7772        None  positive             3285  phenotype_7772  \n",
       "7783        None  positive             3285  phenotype_7783  \n",
       "7784  HP:0000047  positive             3285  phenotype_7784  \n",
       "7785        None  positive             3285  phenotype_7785  \n",
       "7788        None  positive             3285  phenotype_7788  \n",
       "7807        None  positive                0  phenotype_7807  \n",
       "7808  HP:0001748  positive                0  phenotype_7808  \n",
       "7823        None  positive             5475  phenotype_7823  \n",
       "7846        None  positive                0  phenotype_7846  \n",
       "7849        None  positive                0  phenotype_7849  \n",
       "7850        None  positive                0  phenotype_7850  \n",
       "7851        None  positive                0  phenotype_7851  \n",
       "7864        None  positive                0  phenotype_7864  \n",
       "7890        None  positive             2555  phenotype_7890  \n",
       "7892        None  positive             2555  phenotype_7892  \n",
       "7893        None  positive             2555  phenotype_7893  \n",
       "\n",
       "[1170 rows x 7 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = read_phenotype_data()\n",
    "phenos = df[df.observed == 'positive']\n",
    "phenos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>orig_observed</th>\n",
       "      <th>hpo_id</th>\n",
       "      <th>observed</th>\n",
       "      <th>latest_exam_age</th>\n",
       "      <th>phenotype_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phenotype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GENERAL_SKELETAL_ABNORMALITIES_PRESENT</th>\n",
       "      <td>66</td>\n",
       "      <td>66</td>\n",
       "      <td>0</td>\n",
       "      <td>66</td>\n",
       "      <td>66</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABDOMINAL_ABNORMALITIES_PRESENT</th>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DYSMORPHIC_FACIES_PRESENT</th>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAR_ABNORMALITIES_PRESENT</th>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EYE_ABNORMALITIES_PRESENT</th>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NEUROLOGICAL_ABNORMALITIES_PRESENT</th>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENITOURINARY_ABNORMALITIES_PRESENT</th>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RENAL_ABNORMALITIES_PRESENT</th>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HEAD_ABNORMALITIES_PRESENT</th>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AIRWAY_ABNORMALITIES_PRESENT</th>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHEST_ABNORMALITIES_PRESENT</th>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NECK_ABNORMALITIES_PRESENT</th>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MOUTH_ABNORMALITIES_PRESENT</th>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENETIC_SYNDROMES_PRESENT</th>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HAND_ABNORMALITIES_PRESENT</th>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HEPATIC_ABNORMALITIES_PRESENT</th>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PULMONARY_ABNORMALITIES_PRESENT</th>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOW_SET_EARS</th>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENDOCRINOLOGIC_ABNORMALITIES_PRESENT</th>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENDOCRINOLOGIC_ABNORMALITIES</th>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOSE_ABNORMALITIES_PRESENT</th>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WIDE_SPACED_NIPPLES</th>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASPLENIA</th>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRACHEOESOPHAGEAL_FISTULA</th>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DERMATOLOGICAL_ABNORMALITIES_PRESENT</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONGENITAL_SCOLIOSIS</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FEET_ABNORMALITIES_PRESENT</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SACRAL_DIMPLE</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MALROTATION</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LEFT_SIDED_OR_MIDLINE_LIVER</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CAFE_AU_LAIT_SPOTS</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BIFID_UVULA</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DUANE_ANOMALY</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONGENITAL_CYSTIC_ADENOMATOID_MALFORMATION</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SUBGLOTTIC_STENOSIS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TELANGIECTASIA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WIDE_SPACED_BIG_TOE</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONGENITAL_BLINDNESS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRACHEAL_STENOSIS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>URETERAL_ABNORMALITIES</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRISOMY_21_DOWN</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CATARACTS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUTIS_APLASIA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RH_ABO_INCOMPATIBLE</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BILIARY_ATRESIA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BILATERAL_LEFT_SIDEDNESS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HYDROCEPHALUS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHARGE</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HYPERPIGMENTED_LESIONS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EAR_PIT</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CERVICAL_INSTABILITY</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MACROCEPHALY</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MACROGLOSSIA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MICROPENIS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANTERIORLY_PLACED_ANUS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANNULAR_PANCREAS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABSENT_RADIUS</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PECTUS_CARINATUM</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POSTERIOR_EMBRYOTOXIN</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMBIGUOUS_GENITALIA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            subjid  orig_observed  hpo_id  \\\n",
       "phenotype                                                                   \n",
       "GENERAL_SKELETAL_ABNORMALITIES_PRESENT          66             66       0   \n",
       "ABDOMINAL_ABNORMALITIES_PRESENT                 62             62       0   \n",
       "DYSMORPHIC_FACIES_PRESENT                       58             58       0   \n",
       "EAR_ABNORMALITIES_PRESENT                       51             51       0   \n",
       "EYE_ABNORMALITIES_PRESENT                       46             46       0   \n",
       "NEUROLOGICAL_ABNORMALITIES_PRESENT              44             44       0   \n",
       "GENITOURINARY_ABNORMALITIES_PRESENT             42             42       0   \n",
       "RENAL_ABNORMALITIES_PRESENT                     39             39       0   \n",
       "HEAD_ABNORMALITIES_PRESENT                      32             32       0   \n",
       "AIRWAY_ABNORMALITIES_PRESENT                    30             30       0   \n",
       "CHEST_ABNORMALITIES_PRESENT                     25             25       0   \n",
       "NECK_ABNORMALITIES_PRESENT                      24             24       0   \n",
       "MOUTH_ABNORMALITIES_PRESENT                     24             24       0   \n",
       "GENETIC_SYNDROMES_PRESENT                       22             22       0   \n",
       "HAND_ABNORMALITIES_PRESENT                      21             21       0   \n",
       "HEPATIC_ABNORMALITIES_PRESENT                   20             20       0   \n",
       "PULMONARY_ABNORMALITIES_PRESENT                 20             20       0   \n",
       "LOW_SET_EARS                                    20             20       0   \n",
       "ENDOCRINOLOGIC_ABNORMALITIES_PRESENT            19             19       0   \n",
       "ENDOCRINOLOGIC_ABNORMALITIES                    19             19       0   \n",
       "NOSE_ABNORMALITIES_PRESENT                      19             19       0   \n",
       "WIDE_SPACED_NIPPLES                             19             19       0   \n",
       "ASPLENIA                                        18             18      18   \n",
       "TRACHEOESOPHAGEAL_FISTULA                       16             16      16   \n",
       "DERMATOLOGICAL_ABNORMALITIES_PRESENT            15             15       0   \n",
       "CONGENITAL_SCOLIOSIS                            15             15       0   \n",
       "FEET_ABNORMALITIES_PRESENT                      14             14       0   \n",
       "SACRAL_DIMPLE                                   13             13      13   \n",
       "MALROTATION                                     13             13       0   \n",
       "LEFT_SIDED_OR_MIDLINE_LIVER                     13             13       0   \n",
       "...                                            ...            ...     ...   \n",
       "CAFE_AU_LAIT_SPOTS                               2              2       0   \n",
       "BIFID_UVULA                                      2              2       2   \n",
       "DUANE_ANOMALY                                    2              2       2   \n",
       "CONGENITAL_CYSTIC_ADENOMATOID_MALFORMATION       1              1       0   \n",
       "SUBGLOTTIC_STENOSIS                              1              1       1   \n",
       "TELANGIECTASIA                                   1              1       1   \n",
       "WIDE_SPACED_BIG_TOE                              1              1       0   \n",
       "CONGENITAL_BLINDNESS                             1              1       1   \n",
       "TRACHEAL_STENOSIS                                1              1       1   \n",
       "URETERAL_ABNORMALITIES                           1              1       0   \n",
       "TRISOMY_21_DOWN                                  1              1       0   \n",
       "CATARACTS                                        1              1       0   \n",
       "CUTIS_APLASIA                                    1              1       0   \n",
       "RH_ABO_INCOMPATIBLE                              1              1       0   \n",
       "BILIARY_ATRESIA                                  1              1       1   \n",
       "BILATERAL_LEFT_SIDEDNESS                         1              1       0   \n",
       "HYDROCEPHALUS                                    1              1       1   \n",
       "CHARGE                                           1              1       0   \n",
       "HYPERPIGMENTED_LESIONS                           1              1       0   \n",
       "EAR_PIT                                          1              1       0   \n",
       "CERVICAL_INSTABILITY                             1              1       1   \n",
       "MACROCEPHALY                                     1              1       1   \n",
       "MACROGLOSSIA                                     1              1       1   \n",
       "MICROPENIS                                       1              1       1   \n",
       "ANTERIORLY_PLACED_ANUS                           1              1       1   \n",
       "ANNULAR_PANCREAS                                 1              1       1   \n",
       "ABSENT_RADIUS                                    1              1       1   \n",
       "PECTUS_CARINATUM                                 1              1       1   \n",
       "POSTERIOR_EMBRYOTOXIN                            1              1       0   \n",
       "AMBIGUOUS_GENITALIA                              1              1       1   \n",
       "\n",
       "                                            observed  latest_exam_age  \\\n",
       "phenotype                                                               \n",
       "GENERAL_SKELETAL_ABNORMALITIES_PRESENT            66               66   \n",
       "ABDOMINAL_ABNORMALITIES_PRESENT                   62               62   \n",
       "DYSMORPHIC_FACIES_PRESENT                         58               58   \n",
       "EAR_ABNORMALITIES_PRESENT                         51               51   \n",
       "EYE_ABNORMALITIES_PRESENT                         46               46   \n",
       "NEUROLOGICAL_ABNORMALITIES_PRESENT                44               44   \n",
       "GENITOURINARY_ABNORMALITIES_PRESENT               42               42   \n",
       "RENAL_ABNORMALITIES_PRESENT                       39               39   \n",
       "HEAD_ABNORMALITIES_PRESENT                        32               32   \n",
       "AIRWAY_ABNORMALITIES_PRESENT                      30               30   \n",
       "CHEST_ABNORMALITIES_PRESENT                       25               25   \n",
       "NECK_ABNORMALITIES_PRESENT                        24               24   \n",
       "MOUTH_ABNORMALITIES_PRESENT                       24               24   \n",
       "GENETIC_SYNDROMES_PRESENT                         22               22   \n",
       "HAND_ABNORMALITIES_PRESENT                        21               21   \n",
       "HEPATIC_ABNORMALITIES_PRESENT                     20               20   \n",
       "PULMONARY_ABNORMALITIES_PRESENT                   20               20   \n",
       "LOW_SET_EARS                                      20               20   \n",
       "ENDOCRINOLOGIC_ABNORMALITIES_PRESENT              19               19   \n",
       "ENDOCRINOLOGIC_ABNORMALITIES                      19               19   \n",
       "NOSE_ABNORMALITIES_PRESENT                        19               19   \n",
       "WIDE_SPACED_NIPPLES                               19               19   \n",
       "ASPLENIA                                          18               18   \n",
       "TRACHEOESOPHAGEAL_FISTULA                         16               16   \n",
       "DERMATOLOGICAL_ABNORMALITIES_PRESENT              15               15   \n",
       "CONGENITAL_SCOLIOSIS                              15               15   \n",
       "FEET_ABNORMALITIES_PRESENT                        14               14   \n",
       "SACRAL_DIMPLE                                     13               13   \n",
       "MALROTATION                                       13               13   \n",
       "LEFT_SIDED_OR_MIDLINE_LIVER                       13               13   \n",
       "...                                              ...              ...   \n",
       "CAFE_AU_LAIT_SPOTS                                 2                2   \n",
       "BIFID_UVULA                                        2                2   \n",
       "DUANE_ANOMALY                                      2                2   \n",
       "CONGENITAL_CYSTIC_ADENOMATOID_MALFORMATION         1                1   \n",
       "SUBGLOTTIC_STENOSIS                                1                1   \n",
       "TELANGIECTASIA                                     1                1   \n",
       "WIDE_SPACED_BIG_TOE                                1                1   \n",
       "CONGENITAL_BLINDNESS                               1                1   \n",
       "TRACHEAL_STENOSIS                                  1                1   \n",
       "URETERAL_ABNORMALITIES                             1                1   \n",
       "TRISOMY_21_DOWN                                    1                1   \n",
       "CATARACTS                                          1                1   \n",
       "CUTIS_APLASIA                                      1                1   \n",
       "RH_ABO_INCOMPATIBLE                                1                1   \n",
       "BILIARY_ATRESIA                                    1                1   \n",
       "BILATERAL_LEFT_SIDEDNESS                           1                1   \n",
       "HYDROCEPHALUS                                      1                1   \n",
       "CHARGE                                             1                1   \n",
       "HYPERPIGMENTED_LESIONS                             1                1   \n",
       "EAR_PIT                                            1                1   \n",
       "CERVICAL_INSTABILITY                               1                1   \n",
       "MACROCEPHALY                                       1                1   \n",
       "MACROGLOSSIA                                       1                1   \n",
       "MICROPENIS                                         1                1   \n",
       "ANTERIORLY_PLACED_ANUS                             1                1   \n",
       "ANNULAR_PANCREAS                                   1                1   \n",
       "ABSENT_RADIUS                                      1                1   \n",
       "PECTUS_CARINATUM                                   1                1   \n",
       "POSTERIOR_EMBRYOTOXIN                              1                1   \n",
       "AMBIGUOUS_GENITALIA                                1                1   \n",
       "\n",
       "                                            phenotype_id  \n",
       "phenotype                                                 \n",
       "GENERAL_SKELETAL_ABNORMALITIES_PRESENT                66  \n",
       "ABDOMINAL_ABNORMALITIES_PRESENT                       62  \n",
       "DYSMORPHIC_FACIES_PRESENT                             58  \n",
       "EAR_ABNORMALITIES_PRESENT                             51  \n",
       "EYE_ABNORMALITIES_PRESENT                             46  \n",
       "NEUROLOGICAL_ABNORMALITIES_PRESENT                    44  \n",
       "GENITOURINARY_ABNORMALITIES_PRESENT                   42  \n",
       "RENAL_ABNORMALITIES_PRESENT                           39  \n",
       "HEAD_ABNORMALITIES_PRESENT                            32  \n",
       "AIRWAY_ABNORMALITIES_PRESENT                          30  \n",
       "CHEST_ABNORMALITIES_PRESENT                           25  \n",
       "NECK_ABNORMALITIES_PRESENT                            24  \n",
       "MOUTH_ABNORMALITIES_PRESENT                           24  \n",
       "GENETIC_SYNDROMES_PRESENT                             22  \n",
       "HAND_ABNORMALITIES_PRESENT                            21  \n",
       "HEPATIC_ABNORMALITIES_PRESENT                         20  \n",
       "PULMONARY_ABNORMALITIES_PRESENT                       20  \n",
       "LOW_SET_EARS                                          20  \n",
       "ENDOCRINOLOGIC_ABNORMALITIES_PRESENT                  19  \n",
       "ENDOCRINOLOGIC_ABNORMALITIES                          19  \n",
       "NOSE_ABNORMALITIES_PRESENT                            19  \n",
       "WIDE_SPACED_NIPPLES                                   19  \n",
       "ASPLENIA                                              18  \n",
       "TRACHEOESOPHAGEAL_FISTULA                             16  \n",
       "DERMATOLOGICAL_ABNORMALITIES_PRESENT                  15  \n",
       "CONGENITAL_SCOLIOSIS                                  15  \n",
       "FEET_ABNORMALITIES_PRESENT                            14  \n",
       "SACRAL_DIMPLE                                         13  \n",
       "MALROTATION                                           13  \n",
       "LEFT_SIDED_OR_MIDLINE_LIVER                           13  \n",
       "...                                                  ...  \n",
       "CAFE_AU_LAIT_SPOTS                                     2  \n",
       "BIFID_UVULA                                            2  \n",
       "DUANE_ANOMALY                                          2  \n",
       "CONGENITAL_CYSTIC_ADENOMATOID_MALFORMATION             1  \n",
       "SUBGLOTTIC_STENOSIS                                    1  \n",
       "TELANGIECTASIA                                         1  \n",
       "WIDE_SPACED_BIG_TOE                                    1  \n",
       "CONGENITAL_BLINDNESS                                   1  \n",
       "TRACHEAL_STENOSIS                                      1  \n",
       "URETERAL_ABNORMALITIES                                 1  \n",
       "TRISOMY_21_DOWN                                        1  \n",
       "CATARACTS                                              1  \n",
       "CUTIS_APLASIA                                          1  \n",
       "RH_ABO_INCOMPATIBLE                                    1  \n",
       "BILIARY_ATRESIA                                        1  \n",
       "BILATERAL_LEFT_SIDEDNESS                               1  \n",
       "HYDROCEPHALUS                                          1  \n",
       "CHARGE                                                 1  \n",
       "HYPERPIGMENTED_LESIONS                                 1  \n",
       "EAR_PIT                                                1  \n",
       "CERVICAL_INSTABILITY                                   1  \n",
       "MACROCEPHALY                                           1  \n",
       "MACROGLOSSIA                                           1  \n",
       "MICROPENIS                                             1  \n",
       "ANTERIORLY_PLACED_ANUS                                 1  \n",
       "ANNULAR_PANCREAS                                       1  \n",
       "ABSENT_RADIUS                                          1  \n",
       "PECTUS_CARINATUM                                       1  \n",
       "POSTERIOR_EMBRYOTOXIN                                  1  \n",
       "AMBIGUOUS_GENITALIA                                    1  \n",
       "\n",
       "[117 rows x 6 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phenos.groupby(['phenotype']).count().sort_values(['subjid'], ascending=False)                             "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gender\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_gender_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read gender data for all subjects\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DBGAP_DIR,\n",
    "                                '3a_dbGaP_SubjectPhenotypes_GenderDS.txt')\n",
    "    df = pd.read_csv(filepath,\n",
    "                     delimiter='\\t',\n",
    "                     dtype={'SUBJID': str})\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Demographic\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_demographic_data(filepaths=None):\n",
    "    \"\"\"\n",
    "    Read demographic data for all subjects (child, mother, father)\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filenames = ['3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt',\n",
    "                     '3a_dbGaP_SubjectPhenotypes_MaternalDemographicsDS.txt',\n",
    "                     '3a_dbGaP_SubjectPhenotypes_PaternalDemographicsDS.txt']\n",
    "\n",
    "        filepaths = [os.path.join(DBGAP_DIR, filename)\n",
    "                     for filename in filenames\n",
    "                     ]\n",
    "\n",
    "    child_demo_df = pd.read_csv(os.path.join(filepaths[0]),\n",
    "                                delimiter='\\t',\n",
    "                                dtype={'SUBJID': str})\n",
    "\n",
    "    mother_demo_df = pd.read_csv(os.path.join(filepaths[1]),\n",
    "                                 delimiter='\\t',\n",
    "                                 dtype={'SUBJID': str})\n",
    "\n",
    "    father_demo_df = pd.read_csv(os.path.join(filepaths[2]),\n",
    "                                 delimiter='\\t',\n",
    "                                 dtype={'SUBJID': str})\n",
    "\n",
    "    # Combine demographics of all subjects\n",
    "    subject_demo_df = pd.concat(\n",
    "        [child_demo_df, mother_demo_df, father_demo_df])\n",
    "\n",
    "    subject_demo_df.drop_duplicates('SUBJID', inplace=True)\n",
    "    \n",
    "    # Subset of columns\n",
    "    subject_demo_df = subject_demo_df[['RACE', 'ETHNICITY', 'SUBJID']]\n",
    "\n",
    "    def func(row): return \"_\".join(['demographic', str(row.name)])\n",
    "    subject_demo_df['demographic_id'] = subject_demo_df.apply(func, axis=1)\n",
    "\n",
    "\n",
    "    return subject_demo_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Diagnosis\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_diagnosis_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read diagnoses data for all subjects\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filename = '3a_dbGaP_SubjectPhenotypes_PatientDiagnosisDS.txt'\n",
    "        filepath = os.path.join(DBGAP_DIR, filename)\n",
    "\n",
    "    diagnosis_df = pd.read_csv(filepath,\n",
    "                               delimiter='\\t',\n",
    "                               dtype={'SUBJID': str})\n",
    "\n",
    "    def func(row): return \"_\".join(['diagnosis', str(row.name)])\n",
    "    diagnosis_df['diagnosis_id'] = diagnosis_df.apply(func, axis=1)\n",
    "\n",
    "    return diagnosis_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sample\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_subject_sample_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read sample metadata for all subjects\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filename = '6a_dbGaP_SubjectSampleMappingDS.txt'\n",
    "        filepath = os.path.join(DBGAP_DIR, filename)\n",
    "\n",
    "    subject_sample_df = pd.read_csv(filepath,\n",
    "                                    delimiter='\\t',\n",
    "                                    dtype={'SUBJID': str})\n",
    "    subject_sample_df.drop_duplicates('SUBJID', inplace=True)\n",
    "\n",
    "    return subject_sample_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Aliquot\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_sample_shipping_manifest_data(*filepaths):\n",
    "    \"\"\"\n",
    "    Read shipping manifest for samples (from PI/sample source center)\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filepaths = [os.path.join(ALIQUOT_SHIP_DIR, filename)\n",
    "\n",
    "                     for filename in os.listdir(ALIQUOT_SHIP_DIR)\n",
    "                     ]\n",
    "\n",
    "    # Combine all manifest files\n",
    "    dfs = [pd.read_excel(filepath,\n",
    "                         delimiter='/t',\n",
    "                         dtype={'*barcode': str},\n",
    "                         skiprows=[0, 1],\n",
    "                         header=[6])\n",
    "\n",
    "           for filepath in filepaths\n",
    "\n",
    "           if os.path.basename(filepath).startswith(\"PCGC\")\n",
    "\n",
    "           ]\n",
    "    df = pd.concat(dfs)\n",
    "\n",
    "    # Rename columns\n",
    "    df.columns = map((lambda x: x.lower().lstrip(\"*\")), df.columns)\n",
    "    \n",
    "    # Subset of columns\n",
    "    df = df[['barcode',\n",
    "             'external_id',\n",
    "             'sample_collection_site',\n",
    "             'sample_role',\n",
    "             'concentration_ng_per_ul',\n",
    "             'initial_volume_microliters']]\n",
    "\n",
    "    # Drop rows where id cols are nan\n",
    "    id_cols = [col for col in df.columns if \"id\" in col]\n",
    "    df.dropna(subset=id_cols, inplace=True)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sequencing experiment (from read group metadata)\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_seq_experiment_data(filepath=None):\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR, \"seidman_metadata.xlsx\")\n",
    "\n",
    "    df = pd.read_excel(filepath, dtype={\"date\": str})\n",
    "    # Rename some columns\n",
    "    df.rename(columns={\"library_name (in original BAM header)\":\n",
    "                       \"library_name\",\n",
    "                       \"barcode\": \"rg_barcode\"}, inplace=True)\n",
    "    df[\"read_length\"] = df[\"read_length\"].apply(\n",
    "        lambda x: int(x.split(\"x\")[0]))\n",
    "    \n",
    "    # Create new columns\n",
    "    df['max_insert_size'] = df['insert_size'].max()\n",
    "    df['mean_insert_size'] = df['insert_size'].mean()\n",
    "    df['mean_read_length'] = df['read_length'].mean()\n",
    "    df['total_reads'] = df['read_length'].count()\n",
    "    \n",
    "    # Subset of columns\n",
    "    df = df[['sample_name',\n",
    "             'library_name',\n",
    "             'rg_barcode',\n",
    "             'run_name',\n",
    "             'read_length',\n",
    "             'date',\n",
    "             'library_strategy',\n",
    "             'library_source',\n",
    "             'library_selection',\n",
    "             'insert_size',\n",
    "             'instrument',\n",
    "             'library_layout',\n",
    "             'max_insert_size',\n",
    "             'mean_insert_size',\n",
    "             'mean_read_length',\n",
    "             'total_reads']]\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extraction - Execution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Study files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "study_files_df = read_study_file_data()\n",
    "study_files_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Family \n",
    "family_df = read_family_data()\n",
    "family_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Phenotypes\n",
    "phenotype_df = read_phenotype_data()\n",
    "phenotype_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gender\n",
    "gender_df = read_gender_data()\n",
    "gender_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Demographic\n",
    "demographic_df = read_demographic_data()\n",
    "demographic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Diagnosis data\n",
    "diagnosis_df = read_diagnosis_data()\n",
    "diagnosis_df.head()\n",
    "# diagnosis_df[diagnosis_df['subjid'] == '279']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sample data\n",
    "subject_sample_df = read_subject_sample_data()\n",
    "subject_sample_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Aliquot/Sample Shipping data\n",
    "shipping_manifest_df = read_sample_shipping_manifest_data()\n",
    "shipping_manifest_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sequencing experiments\n",
    "seq_exp_df = read_seq_experiment_data()\n",
    "seq_exp_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Explore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Participants\n",
    "print(\"Family\")\n",
    "print(family_df.nunique())\n",
    "print(\"\\nDemographics\")\n",
    "print(demographic_df.nunique())\n",
    "print(\"\\nGender\")\n",
    "print(demographic_df.nunique())\n",
    "print(\"\\nDiagnosis\")\n",
    "print(demographic_df.nunique())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Investigator\n",
    "investigator_df = read_investigator_data()\n",
    "investigator_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Study\n",
    "study_df = read_study_data()\n",
    "study_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Study files\n",
    "study_files_df = read_study_file_data()\n",
    "study_files_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Family\n",
    "family_df = read_family_data()\n",
    "family_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create participant df\n",
    "# Merge Gender + Demographics\n",
    "gender_demo_df = pd.merge(gender_df, demographic_df, on='subjid')\n",
    "# Add Family\n",
    "df1 = pd.merge(gender_demo_df, family_df, on='subjid')\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge Diagnosis\n",
    "df2 = pd.merge(df1, diagnosis_df, on='subjid')\n",
    "df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Merge Sample\n",
    "df3 = pd.merge(df2, subject_sample_df, on='subjid')\n",
    "df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Merge Aliquot\n",
    "df4 = pd.merge(df3, shipping_manifest_df, left_on='sampid', right_on='external_id')\n",
    "df4.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge Sequencing Experiment\n",
    "full_participant_df = pd.merge(df4, seq_exp_df, left_on='external_id', right_on='sample_name')\n",
    "full_participant_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create study\n",
    "study = {\n",
    "    'data_access_authority': 'dbGaP',\n",
    "    'study_id': 'phs001138',\n",
    "    'study_version': 'v1.p2',\n",
    "    'study_name': 'Discovery of the Genetic Basis of Structural Heart'\n",
    "    'and Other Birth Defects',\n",
    "    'attribution': 'https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/'\n",
    "    'GetAcknowledgementStatement.cgi?study_id=phs001138.v1.p2'\n",
    "}\n",
    "study_df = pd.DataFrame([study])\n",
    "study_df.to_csv(os.path.join(DATA_DIR, 'study.txt'))\n",
    "\n",
    "# Create investigator\n",
    "invest = {\n",
    "    'investigator_name': 'Christine E. Seidman',\n",
    "    'institution': 'Harvard Medical School'\n",
    "}\n",
    "inv_df = pd.DataFrame([invest])\n",
    "inv_df.to_csv(os.path.join(DATA_DIR, 'investigator.txt'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def _add_study_cols(study_df, df):\n",
    "    # Add study cols to a df\n",
    "    cols = study_df.columns.tolist()\n",
    "    row = study_df.iloc[0]\n",
    "    for col in cols:\n",
    "        df[col] = row[col]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add study to full participant df\n",
    "_add_study_cols(study_df, full_participant_df)\n",
    "\n",
    "# Add study to basic participant df\n",
    "participant_df = _add_study_cols(study_df, family_df)\n",
    "\n",
    "# Add study to investigator df\n",
    "study_investigator_df =_add_study_cols(study_df, investigator_df)\n",
    "\n",
    "# Add study to study files df\n",
    "study_study_files_df = _add_study_cols(study_df, study_files_df)\n",
    "\n",
    "# Phenotype df\n",
    "phenotype_participant_df = pd.merge(phenotype_df, participant_df,\n",
    "                                    on='subjid')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pprint import pprint\n",
    "n = 22\n",
    "chunk_size = 10\n",
    "entity_type ='participant'\n",
    "entities = ['{}_{}'.format(entity_type, j) for j in range(n)]\n",
    "for i in range(0, n, chunk_size):\n",
    "    chunk = entities[i - chunk_size:i]\n",
    "    if chunk:\n",
    "        start = i - chunk_size + 1\n",
    "        print('Adding {}:{} {}s to session'.format(start, i,\n",
    "                                                   entity_type))\n",
    "        pprint([e for e in entities[start:i]])\n",
    "        print('Flushing {} {}\\ns'.format(chunk_size, entity_type))\n",
    "\n",
    "print('Flushing remaining {} {}s to session\\n'.format(\n",
    "    len(entities[i:]) + 1, entity_type))\n",
    "\n",
    "remaining = entities[0:1] + entities[i:]\n",
    "\n",
    "pprint([e for e in remaining])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "def read_json(filepath):\n",
    "    with open(filepath, 'r') as json_file:\n",
    "        return json.load(json_file)\n",
    "\n",
    "\n",
    "def write_json(data, filepath):\n",
    "    with open(filepath, 'w') as json_file:\n",
    "        json.dump(data, json_file, sort_keys=True, indent=4, separators=(',', ':'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fp = os.path.join(DATA_DIR, 'genomic_file_uuid.json')\n",
    "file_json = read_json(fp)\n",
    "write_json(file_json, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Genomic file info df\n",
    "def read_genomic_file_info(filepath=None):\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR, 'genomic_file_uuid.json')\n",
    "        \n",
    "    def get_ext(fp):\n",
    "        filename = os.path.basename(fp)\n",
    "        parts = filename.split('.')\n",
    "        if len(parts) > 2:\n",
    "            ext = '.'.join(parts[1:])\n",
    "        else:\n",
    "            ext = parts[-1]\n",
    "        return ext\n",
    "\n",
    "    with open(filepath, 'r') as json_file:\n",
    "        uuid_dict = json.load(json_file)\n",
    "\n",
    "    gf_dicts = []\n",
    "    for k, v in uuid_dict.items():\n",
    "        file_info = {\n",
    "            'uuid': v['did'],\n",
    "            'md5sum': v['hashes']['md5'],\n",
    "            'file_url': v['urls'][0],\n",
    "            'file_size': v['size'],\n",
    "            'data_type': 'submitted aligned reads',\n",
    "            'file_format': get_ext(v['urls'][0]),\n",
    "            'file_name': os.path.basename(v['urls'][0])\n",
    "        }\n",
    "        gf_dicts.append(file_info)\n",
    "        \n",
    "    return pd.DataFrame(gf_dicts)\n",
    "\n",
    "def read_sample_gf_data(filepath=None):\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR, 'manifests', 'GMKF_BAMsampleIDs.xlsx')\n",
    "    df = pd.read_excel(filepath)\n",
    "    df = df.loc[df['Cohort'] == 'GMKF-Seidman']\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read genomic file info\n",
    "gf_file_info_df = read_genomic_file_info()\n",
    "# Sample and BAM File df\n",
    "sample_gf_df = read_sample_gf_data()\n",
    "# Merge with sequencing experiment df\n",
    "df1 = pd.merge(sample_gf_df, seq_exp_df, left_on='dbgap_subject_id', right_on='sample_name')\n",
    "# Merge with genomic file info df\n",
    "df2 = pd.merge(df1, gf_file_info_df, left_on='BAM sample ID', right_on='file_name')\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1.describe(include=['O']).T.sort_values('unique', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = read_genomic_file_info()\n",
    "print(df['file_size'].max()/1000000000)\n",
    "print(df['file_size'].min()/1000000000)\n",
    "print(df['file_size'].mean()/1000000000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = read_phenotype_data()\n",
    "df['observed'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('/Users/singhn4/Desktop/phenotype.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.where((pd.notnull(df)), None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
