{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "DATA_DIR = '/Users/singhn4/Projects/kids_first/data/Seidman_2015'\n",
    "DBGAP_DIR = os.path.join(DATA_DIR, 'dbgap')\n",
    "ALIQUOT_SHIP_DIR = os.path.join(DATA_DIR, 'manifests', 'shipping')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extraction - Methods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Helper functions\n",
    "def dropna_rows_cols(df_func):\n",
    "    \"\"\"\n",
    "    Decorator to drop rows and cols w all nan values\n",
    "    \"\"\"\n",
    "\n",
    "    def wrapper(*args, **kwargs):\n",
    "        df = df_func(*args, **kwargs)\n",
    "\n",
    "        # None or empty df\n",
    "        try:\n",
    "            if df.empty:\n",
    "                return df\n",
    "        except AttributeError:\n",
    "            return df\n",
    "\n",
    "        # Rows\n",
    "        df.dropna(how=\"all\", inplace=True)\n",
    "        # Cols\n",
    "        df.dropna(how=\"all\", axis=1, inplace=True)\n",
    "        return df\n",
    "\n",
    "    return wrapper\n",
    "\n",
    "def reformat_column_names(df_func):\n",
    "    \"\"\"\n",
    "    Decorator to reformat DataFrame column names.\n",
    "\n",
    "    Replace all column names having whitespace with underscore\n",
    "    and make lowercase\n",
    "    \"\"\"\n",
    "\n",
    "    def wrapper(*args, **kwargs):\n",
    "        df = df_func(*args, **kwargs)\n",
    "        # None or empty df\n",
    "        try:\n",
    "            if df.empty:\n",
    "                return df\n",
    "        except AttributeError:\n",
    "            return df\n",
    "        df.columns = map((lambda x: x.replace(\" \", \"_\").lower()),\n",
    "                         df.columns)\n",
    "        return df\n",
    "\n",
    "    return wrapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_study_file_data(filepaths=None):\n",
    "    \"\"\"\n",
    "    Read in raw study files\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filepaths = os.listdir(DBGAP_DIR)\n",
    "\n",
    "    study_files = [{\"study_file_name\": f}\n",
    "                   for f in filepaths if 'dbGaP' in f]\n",
    "    return pd.DataFrame(study_files)\n",
    "\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_study_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read study data\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR,\n",
    "                                'study.txt')\n",
    "    df = pd.read_csv(filepath)\n",
    "\n",
    "    return df\n",
    "\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_investigator_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read investigator data\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR,\n",
    "                                'investigator.txt')\n",
    "    df = pd.read_csv(filepath)\n",
    "\n",
    "    return df\n",
    "\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_family_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read family data for all participants\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DBGAP_DIR,\n",
    "                                '7a_dbGaP_PedigreeDS.txt')\n",
    "    df = pd.read_csv(filepath,\n",
    "                     delimiter='\\t',\n",
    "                     dtype={'SUBJID': str})\n",
    "    # Subset of columns\n",
    "    df.drop(['SEX'], axis=1, inplace=True)\n",
    "\n",
    "    # Add proband column\n",
    "    def func(row): return bool(row['MOTHER'] and row['FATHER'])\n",
    "    df['is_proband'] = df.apply(func, axis=1)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['No' 'Yes']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'No' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'abnormal fetal calvarium'\n",
      " 'Dolichocephaly with bony synostosis of sagittal structure.'\n",
      " 'dolichochocehaly' 'unilateral coronal synostosis']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'frontal bossing (occipital protuberance)'\n",
      " 'cephalohematoma, Caput Succedaneum'\n",
      " 'short, sloping forehead with prominent occiput' 'bitemporal hallowing'\n",
      " 'small anterior fontanelle' 'congenital scar from eye to scalp per Mom'\n",
      " 'brachycephaly' 'parietal encephalocele, plagiocephaly'\n",
      " 'overriding coronal sutures'\n",
      " 'medical record indicates that she had a cyst in her brain'\n",
      " 'narrow forehead' 'metopic prominence' 'plagiocephaly' 'Plagiocephaly'\n",
      " 'sloped forehead, narrow' 'flat face' '3 fontanelles'\n",
      " 'Plagiocephaly, triangular shaped in coronal plane' 'bitemporal narrowing']\n",
      "['None' 'No' 'Yes']\n",
      "['Unknown' 'None' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Glaucoma' 'myopia' 'intraretinal hemorrhages'\n",
      " 'mildly proptotic eyes' 'bilateral low vision' 'sclerae anicteric'\n",
      " 'downslanting' 'Exotropia' 'short palpebral fissures'\n",
      " 'left microphthalmia' 'proptotic, long eyelashes' 'Belpharoptosis'\n",
      " 'Microphthalmia and cloudy corneas'\n",
      " 'prominant eyes, long eyelashes, midface hypoplasia' 'epibulbar dermoid'\n",
      " 'downslanting palepbral fissures' 'hypertelorism' 'mild ROP OU'\n",
      " 'upslanting palpebrals' 'slightly upslanted palpebral fissures'\n",
      " 'palpebral fissures' 'eye deviation\"\" per  note' 'Amblyopia'\n",
      " 'Hypotelorism, downslanted palpebral fissures'\n",
      " 'hypotelorism, downslanting papebral fissures' 'amblyopia'\n",
      " 'Born with retinoblastoma']\n",
      "['None' 'No' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Atresia of left external ear'\n",
      " 'R ear overfolded and posteriorly rotated; L ear slightly pointed'\n",
      " 'cupped ears' 'pointy superiorly' 'small' 'slight redundancy in helices'\n",
      " 'rail road track deformity' 'Bilateral glue ear'\n",
      " 'overriding helices and relatively small' 'squared helix'\n",
      " 'Bilateral Congenital Ossicular Anomaly' 'prominent'\n",
      " 'R ear long and underdeveloped' 'posteriorly rotated'\n",
      " 'Underdeveloped ears' 'pointy' 'enlarged' 'overfolded superior helices'\n",
      " 'Left ear microtia' 'Posteriorly rotated' 'r ear is smaller in size'\n",
      " 'posteriorly set' 'thick helices' 'congenital deafness, R pinna cupped.']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'No' 'Yes']\n",
      "['Unknown' 'None' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Torticollis' 'torticollis' 'short neck'\n",
      " 'neck bent to one side;needed PT' 'short' 'cystic hygroma' 'mildly short'\n",
      " 'short, thick' 'Short']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Short frenulum, hypoplastic left mandible' 'tongue-tied' 'thrush'\n",
      " 'ankyloglossia' 'asymmetric' 'small' 'thin upper lip'\n",
      " 'small mouth,thin upper lip' 'missing teeth'\n",
      " 'small maxilla/mandible on right side' 'thin lips, short philtrum'\n",
      " 'Asymmetric mouth movement; hypoplastic left jaw']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'high palate' 'high arched' 'high arched and narrowed'\n",
      " 'mildly high-arched']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'stridor, left vocal cord paralysis' 'esophagitis.  G-tube.'\n",
      " 'hypoplastic right side, anteriorly displaced' 'Asthma'\n",
      " 'esophageal reflux']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['None' 'sternum asymmetry']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'asthma' 'scimitar syndrome' 'small lungs'\n",
      " 'pulmonary lymphangiectasia' 'pulmonary hypertension'\n",
      " 'hypoplasia L mainstem bronchus'\n",
      " 'pulmonary hypertension in newborn period'\n",
      " 'CHRONIC LUNG DISEASE-PREMATURITY'\n",
      " 'Frequent episodes of migratory atelectasis'\n",
      " 'reversed pulmonary anatomy (L-sided on R, R on L)'\n",
      " 'inverted bronchial pattern']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' '3 cm. liver' 'enlarged tranverse liver' 'Liver hepatomegaly'\n",
      " 'infantile hepatic hemangioma' 'indirect hyperbilirubinemia'\n",
      " 'portacaval shunt']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'heterotaxy' 'Gastroesophageal reflux'\n",
      " 'Protein-losing enteropathy: Celiac confirmed' 'cecal carcinoma'\n",
      " 'deformity of the right anterolateral abdominal wall' '2 vessel cord'\n",
      " '2 vessel umbilical cord' 'pancreas not seen on ultrasound'\n",
      " 'Hernia, type unk' 'stricture or obstruction and medical NEC'\n",
      " 'Moragni hernia' 'heterotaxy syndrome' 'ileal atresia'\n",
      " 'Protein Losing Enteropathy' 'hernia x2 type unk' 'right-sided stomach'\n",
      " 'rt-sided stomach' 'esophageal atresai' 'cholecystectomy  1994'\n",
      " 'rt-sided stomach & rt-sided multi-lobulated spleen'\n",
      " 'Atresia of the large intestine' 'Pancreatitis' 'GERD' 'situs inversus']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Pyelectasis' 'Hypoplastic'\n",
      " 'non-functioning Rt. kidney, obstruction of bladder outlet, hydroureteral nephrosis of the Lt. kidney'\n",
      " 'L ureterocele with absent R kidney' 'R and L kidneys pelvicaliectasis'\n",
      " 'left pyelectasis' 'double left renal collecting systems and ureters'\n",
      " 'double kidney' 'mild upper pole caliectasis'\n",
      " 'ureteropelvic junction obstruction' 'Chronic renal insufficiency'\n",
      " 'Small right kidney' 'R KIDNEY IS ABSENT' 'hyperechoic renal cortex'\n",
      " 'ureterocele, Right vesicoureteric reflux, grade 1' 'pelviectasis'\n",
      " 'rectourethral fistula' 'renal agenesis' 'hypoplasia of kidney']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'meatus stenosis' 'chordee' 'phimosis' 'rectovaginal fistula'\n",
      " 'hydrocele' 'labial adhesions' 'Hydrocele' 'high riding testicles'\n",
      " 'neurogenic bladder' 'ventral displacement of meatus' 'hydroceles'\n",
      " 'rectovestibular fistula' 'hypogonadotropic hypogonadism'\n",
      " 'varicoceles and phimosis.' 'prominent labia']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'spina bifida'\n",
      " 'asymmetric sacral cleft, low lying conus terminalis L3' 'hemi-vertebrae'\n",
      " 'Klippel-Feil syndrome'\n",
      " 'sacral agenesis with absent coccyx and missing vertebrae at S1'\n",
      " 'abnormal cervical and thoracic bodies' 'absent sacrum and lumbar spine'\n",
      " 'incompletely formed T10' 'segmental abnormality of sacrum'\n",
      " 'left sided hemivertebrae'\n",
      " 'segmental anomalies involving upper thoracic spine']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Thoracic spinal cord syrinx' 'Congenital hemivertebra'\n",
      " 'stiff musculature' 'thickened rib, mild scoliosis, difuse osteopenia'\n",
      " 'Osteopenia (updated )' 'Spina bifida' 'tibial torsion,'\n",
      " 'Right sided Sternocleidomastoid Contracture' 'short stature'\n",
      " 'shortened long bones' 'asymmetrical gluteal crease'\n",
      " 'underdeveloped pelvis, lower extremity contractures'\n",
      " 'low lying conus at S1-2 w/ fatty filum'\n",
      " 'short achilles tendon on one side'\n",
      " 'leg length discrepancy and short stature' 'short sternum'\n",
      " 'mild ligamentous laxity' 'shallow acetabula' 'idiopathic short stature'\n",
      " 'sacral hemangioma'\n",
      " 'no humerus, radius/ulna or hand bones on the left side'\n",
      " 'hypotonia lower extremities']\n",
      "['None' 'Yes' 'No']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'Yes' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Clinodactyly, right hand' 'mildly hypoplastic proximal thumbs'\n",
      " 'clinodactyly 5th finger' 'hypoplastic left thumb' 'tapering fingers'\n",
      " 'left transverse crease'\n",
      " 'bilateral clinodactyly 5th fingers, left hand simian creases'\n",
      " 'decreased thenar eminence' 'clubbing'\n",
      " 'skintags after the 5th digit on each hand.'\n",
      " 'underdeveloped thenar prominence'\n",
      " 'distal limb deformities and radial anomaly'\n",
      " 'Left hand single palmar crease, clinodactyly'\n",
      " 'no humerus, radius/ulna or hand bones on the left side']\n",
      "['None' 'Yes' 'No']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'Yes' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'short toe bones 4th/5th digits' 'Broad big toe'\n",
      " 'metatarsus adductus bilaterally' 'bilat toes 2-4 syndactyly'\n",
      " 'clindactyly of fifth digits, L partial simian crease' 'Tarsal Coalition'\n",
      " 'extra digit bilat' 'overlapping toes on right: 4th overlaps 5th'\n",
      " 'hallux valgus']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'learning disability, ADHD' 'LS Myelomeningocele' 'seizures'\n",
      " 'monoparesis of right arm' 'developmental & learning issues' 'PVL'\n",
      " 'Gross motor delay' 'Hypertonic' 'choroid plexus cyst'\n",
      " 'Assymetric crying facies' 'anxiety disorder' 'Seizure disorder.'\n",
      " 'Agenesis of corpus callosum' 'cerebral atrophy' 'low muscle tone'\n",
      " 'hamartoma' 'sinus dermal tract'\n",
      " 'pericallsol lipoma; splenium of corpus callosum'\n",
      " 'severe developmental delay'\n",
      " 'severe developmental delay, non-verbal, autistic, history of seizures'\n",
      " 'Congenital Bulbar Palsy' 'parietal encephalocele'\n",
      " 'global developmental delay, hypotonia' 'Caudal regression syndrome'\n",
      " 'hypotonia' 'muscle tone' 'mild global developmental delay'\n",
      " 'hypotonia, pt with head lag, cannot sit on her own' 'severe seizures'\n",
      " 'hypoplastic corpus callosum, vermis hypoplasia'\n",
      " 'seizures related to hypocalcemia' 'polymicrogyria' 'HORNER SYNDROME'\n",
      " 'seizure disorder' 'stroke and seizure per physician note '\n",
      " 'tethered cord malformation sequence' 'periventricular leukomalacia'\n",
      " 'subject has CP.' 'hemiplegic cerebral palsy; abnormal gait']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'skin tag on lower part of sternum' '3 facial skin tags'\n",
      " 'left sided auricular skin tags as well as one tag on left cheek.'\n",
      " 'very dry skin\"\"' 'anal skin tag']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'congenital hypoglycemia' 'Hypothyroidism, Vit D deficiency'\n",
      " 'hypothyroid' 'late onset of puberty/growth hormone deficiency'\n",
      " 'Hypothyroidism (update )' 'congenital hypothyroidism' 'short stature'\n",
      " 'Hypothyroidism' 'central hypothyrodism' 'hypercholesterolemia'\n",
      " 'hypothyroidism' 'hypothyroidism, hyperbilirubinemia'\n",
      " 'hypoglycemia, hypoparathyroidism'\n",
      " 'delayed puberty-no facial hair,pubic or underarm hair.'\n",
      " 'low growth factor/idiopathic short stature' 'pheochromocytoma   surgery '\n",
      " 'hypoglycemia']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'Hypoplastic' 'Absent' 'Other']\n",
      "['None' 'very small']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'anemia' 'Factor V Leiden' 'Factor XII deficiency'\n",
      " 'thrombocytopenia' 'ANEMIA OF PREMATURITY'\n",
      " 'Factor V Leiden heterozygous, MTHFR heterozygous' 'hypogammaglobulinemia'\n",
      " 'decrease vit K and Factor XII deficiency']\n",
      "['No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['None']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Homozygous familial hypercholesterolemia'\n",
      " 'alpha and beta Thalassemia trait']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'Immune Globulin deficiency for unknown reasons'\n",
      " 'Exposed to toxoplasma in utero - born negative'\n",
      " 'juvenile rheumatoid arthritis']\n"
     ]
    }
   ],
   "source": [
    "# Find the potential negative/positive observed vaue\n",
    "from pandas.api.types import is_numeric_dtype\n",
    "filepath = os.path.join(DBGAP_DIR, '3a_dbGaP_SubjectPhenotypes_ExtracardiacFindingsDS.txt')\n",
    "\n",
    "# Read csv\n",
    "df = pd.read_csv(filepath,\n",
    "             delimiter='\\t',\n",
    "             dtype={'SUBJID': str})\n",
    "\n",
    "cols = df.columns.tolist()[1:]\n",
    "for col in cols:\n",
    "    if not is_numeric_dtype(df[col]):\n",
    "        print(df[col].unique())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_phenotype_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read phenotype data\n",
    "    \"\"\"\n",
    "    filepath = os.path.join(\n",
    "    DBGAP_DIR,\n",
    "    '3a_dbGaP_SubjectPhenotypes_ExtracardiacFindingsDS.txt')\n",
    "\n",
    "    # Read csv\n",
    "    df = pd.read_csv(filepath,\n",
    "                     delimiter='\\t',\n",
    "                     dtype={'SUBJID': str})\n",
    "\n",
    "    # Convert age years to days\n",
    "    age_at_event_days = df[\"LATEST_EXAM_AGE\"].apply(\n",
    "        lambda x: float(x) * 365)\n",
    "\n",
    "    # Select string based phenotypes\n",
    "    df = df.select_dtypes(include='object')\n",
    "\n",
    "    # Make all values lower case\n",
    "    for col in df.columns.tolist():\n",
    "        df[col] = df[col].apply(lambda x: str(x).lower())\n",
    "\n",
    "    # Reshape to build the phenotypes df\n",
    "    cols = df.columns.tolist()[2:]\n",
    "    phenotype_cols = [col for col in cols if not col.startswith('OTHER')]\n",
    "    phenotype_df = pd.melt(df, id_vars='SUBJID', value_vars=phenotype_cols,\n",
    "                           var_name='phenotype', value_name='observed')\n",
    "\n",
    "    phenotype_df = pd.concat([age_at_event_days, phenotype_df], axis=1)\n",
    "\n",
    "    # Merge with HPO mapping\n",
    "    mapping_filepath = os.path.join(DATA_DIR, 'phenotype_hpo_mapping.txt')\n",
    "    if os.path.isfile(mapping_filepath):\n",
    "        hpo_df = pd.read_csv(mapping_filepath)\n",
    "        new_df = pd.merge(hpo_df, phenotype_df, on='phenotype')\n",
    "        new_df.rename(columns={\"Yes\": \"hpo_id\"}, inplace=True)\n",
    "        phenotype_df = new_df[['hpo_id', 'phenotype', 'observed', 'LATEST_EXAM_AGE', 'SUBJID']]\n",
    "\n",
    "    def func(row): return \"_\".join(['phenotype', str(row.name)])\n",
    "    phenotype_df['phenotype_id'] = phenotype_df.apply(func, axis=1)\n",
    "    \n",
    "    return phenotype_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gender\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_gender_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read gender data for all subjects\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DBGAP_DIR,\n",
    "                                '3a_dbGaP_SubjectPhenotypes_GenderDS.txt')\n",
    "    df = pd.read_csv(filepath,\n",
    "                     delimiter='\\t',\n",
    "                     dtype={'SUBJID': str})\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Demographic\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_demographic_data(filepaths=None):\n",
    "    \"\"\"\n",
    "    Read demographic data for all subjects (child, mother, father)\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filenames = ['3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt',\n",
    "                     '3a_dbGaP_SubjectPhenotypes_MaternalDemographicsDS.txt',\n",
    "                     '3a_dbGaP_SubjectPhenotypes_PaternalDemographicsDS.txt']\n",
    "\n",
    "        filepaths = [os.path.join(DBGAP_DIR, filename)\n",
    "                     for filename in filenames\n",
    "                     ]\n",
    "\n",
    "    child_demo_df = pd.read_csv(os.path.join(filepaths[0]),\n",
    "                                delimiter='\\t',\n",
    "                                dtype={'SUBJID': str})\n",
    "\n",
    "    mother_demo_df = pd.read_csv(os.path.join(filepaths[1]),\n",
    "                                 delimiter='\\t',\n",
    "                                 dtype={'SUBJID': str})\n",
    "\n",
    "    father_demo_df = pd.read_csv(os.path.join(filepaths[2]),\n",
    "                                 delimiter='\\t',\n",
    "                                 dtype={'SUBJID': str})\n",
    "\n",
    "    # Combine demographics of all subjects\n",
    "    subject_demo_df = pd.concat(\n",
    "        [child_demo_df, mother_demo_df, father_demo_df])\n",
    "\n",
    "    subject_demo_df.drop_duplicates('SUBJID', inplace=True)\n",
    "    \n",
    "    # Subset of columns\n",
    "    subject_demo_df = subject_demo_df[['RACE', 'ETHNICITY', 'SUBJID']]\n",
    "\n",
    "    def func(row): return \"_\".join(['demographic', str(row.name)])\n",
    "    subject_demo_df['demographic_id'] = subject_demo_df.apply(func, axis=1)\n",
    "\n",
    "\n",
    "    return subject_demo_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Diagnosis\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_diagnosis_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read diagnoses data for all subjects\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filename = '3a_dbGaP_SubjectPhenotypes_PatientDiagnosisDS.txt'\n",
    "        filepath = os.path.join(DBGAP_DIR, filename)\n",
    "\n",
    "    diagnosis_df = pd.read_csv(filepath,\n",
    "                               delimiter='\\t',\n",
    "                               dtype={'SUBJID': str})\n",
    "\n",
    "    def func(row): return \"_\".join(['diagnosis', str(row.name)])\n",
    "    diagnosis_df['diagnosis_id'] = diagnosis_df.apply(func, axis=1)\n",
    "\n",
    "    return diagnosis_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sample\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_subject_sample_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read sample metadata for all subjects\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filename = '6a_dbGaP_SubjectSampleMappingDS.txt'\n",
    "        filepath = os.path.join(DBGAP_DIR, filename)\n",
    "\n",
    "    subject_sample_df = pd.read_csv(filepath,\n",
    "                                    delimiter='\\t',\n",
    "                                    dtype={'SUBJID': str})\n",
    "    subject_sample_df.drop_duplicates('SUBJID', inplace=True)\n",
    "\n",
    "    return subject_sample_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Aliquot\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_sample_shipping_manifest_data(*filepaths):\n",
    "    \"\"\"\n",
    "    Read shipping manifest for samples (from PI/sample source center)\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filepaths = [os.path.join(ALIQUOT_SHIP_DIR, filename)\n",
    "\n",
    "                     for filename in os.listdir(ALIQUOT_SHIP_DIR)\n",
    "                     ]\n",
    "\n",
    "    # Combine all manifest files\n",
    "    dfs = [pd.read_excel(filepath,\n",
    "                         delimiter='/t',\n",
    "                         dtype={'*barcode': str},\n",
    "                         skiprows=[0, 1],\n",
    "                         header=[6])\n",
    "\n",
    "           for filepath in filepaths\n",
    "\n",
    "           if os.path.basename(filepath).startswith(\"PCGC\")\n",
    "\n",
    "           ]\n",
    "    df = pd.concat(dfs)\n",
    "\n",
    "    # Rename columns\n",
    "    df.columns = map((lambda x: x.lower().lstrip(\"*\")), df.columns)\n",
    "    \n",
    "    # Subset of columns\n",
    "    df = df[['barcode',\n",
    "             'external_id',\n",
    "             'sample_collection_site',\n",
    "             'sample_role',\n",
    "             'concentration_ng_per_ul',\n",
    "             'initial_volume_microliters']]\n",
    "\n",
    "    # Drop rows where id cols are nan\n",
    "    id_cols = [col for col in df.columns if \"id\" in col]\n",
    "    df.dropna(subset=id_cols, inplace=True)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sequencing experiment (from read group metadata)\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_seq_experiment_data(filepath=None):\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR, \"seidman_metadata.xlsx\")\n",
    "\n",
    "    df = pd.read_excel(filepath, dtype={\"date\": str})\n",
    "    # Rename some columns\n",
    "    df.rename(columns={\"library_name (in original BAM header)\":\n",
    "                       \"library_name\",\n",
    "                       \"barcode\": \"rg_barcode\"}, inplace=True)\n",
    "    df[\"read_length\"] = df[\"read_length\"].apply(\n",
    "        lambda x: int(x.split(\"x\")[0]))\n",
    "    \n",
    "    # Create new columns\n",
    "    df['max_insert_size'] = df['insert_size'].max()\n",
    "    df['mean_insert_size'] = df['insert_size'].mean()\n",
    "    df['mean_read_length'] = df['read_length'].mean()\n",
    "    df['total_reads'] = df['read_length'].count()\n",
    "    \n",
    "    # Subset of columns\n",
    "    df = df[['sample_name',\n",
    "             'library_name',\n",
    "             'rg_barcode',\n",
    "             'run_name',\n",
    "             'read_length',\n",
    "             'date',\n",
    "             'library_strategy',\n",
    "             'library_source',\n",
    "             'library_selection',\n",
    "             'insert_size',\n",
    "             'instrument',\n",
    "             'library_layout',\n",
    "             'max_insert_size',\n",
    "             'mean_insert_size',\n",
    "             'mean_read_length',\n",
    "             'total_reads']]\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extraction - Execution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Study files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CardiacAbnormalitie...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CardiacSurgeriesPro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CatheterizationDS.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CopynumberResultsDS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     study_file_name\n",
       "0  3a_dbGaP_SubjectPhenotypes_CardiacAbnormalitie...\n",
       "1  3a_dbGaP_SubjectPhenotypes_CardiacSurgeriesPro...\n",
       "2   3a_dbGaP_SubjectPhenotypes_CatheterizationDS.txt\n",
       "3  3a_dbGaP_SubjectPhenotypes_CopynumberResultsDS...\n",
       "4      3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_files_df = read_study_file_data()\n",
    "study_files_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>famid</th>\n",
       "      <th>subjid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>210</td>\n",
       "      <td>211</td>\n",
       "      <td>212</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>211</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>212</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>272</td>\n",
       "      <td>273</td>\n",
       "      <td>274</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>273</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   famid subjid  mother  father  is_proband\n",
       "0      1    210     211     212        True\n",
       "1      1    211       0       0       False\n",
       "2      1    212       0       0       False\n",
       "3      2    272     273     274        True\n",
       "4      2    273       0       0       False"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Family \n",
    "family_df = read_family_data()\n",
    "family_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/singhn4/Projects/kids_first/kf-api-dataservice/venv/lib/python3.6/site-packages/ipykernel_launcher.py:44: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hpo_id</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>observed</th>\n",
       "      <th>latest_exam_age</th>\n",
       "      <th>subjid</th>\n",
       "      <th>phenotype_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3673</td>\n",
       "      <td>phenotype_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21296</td>\n",
       "      <td>phenotype_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9653</td>\n",
       "      <td>phenotype_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15386</td>\n",
       "      <td>phenotype_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26312</td>\n",
       "      <td>phenotype_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  hpo_id                        phenotype observed  latest_exam_age subjid  \\\n",
       "0   None  ABDOMINAL_ABNORMALITIES_PRESENT     none              NaN   3673   \n",
       "1   None  ABDOMINAL_ABNORMALITIES_PRESENT       no              NaN  21296   \n",
       "2   None  ABDOMINAL_ABNORMALITIES_PRESENT       no              NaN   9653   \n",
       "3   None  ABDOMINAL_ABNORMALITIES_PRESENT     none              NaN  15386   \n",
       "4   None  ABDOMINAL_ABNORMALITIES_PRESENT     none              NaN  26312   \n",
       "\n",
       "  phenotype_id  \n",
       "0  phenotype_0  \n",
       "1  phenotype_1  \n",
       "2  phenotype_2  \n",
       "3  phenotype_3  \n",
       "4  phenotype_4  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Phenotypes\n",
    "phenotype_df = read_phenotype_data()\n",
    "phenotype_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5787</td>\n",
       "      <td>M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16252</td>\n",
       "      <td>F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26290</td>\n",
       "      <td>F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14665</td>\n",
       "      <td>F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2248</td>\n",
       "      <td>M</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex\n",
       "0   5787   M\n",
       "1  16252   F\n",
       "2  26290   F\n",
       "3  14665   F\n",
       "4   2248   M"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Gender\n",
    "gender_df = read_gender_data()\n",
    "gender_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>subjid</th>\n",
       "      <th>demographic_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>3673</td>\n",
       "      <td>demographic_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Black or African American</td>\n",
       "      <td>No</td>\n",
       "      <td>21296</td>\n",
       "      <td>demographic_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>9653</td>\n",
       "      <td>demographic_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>15386</td>\n",
       "      <td>demographic_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>White</td>\n",
       "      <td>Yes</td>\n",
       "      <td>26312</td>\n",
       "      <td>demographic_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        race ethnicity subjid demographic_id\n",
       "0                      White        No   3673  demographic_0\n",
       "1  Black or African American        No  21296  demographic_1\n",
       "2                      White        No   9653  demographic_2\n",
       "3                      White        No  15386  demographic_3\n",
       "4                      White       Yes  26312  demographic_4"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Demographic\n",
    "demographic_df = read_demographic_data()\n",
    "demographic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>diagnosis_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3673</td>\n",
       "      <td>L-loop corrected transposition of the great ar...</td>\n",
       "      <td>diagnosis_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3673</td>\n",
       "      <td>Complete heart block</td>\n",
       "      <td>diagnosis_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>21296</td>\n",
       "      <td>Tetralogy of Fallot with pulmonary atresia</td>\n",
       "      <td>diagnosis_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9653</td>\n",
       "      <td>Right aortic arch with mirror image branching ...</td>\n",
       "      <td>diagnosis_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9653</td>\n",
       "      <td>Ventricular septal defect, membranous</td>\n",
       "      <td>diagnosis_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid                                          diagnosis diagnosis_id\n",
       "0   3673  L-loop corrected transposition of the great ar...  diagnosis_0\n",
       "1   3673                               Complete heart block  diagnosis_1\n",
       "2  21296         Tetralogy of Fallot with pulmonary atresia  diagnosis_2\n",
       "3   9653  Right aortic arch with mirror image branching ...  diagnosis_3\n",
       "4   9653              Ventricular septal defect, membranous  diagnosis_4"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Diagnosis data\n",
    "diagnosis_df = read_diagnosis_data()\n",
    "diagnosis_df.head()\n",
    "# diagnosis_df[diagnosis_df['subjid'] == '279']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sampid</th>\n",
       "      <th>samp_source</th>\n",
       "      <th>source_sampid</th>\n",
       "      <th>sample_use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>22921</td>\n",
       "      <td>CG0019-7254</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0019-7254</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16165</td>\n",
       "      <td>CG0012-4474</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0012-4474</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9800</td>\n",
       "      <td>CG0009-6098</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0009-6098</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15386</td>\n",
       "      <td>CG0011-4770</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0011-4770</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>570</td>\n",
       "      <td>CG0000-1854</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0000-1854</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid       sampid samp_source source_sampid           sample_use\n",
       "0  22921  CG0019-7254     CORIELL   CG0019-7254  Seq_DNA_WholeGenome\n",
       "1  16165  CG0012-4474     CORIELL   CG0012-4474  Seq_DNA_WholeGenome\n",
       "2   9800  CG0009-6098     CORIELL   CG0009-6098  Seq_DNA_WholeGenome\n",
       "3  15386  CG0011-4770     CORIELL   CG0011-4770  Seq_DNA_WholeGenome\n",
       "4    570  CG0000-1854     CORIELL   CG0000-1854  Seq_DNA_WholeGenome"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sample data\n",
    "subject_sample_df = read_subject_sample_data()\n",
    "subject_sample_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>external_id</th>\n",
       "      <th>sample_collection_site</th>\n",
       "      <th>sample_role</th>\n",
       "      <th>concentration_ng_per_ul</th>\n",
       "      <th>initial_volume_microliters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1125743864</td>\n",
       "      <td>CG0000-6676</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>66.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1125743887</td>\n",
       "      <td>CG0000-6978</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Parent</td>\n",
       "      <td>50.0</td>\n",
       "      <td>65.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1125743888</td>\n",
       "      <td>CG0000-6989</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Parent</td>\n",
       "      <td>50.0</td>\n",
       "      <td>66.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1125743911</td>\n",
       "      <td>CG0000-6731</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>65.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1125743912</td>\n",
       "      <td>CG0000-6738</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Parent</td>\n",
       "      <td>50.0</td>\n",
       "      <td>66.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      barcode  external_id sample_collection_site sample_role  \\\n",
       "0  1125743864  CG0000-6676                  Blood    Affected   \n",
       "1  1125743887  CG0000-6978                  Blood      Parent   \n",
       "2  1125743888  CG0000-6989                  Blood      Parent   \n",
       "3  1125743911  CG0000-6731                  Blood    Affected   \n",
       "4  1125743912  CG0000-6738                  Blood      Parent   \n",
       "\n",
       "   concentration_ng_per_ul  initial_volume_microliters  \n",
       "0                     50.0                        66.1  \n",
       "1                     50.0                        65.7  \n",
       "2                     50.0                        66.0  \n",
       "3                     50.0                        65.7  \n",
       "4                     50.0                        66.2  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Aliquot/Sample Shipping data\n",
    "shipping_manifest_df = read_sample_shipping_manifest_data()\n",
    "shipping_manifest_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>library_name</th>\n",
       "      <th>rg_barcode</th>\n",
       "      <th>run_name</th>\n",
       "      <th>read_length</th>\n",
       "      <th>date</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>insert_size</th>\n",
       "      <th>instrument</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>max_insert_size</th>\n",
       "      <th>mean_insert_size</th>\n",
       "      <th>mean_read_length</th>\n",
       "      <th>total_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CG0023-9153</td>\n",
       "      <td>GMKFCS.ST-00000936-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-1</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_1</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>361</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CG0002-6325</td>\n",
       "      <td>GMKFCS.ST-00000953-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-2</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_2</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>369</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CG0002-6931</td>\n",
       "      <td>GMKFCS.ST-00000954-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-3</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>360</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CG0002-7519</td>\n",
       "      <td>GMKFCS.ST-00003538-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-4</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>365</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CG0002-7976</td>\n",
       "      <td>GMKFCS.ST-00003539-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-5</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>352</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_name               library_name   rg_barcode  \\\n",
       "0  CG0023-9153  GMKFCS.ST-00000936-1_2AMP  HMNMFCCXX-1   \n",
       "1  CG0002-6325  GMKFCS.ST-00000953-1_2AMP  HMNMFCCXX-2   \n",
       "2  CG0002-6931  GMKFCS.ST-00000954-1_2AMP  HMNMFCCXX-3   \n",
       "3  CG0002-7519  GMKFCS.ST-00003538-1_2AMP  HMNMFCCXX-4   \n",
       "4  CG0002-7976  GMKFCS.ST-00003539-1_2AMP  HMNMFCCXX-5   \n",
       "\n",
       "                             run_name  read_length                 date  \\\n",
       "0  160423_ST-E00212_0248_AHMNMFCCXX_1          150  2016-05-10 00:00:00   \n",
       "1  160423_ST-E00212_0248_AHMNMFCCXX_2          150  2016-05-10 00:00:00   \n",
       "2  160423_ST-E00212_0248_AHMNMFCCXX_3          150  2016-05-10 00:00:00   \n",
       "3  160423_ST-E00212_0248_AHMNMFCCXX_4          150  2016-05-10 00:00:00   \n",
       "4  160423_ST-E00212_0248_AHMNMFCCXX_5          150  2016-05-10 00:00:00   \n",
       "\n",
       "  library_strategy library_source library_selection  insert_size   instrument  \\\n",
       "0              WGS        GENOMIC            RANDOM          361  HiSeq X Ten   \n",
       "1              WGS        GENOMIC            RANDOM          369  HiSeq X Ten   \n",
       "2              WGS        GENOMIC            RANDOM          360  HiSeq X Ten   \n",
       "3              WGS        GENOMIC            RANDOM          365  HiSeq X Ten   \n",
       "4              WGS        GENOMIC            RANDOM          352  HiSeq X Ten   \n",
       "\n",
       "  library_layout  max_insert_size  mean_insert_size  mean_read_length  \\\n",
       "0     Paired-End              428        392.684444             150.0   \n",
       "1     Paired-End              428        392.684444             150.0   \n",
       "2     Paired-End              428        392.684444             150.0   \n",
       "3     Paired-End              428        392.684444             150.0   \n",
       "4     Paired-End              428        392.684444             150.0   \n",
       "\n",
       "   total_reads  \n",
       "0          900  \n",
       "1          900  \n",
       "2          900  \n",
       "3          900  \n",
       "4          900  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sequencing experiments\n",
    "seq_exp_df = read_seq_experiment_data()\n",
    "seq_exp_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Explore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Family\n",
      "famid          736\n",
      "subjid        2208\n",
      "mother         737\n",
      "father         737\n",
      "is_proband       2\n",
      "dtype: int64\n",
      "\n",
      "Demographics\n",
      "race                6\n",
      "ethnicity           2\n",
      "subjid            736\n",
      "demographic_id    736\n",
      "dtype: int64\n",
      "\n",
      "Gender\n",
      "race                6\n",
      "ethnicity           2\n",
      "subjid            736\n",
      "demographic_id    736\n",
      "dtype: int64\n",
      "\n",
      "Diagnosis\n",
      "race                6\n",
      "ethnicity           2\n",
      "subjid            736\n",
      "demographic_id    736\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# Participants\n",
    "print(\"Family\")\n",
    "print(family_df.nunique())\n",
    "print(\"\\nDemographics\")\n",
    "print(demographic_df.nunique())\n",
    "print(\"\\nGender\")\n",
    "print(demographic_df.nunique())\n",
    "print(\"\\nDiagnosis\")\n",
    "print(demographic_df.nunique())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unnamed:_0</th>\n",
       "      <th>institution</th>\n",
       "      <th>investigator_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>Harvard Medical School</td>\n",
       "      <td>Christine E. Seidman</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   unnamed:_0             institution     investigator_name\n",
       "0           0  Harvard Medical School  Christine E. Seidman"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Investigator\n",
    "investigator_df = read_investigator_data()\n",
    "investigator_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unnamed:_0</th>\n",
       "      <th>attribution</th>\n",
       "      <th>data_access_authority</th>\n",
       "      <th>study_id</th>\n",
       "      <th>study_name</th>\n",
       "      <th>study_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/projects/gap/cgi-...</td>\n",
       "      <td>dbGaP</td>\n",
       "      <td>phs001138</td>\n",
       "      <td>Discovery of the Genetic Basis of Structural H...</td>\n",
       "      <td>v1.p2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   unnamed:_0                                        attribution  \\\n",
       "0           0  https://www.ncbi.nlm.nih.gov/projects/gap/cgi-...   \n",
       "\n",
       "  data_access_authority   study_id  \\\n",
       "0                 dbGaP  phs001138   \n",
       "\n",
       "                                          study_name study_version  \n",
       "0  Discovery of the Genetic Basis of Structural H...         v1.p2  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Study\n",
    "study_df = read_study_data()\n",
    "study_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CardiacAbnormalitie...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CardiacSurgeriesPro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CatheterizationDS.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CopynumberResultsDS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     study_file_name\n",
       "0  3a_dbGaP_SubjectPhenotypes_CardiacAbnormalitie...\n",
       "1  3a_dbGaP_SubjectPhenotypes_CardiacSurgeriesPro...\n",
       "2   3a_dbGaP_SubjectPhenotypes_CatheterizationDS.txt\n",
       "3  3a_dbGaP_SubjectPhenotypes_CopynumberResultsDS...\n",
       "4      3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Study files\n",
    "study_files_df = read_study_file_data()\n",
    "study_files_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>famid</th>\n",
       "      <th>subjid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>210</td>\n",
       "      <td>211</td>\n",
       "      <td>212</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>211</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>212</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>272</td>\n",
       "      <td>273</td>\n",
       "      <td>274</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>273</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   famid subjid  mother  father  is_proband\n",
       "0      1    210     211     212        True\n",
       "1      1    211       0       0       False\n",
       "2      1    212       0       0       False\n",
       "3      2    272     273     274        True\n",
       "4      2    273       0       0       False"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Family\n",
    "family_df = read_family_data()\n",
    "family_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "subjid            2208\n",
       "sex                  2\n",
       "race                 6\n",
       "ethnicity            2\n",
       "demographic_id     736\n",
       "famid              736\n",
       "mother             737\n",
       "father             737\n",
       "is_proband           2\n",
       "dtype: int64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create participant df\n",
    "# Merge Gender + Demographics\n",
    "gender_demo_df = pd.merge(gender_df, demographic_df, on='subjid')\n",
    "# Add Family\n",
    "df1 = pd.merge(gender_demo_df, family_df, on='subjid', how='outer')\n",
    "df1.head()\n",
    "df1.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>demographic_id</th>\n",
       "      <th>famid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>diagnosis_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis, bilateral branch pulmonary...</td>\n",
       "      <td>diagnosis_232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Single ventricle comprised of mostly left vent...</td>\n",
       "      <td>diagnosis_233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>L-loop transposition of the great arteries</td>\n",
       "      <td>diagnosis_234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Double outlet left ventricle</td>\n",
       "      <td>diagnosis_235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis</td>\n",
       "      <td>diagnosis_236</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex   race ethnicity  demographic_id  famid  mother  father  \\\n",
       "0    390   F  White        No  demographic_64      8     391     392   \n",
       "1    390   F  White        No  demographic_64      8     391     392   \n",
       "2    390   F  White        No  demographic_64      8     391     392   \n",
       "3    390   F  White        No  demographic_64      8     391     392   \n",
       "4    390   F  White        No  demographic_64      8     391     392   \n",
       "\n",
       "   is_proband                                          diagnosis  \\\n",
       "0        True  Pulmonary stenosis, bilateral branch pulmonary...   \n",
       "1        True  Single ventricle comprised of mostly left vent...   \n",
       "2        True         L-loop transposition of the great arteries   \n",
       "3        True                       Double outlet left ventricle   \n",
       "4        True                                 Pulmonary stenosis   \n",
       "\n",
       "    diagnosis_id  \n",
       "0  diagnosis_232  \n",
       "1  diagnosis_233  \n",
       "2  diagnosis_234  \n",
       "3  diagnosis_235  \n",
       "4  diagnosis_236  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge Diagnosis\n",
    "df2 = pd.merge(df1, diagnosis_df, on='subjid')\n",
    "df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>demographic_id</th>\n",
       "      <th>famid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>diagnosis_id</th>\n",
       "      <th>sampid</th>\n",
       "      <th>samp_source</th>\n",
       "      <th>source_sampid</th>\n",
       "      <th>sample_use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis, bilateral branch pulmonary...</td>\n",
       "      <td>diagnosis_232</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Single ventricle comprised of mostly left vent...</td>\n",
       "      <td>diagnosis_233</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>L-loop transposition of the great arteries</td>\n",
       "      <td>diagnosis_234</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Double outlet left ventricle</td>\n",
       "      <td>diagnosis_235</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis</td>\n",
       "      <td>diagnosis_236</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex   race ethnicity  demographic_id  famid  mother  father  \\\n",
       "0    390   F  White        No  demographic_64      8     391     392   \n",
       "1    390   F  White        No  demographic_64      8     391     392   \n",
       "2    390   F  White        No  demographic_64      8     391     392   \n",
       "3    390   F  White        No  demographic_64      8     391     392   \n",
       "4    390   F  White        No  demographic_64      8     391     392   \n",
       "\n",
       "   is_proband                                          diagnosis  \\\n",
       "0        True  Pulmonary stenosis, bilateral branch pulmonary...   \n",
       "1        True  Single ventricle comprised of mostly left vent...   \n",
       "2        True         L-loop transposition of the great arteries   \n",
       "3        True                       Double outlet left ventricle   \n",
       "4        True                                 Pulmonary stenosis   \n",
       "\n",
       "    diagnosis_id       sampid samp_source source_sampid           sample_use  \n",
       "0  diagnosis_232  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  \n",
       "1  diagnosis_233  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  \n",
       "2  diagnosis_234  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  \n",
       "3  diagnosis_235  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  \n",
       "4  diagnosis_236  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge Sample\n",
    "df3 = pd.merge(df2, subject_sample_df, on='subjid')\n",
    "df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>demographic_id</th>\n",
       "      <th>famid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>sampid</th>\n",
       "      <th>samp_source</th>\n",
       "      <th>source_sampid</th>\n",
       "      <th>sample_use</th>\n",
       "      <th>barcode</th>\n",
       "      <th>external_id</th>\n",
       "      <th>sample_collection_site</th>\n",
       "      <th>sample_role</th>\n",
       "      <th>concentration_ng_per_ul</th>\n",
       "      <th>initial_volume_microliters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis, bilateral branch pulmonary...</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Single ventricle comprised of mostly left vent...</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>L-loop transposition of the great arteries</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Double outlet left ventricle</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex   race ethnicity  demographic_id  famid  mother  father  \\\n",
       "0    390   F  White        No  demographic_64      8     391     392   \n",
       "1    390   F  White        No  demographic_64      8     391     392   \n",
       "2    390   F  White        No  demographic_64      8     391     392   \n",
       "3    390   F  White        No  demographic_64      8     391     392   \n",
       "4    390   F  White        No  demographic_64      8     391     392   \n",
       "\n",
       "   is_proband                                          diagnosis  \\\n",
       "0        True  Pulmonary stenosis, bilateral branch pulmonary...   \n",
       "1        True  Single ventricle comprised of mostly left vent...   \n",
       "2        True         L-loop transposition of the great arteries   \n",
       "3        True                       Double outlet left ventricle   \n",
       "4        True                                 Pulmonary stenosis   \n",
       "\n",
       "              ...                   sampid samp_source source_sampid  \\\n",
       "0             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "1             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "2             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "3             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "4             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "\n",
       "            sample_use     barcode  external_id sample_collection_site  \\\n",
       "0  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "1  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "2  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "3  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "4  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "\n",
       "  sample_role concentration_ng_per_ul  initial_volume_microliters  \n",
       "0    Affected                    50.0                        60.1  \n",
       "1    Affected                    50.0                        60.1  \n",
       "2    Affected                    50.0                        60.1  \n",
       "3    Affected                    50.0                        60.1  \n",
       "4    Affected                    50.0                        60.1  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge Aliquot\n",
    "df4 = pd.merge(df3, shipping_manifest_df, left_on='sampid', right_on='external_id')\n",
    "df4.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>demographic_id</th>\n",
       "      <th>famid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>insert_size</th>\n",
       "      <th>instrument</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>max_insert_size</th>\n",
       "      <th>mean_insert_size</th>\n",
       "      <th>mean_read_length</th>\n",
       "      <th>total_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis, bilateral branch pulmonary...</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Single ventricle comprised of mostly left vent...</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>L-loop transposition of the great arteries</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Double outlet left ventricle</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex   race ethnicity  demographic_id  famid  mother  father  \\\n",
       "0    390   F  White        No  demographic_64      8     391     392   \n",
       "1    390   F  White        No  demographic_64      8     391     392   \n",
       "2    390   F  White        No  demographic_64      8     391     392   \n",
       "3    390   F  White        No  demographic_64      8     391     392   \n",
       "4    390   F  White        No  demographic_64      8     391     392   \n",
       "\n",
       "   is_proband                                          diagnosis     ...       \\\n",
       "0        True  Pulmonary stenosis, bilateral branch pulmonary...     ...        \n",
       "1        True  Single ventricle comprised of mostly left vent...     ...        \n",
       "2        True         L-loop transposition of the great arteries     ...        \n",
       "3        True                       Double outlet left ventricle     ...        \n",
       "4        True                                 Pulmonary stenosis     ...        \n",
       "\n",
       "  library_strategy library_source library_selection insert_size   instrument  \\\n",
       "0              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "1              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "2              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "3              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "4              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "\n",
       "  library_layout max_insert_size mean_insert_size mean_read_length  \\\n",
       "0     Paired-End             428       392.684444            150.0   \n",
       "1     Paired-End             428       392.684444            150.0   \n",
       "2     Paired-End             428       392.684444            150.0   \n",
       "3     Paired-End             428       392.684444            150.0   \n",
       "4     Paired-End             428       392.684444            150.0   \n",
       "\n",
       "   total_reads  \n",
       "0          900  \n",
       "1          900  \n",
       "2          900  \n",
       "3          900  \n",
       "4          900  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge Sequencing Experiment\n",
    "full_participant_df = pd.merge(df4, seq_exp_df, left_on='external_id', right_on='sample_name')\n",
    "full_participant_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create study\n",
    "study = {\n",
    "    'data_access_authority': 'dbGaP',\n",
    "    'study_id': 'phs001138',\n",
    "    'study_version': 'v1.p2',\n",
    "    'study_name': 'Discovery of the Genetic Basis of Structural Heart'\n",
    "    'and Other Birth Defects',\n",
    "    'attribution': 'https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/'\n",
    "    'GetAcknowledgementStatement.cgi?study_id=phs001138.v1.p2'\n",
    "}\n",
    "study_df = pd.DataFrame([study])\n",
    "study_df.to_csv(os.path.join(DATA_DIR, 'study.txt'))\n",
    "\n",
    "# Create investigator\n",
    "invest = {\n",
    "    'investigator_name': 'Christine E. Seidman',\n",
    "    'institution': 'Harvard Medical School'\n",
    "}\n",
    "inv_df = pd.DataFrame([invest])\n",
    "inv_df.to_csv(os.path.join(DATA_DIR, 'investigator.txt'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "def _add_study_cols(study_df, df):\n",
    "    # Add study cols to a df\n",
    "    cols = study_df.columns.tolist()\n",
    "    row = study_df.iloc[0]\n",
    "    for col in cols:\n",
    "        df[col] = row[col]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add study to full participant df\n",
    "_add_study_cols(study_df, full_participant_df)\n",
    "\n",
    "# Add study to basic participant df\n",
    "participant_df = _add_study_cols(study_df, family_df)\n",
    "\n",
    "# Add study to investigator df\n",
    "study_investigator_df =_add_study_cols(study_df, investigator_df)\n",
    "\n",
    "# Add study to study files df\n",
    "study_study_files_df = _add_study_cols(study_df, study_files_df)\n",
    "\n",
    "# Phenotype df\n",
    "phenotype_participant_df = pd.merge(phenotype_df, participant_df,\n",
    "                                    on='subjid')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adding 1:10 participants to session\n",
      "['participant_1',\n",
      " 'participant_2',\n",
      " 'participant_3',\n",
      " 'participant_4',\n",
      " 'participant_5',\n",
      " 'participant_6',\n",
      " 'participant_7',\n",
      " 'participant_8',\n",
      " 'participant_9']\n",
      "Flushing 10 participant\n",
      "s\n",
      "Adding 11:20 participants to session\n",
      "['participant_11',\n",
      " 'participant_12',\n",
      " 'participant_13',\n",
      " 'participant_14',\n",
      " 'participant_15',\n",
      " 'participant_16',\n",
      " 'participant_17',\n",
      " 'participant_18',\n",
      " 'participant_19']\n",
      "Flushing 10 participant\n",
      "s\n",
      "Flushing remaining 3 participants to session\n",
      "\n",
      "['participant_0', 'participant_20', 'participant_21']\n"
     ]
    }
   ],
   "source": [
    "from pprint import pprint\n",
    "n = 22\n",
    "chunk_size = 10\n",
    "entity_type ='participant'\n",
    "entities = ['{}_{}'.format(entity_type, j) for j in range(n)]\n",
    "for i in range(0, n, chunk_size):\n",
    "    chunk = entities[i - chunk_size:i]\n",
    "    if chunk:\n",
    "        start = i - chunk_size + 1\n",
    "        print('Adding {}:{} {}s to session'.format(start, i,\n",
    "                                                   entity_type))\n",
    "        pprint([e for e in entities[start:i]])\n",
    "        print('Flushing {} {}\\ns'.format(chunk_size, entity_type))\n",
    "\n",
    "print('Flushing remaining {} {}s to session\\n'.format(\n",
    "    len(entities[i:]) + 1, entity_type))\n",
    "\n",
    "remaining = entities[0:1] + entities[i:]\n",
    "\n",
    "pprint([e for e in remaining])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "def read_json(filepath):\n",
    "    with open(filepath, 'r') as json_file:\n",
    "        return json.load(json_file)\n",
    "\n",
    "\n",
    "def write_json(data, filepath):\n",
    "    with open(filepath, 'w') as json_file:\n",
    "        json.dump(data, json_file, sort_keys=True, indent=4, separators=(',', ':'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "fp = os.path.join(DATA_DIR, 'uuid_path_size_md5.json')\n",
    "file_json = read_json(fp)\n",
    "write_json(file_json, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
