{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "DATA_DIR = '/Users/singhn4/Projects/kids_first/data/Seidman_2015'\n",
    "DBGAP_DIR = os.path.join(DATA_DIR, 'dbgap')\n",
    "ALIQUOT_SHIP_DIR = os.path.join(DATA_DIR, 'manifests', 'shipping')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extraction - Methods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Helper functions\n",
    "def dropna_rows_cols(df_func):\n",
    "    \"\"\"\n",
    "    Decorator to drop rows and cols w all nan values\n",
    "    \"\"\"\n",
    "\n",
    "    def wrapper(*args, **kwargs):\n",
    "        df = df_func(*args, **kwargs)\n",
    "\n",
    "        # None or empty df\n",
    "        try:\n",
    "            if df.empty:\n",
    "                return df\n",
    "        except AttributeError:\n",
    "            return df\n",
    "\n",
    "        # Rows\n",
    "        df.dropna(how=\"all\", inplace=True)\n",
    "        # Cols\n",
    "        df.dropna(how=\"all\", axis=1, inplace=True)\n",
    "        return df\n",
    "\n",
    "    return wrapper\n",
    "\n",
    "def reformat_column_names(df_func):\n",
    "    \"\"\"\n",
    "    Decorator to reformat DataFrame column names.\n",
    "\n",
    "    Replace all column names having whitespace with underscore\n",
    "    and make lowercase\n",
    "    \"\"\"\n",
    "\n",
    "    def wrapper(*args, **kwargs):\n",
    "        df = df_func(*args, **kwargs)\n",
    "        # None or empty df\n",
    "        try:\n",
    "            if df.empty:\n",
    "                return df\n",
    "        except AttributeError:\n",
    "            return df\n",
    "        df.columns = map((lambda x: x.replace(\" \", \"_\").lower()),\n",
    "                         df.columns)\n",
    "        return df\n",
    "\n",
    "    return wrapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_study_file_data(filepaths=None):\n",
    "    \"\"\"\n",
    "    Read in raw study files\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filepaths = os.listdir(DBGAP_DIR)\n",
    "\n",
    "    study_files = [{\"study_file_name\": f}\n",
    "                   for f in filepaths if 'dbGaP' in f]\n",
    "    return pd.DataFrame(study_files)\n",
    "\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_study_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read study data\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR,\n",
    "                                'study.txt')\n",
    "    df = pd.read_csv(filepath)\n",
    "\n",
    "    return df\n",
    "\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_investigator_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read investigator data\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR,\n",
    "                                'investigator.txt')\n",
    "    df = pd.read_csv(filepath)\n",
    "\n",
    "    return df\n",
    "\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_family_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read family data for all participants\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DBGAP_DIR,\n",
    "                                '7a_dbGaP_PedigreeDS.txt')\n",
    "    df = pd.read_csv(filepath,\n",
    "                     delimiter='\\t',\n",
    "                     dtype={'SUBJID': str})\n",
    "    # Subset of columns\n",
    "    df.drop(['SEX'], axis=1, inplace=True)\n",
    "\n",
    "    # Add proband column\n",
    "    def func(row): return bool(row['MOTHER'] and row['FATHER'])\n",
    "    df['is_proband'] = df.apply(func, axis=1)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['No' 'Yes']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'No' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'abnormal fetal calvarium'\n",
      " 'Dolichocephaly with bony synostosis of sagittal structure.'\n",
      " 'dolichochocehaly' 'unilateral coronal synostosis']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'frontal bossing (occipital protuberance)'\n",
      " 'cephalohematoma, Caput Succedaneum'\n",
      " 'short, sloping forehead with prominent occiput' 'bitemporal hallowing'\n",
      " 'small anterior fontanelle' 'congenital scar from eye to scalp per Mom'\n",
      " 'brachycephaly' 'parietal encephalocele, plagiocephaly'\n",
      " 'overriding coronal sutures'\n",
      " 'medical record indicates that she had a cyst in her brain'\n",
      " 'narrow forehead' 'metopic prominence' 'plagiocephaly' 'Plagiocephaly'\n",
      " 'sloped forehead, narrow' 'flat face' '3 fontanelles'\n",
      " 'Plagiocephaly, triangular shaped in coronal plane' 'bitemporal narrowing']\n",
      "['None' 'No' 'Yes']\n",
      "['Unknown' 'None' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Glaucoma' 'myopia' 'intraretinal hemorrhages'\n",
      " 'mildly proptotic eyes' 'bilateral low vision' 'sclerae anicteric'\n",
      " 'downslanting' 'Exotropia' 'short palpebral fissures'\n",
      " 'left microphthalmia' 'proptotic, long eyelashes' 'Belpharoptosis'\n",
      " 'Microphthalmia and cloudy corneas'\n",
      " 'prominant eyes, long eyelashes, midface hypoplasia' 'epibulbar dermoid'\n",
      " 'downslanting palepbral fissures' 'hypertelorism' 'mild ROP OU'\n",
      " 'upslanting palpebrals' 'slightly upslanted palpebral fissures'\n",
      " 'palpebral fissures' 'eye deviation\"\" per  note' 'Amblyopia'\n",
      " 'Hypotelorism, downslanted palpebral fissures'\n",
      " 'hypotelorism, downslanting papebral fissures' 'amblyopia'\n",
      " 'Born with retinoblastoma']\n",
      "['None' 'No' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'No/Not checked' 'Yes']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Atresia of left external ear'\n",
      " 'R ear overfolded and posteriorly rotated; L ear slightly pointed'\n",
      " 'cupped ears' 'pointy superiorly' 'small' 'slight redundancy in helices'\n",
      " 'rail road track deformity' 'Bilateral glue ear'\n",
      " 'overriding helices and relatively small' 'squared helix'\n",
      " 'Bilateral Congenital Ossicular Anomaly' 'prominent'\n",
      " 'R ear long and underdeveloped' 'posteriorly rotated'\n",
      " 'Underdeveloped ears' 'pointy' 'enlarged' 'overfolded superior helices'\n",
      " 'Left ear microtia' 'Posteriorly rotated' 'r ear is smaller in size'\n",
      " 'posteriorly set' 'thick helices' 'congenital deafness, R pinna cupped.']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'No' 'Yes']\n",
      "['Unknown' 'None' 'No/Not checked']\n",
      "['Unknown' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Torticollis' 'torticollis' 'short neck'\n",
      " 'neck bent to one side;needed PT' 'short' 'cystic hygroma' 'mildly short'\n",
      " 'short, thick' 'Short']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Short frenulum, hypoplastic left mandible' 'tongue-tied' 'thrush'\n",
      " 'ankyloglossia' 'asymmetric' 'small' 'thin upper lip'\n",
      " 'small mouth,thin upper lip' 'missing teeth'\n",
      " 'small maxilla/mandible on right side' 'thin lips, short philtrum'\n",
      " 'Asymmetric mouth movement; hypoplastic left jaw']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'high palate' 'high arched' 'high arched and narrowed'\n",
      " 'mildly high-arched']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'stridor, left vocal cord paralysis' 'esophagitis.  G-tube.'\n",
      " 'hypoplastic right side, anteriorly displaced' 'Asthma'\n",
      " 'esophageal reflux']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['None' 'sternum asymmetry']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'asthma' 'scimitar syndrome' 'small lungs'\n",
      " 'pulmonary lymphangiectasia' 'pulmonary hypertension'\n",
      " 'hypoplasia L mainstem bronchus'\n",
      " 'pulmonary hypertension in newborn period'\n",
      " 'CHRONIC LUNG DISEASE-PREMATURITY'\n",
      " 'Frequent episodes of migratory atelectasis'\n",
      " 'reversed pulmonary anatomy (L-sided on R, R on L)'\n",
      " 'inverted bronchial pattern']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' '3 cm. liver' 'enlarged tranverse liver' 'Liver hepatomegaly'\n",
      " 'infantile hepatic hemangioma' 'indirect hyperbilirubinemia'\n",
      " 'portacaval shunt']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'heterotaxy' 'Gastroesophageal reflux'\n",
      " 'Protein-losing enteropathy: Celiac confirmed' 'cecal carcinoma'\n",
      " 'deformity of the right anterolateral abdominal wall' '2 vessel cord'\n",
      " '2 vessel umbilical cord' 'pancreas not seen on ultrasound'\n",
      " 'Hernia, type unk' 'stricture or obstruction and medical NEC'\n",
      " 'Moragni hernia' 'heterotaxy syndrome' 'ileal atresia'\n",
      " 'Protein Losing Enteropathy' 'hernia x2 type unk' 'right-sided stomach'\n",
      " 'rt-sided stomach' 'esophageal atresai' 'cholecystectomy  1994'\n",
      " 'rt-sided stomach & rt-sided multi-lobulated spleen'\n",
      " 'Atresia of the large intestine' 'Pancreatitis' 'GERD' 'situs inversus']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Pyelectasis' 'Hypoplastic'\n",
      " 'non-functioning Rt. kidney, obstruction of bladder outlet, hydroureteral nephrosis of the Lt. kidney'\n",
      " 'L ureterocele with absent R kidney' 'R and L kidneys pelvicaliectasis'\n",
      " 'left pyelectasis' 'double left renal collecting systems and ureters'\n",
      " 'double kidney' 'mild upper pole caliectasis'\n",
      " 'ureteropelvic junction obstruction' 'Chronic renal insufficiency'\n",
      " 'Small right kidney' 'R KIDNEY IS ABSENT' 'hyperechoic renal cortex'\n",
      " 'ureterocele, Right vesicoureteric reflux, grade 1' 'pelviectasis'\n",
      " 'rectourethral fistula' 'renal agenesis' 'hypoplasia of kidney']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'meatus stenosis' 'chordee' 'phimosis' 'rectovaginal fistula'\n",
      " 'hydrocele' 'labial adhesions' 'Hydrocele' 'high riding testicles'\n",
      " 'neurogenic bladder' 'ventral displacement of meatus' 'hydroceles'\n",
      " 'rectovestibular fistula' 'hypogonadotropic hypogonadism'\n",
      " 'varicoceles and phimosis.' 'prominent labia']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'spina bifida'\n",
      " 'asymmetric sacral cleft, low lying conus terminalis L3' 'hemi-vertebrae'\n",
      " 'Klippel-Feil syndrome'\n",
      " 'sacral agenesis with absent coccyx and missing vertebrae at S1'\n",
      " 'abnormal cervical and thoracic bodies' 'absent sacrum and lumbar spine'\n",
      " 'incompletely formed T10' 'segmental abnormality of sacrum'\n",
      " 'left sided hemivertebrae'\n",
      " 'segmental anomalies involving upper thoracic spine']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Thoracic spinal cord syrinx' 'Congenital hemivertebra'\n",
      " 'stiff musculature' 'thickened rib, mild scoliosis, difuse osteopenia'\n",
      " 'Osteopenia (updated )' 'Spina bifida' 'tibial torsion,'\n",
      " 'Right sided Sternocleidomastoid Contracture' 'short stature'\n",
      " 'shortened long bones' 'asymmetrical gluteal crease'\n",
      " 'underdeveloped pelvis, lower extremity contractures'\n",
      " 'low lying conus at S1-2 w/ fatty filum'\n",
      " 'short achilles tendon on one side'\n",
      " 'leg length discrepancy and short stature' 'short sternum'\n",
      " 'mild ligamentous laxity' 'shallow acetabula' 'idiopathic short stature'\n",
      " 'sacral hemangioma'\n",
      " 'no humerus, radius/ulna or hand bones on the left side'\n",
      " 'hypotonia lower extremities']\n",
      "['None' 'Yes' 'No']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'Yes' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Clinodactyly, right hand' 'mildly hypoplastic proximal thumbs'\n",
      " 'clinodactyly 5th finger' 'hypoplastic left thumb' 'tapering fingers'\n",
      " 'left transverse crease'\n",
      " 'bilateral clinodactyly 5th fingers, left hand simian creases'\n",
      " 'decreased thenar eminence' 'clubbing'\n",
      " 'skintags after the 5th digit on each hand.'\n",
      " 'underdeveloped thenar prominence'\n",
      " 'distal limb deformities and radial anomaly'\n",
      " 'Left hand single palmar crease, clinodactyly'\n",
      " 'no humerus, radius/ulna or hand bones on the left side']\n",
      "['None' 'Yes' 'No']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'Yes' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'short toe bones 4th/5th digits' 'Broad big toe'\n",
      " 'metatarsus adductus bilaterally' 'bilat toes 2-4 syndactyly'\n",
      " 'clindactyly of fifth digits, L partial simian crease' 'Tarsal Coalition'\n",
      " 'extra digit bilat' 'overlapping toes on right: 4th overlaps 5th'\n",
      " 'hallux valgus']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'learning disability, ADHD' 'LS Myelomeningocele' 'seizures'\n",
      " 'monoparesis of right arm' 'developmental & learning issues' 'PVL'\n",
      " 'Gross motor delay' 'Hypertonic' 'choroid plexus cyst'\n",
      " 'Assymetric crying facies' 'anxiety disorder' 'Seizure disorder.'\n",
      " 'Agenesis of corpus callosum' 'cerebral atrophy' 'low muscle tone'\n",
      " 'hamartoma' 'sinus dermal tract'\n",
      " 'pericallsol lipoma; splenium of corpus callosum'\n",
      " 'severe developmental delay'\n",
      " 'severe developmental delay, non-verbal, autistic, history of seizures'\n",
      " 'Congenital Bulbar Palsy' 'parietal encephalocele'\n",
      " 'global developmental delay, hypotonia' 'Caudal regression syndrome'\n",
      " 'hypotonia' 'muscle tone' 'mild global developmental delay'\n",
      " 'hypotonia, pt with head lag, cannot sit on her own' 'severe seizures'\n",
      " 'hypoplastic corpus callosum, vermis hypoplasia'\n",
      " 'seizures related to hypocalcemia' 'polymicrogyria' 'HORNER SYNDROME'\n",
      " 'seizure disorder' 'stroke and seizure per physician note '\n",
      " 'tethered cord malformation sequence' 'periventricular leukomalacia'\n",
      " 'subject has CP.' 'hemiplegic cerebral palsy; abnormal gait']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'skin tag on lower part of sternum' '3 facial skin tags'\n",
      " 'left sided auricular skin tags as well as one tag on left cheek.'\n",
      " 'very dry skin\"\"' 'anal skin tag']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'congenital hypoglycemia' 'Hypothyroidism, Vit D deficiency'\n",
      " 'hypothyroid' 'late onset of puberty/growth hormone deficiency'\n",
      " 'Hypothyroidism (update )' 'congenital hypothyroidism' 'short stature'\n",
      " 'Hypothyroidism' 'central hypothyrodism' 'hypercholesterolemia'\n",
      " 'hypothyroidism' 'hypothyroidism, hyperbilirubinemia'\n",
      " 'hypoglycemia, hypoparathyroidism'\n",
      " 'delayed puberty-no facial hair,pubic or underarm hair.'\n",
      " 'low growth factor/idiopathic short stature' 'pheochromocytoma   surgery '\n",
      " 'hypoglycemia']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'Hypoplastic' 'Absent' 'Other']\n",
      "['None' 'very small']\n",
      "['None' 'No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'anemia' 'Factor V Leiden' 'Factor XII deficiency'\n",
      " 'thrombocytopenia' 'ANEMIA OF PREMATURITY'\n",
      " 'Factor V Leiden heterozygous, MTHFR heterozygous' 'hypogammaglobulinemia'\n",
      " 'decrease vit K and Factor XII deficiency']\n",
      "['No' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked' 'Yes']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['None']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['Not applicable' 'None' 'No/Not checked']\n",
      "['Not applicable' 'None' 'Yes' 'No/Not checked']\n",
      "['None' 'Homozygous familial hypercholesterolemia'\n",
      " 'alpha and beta Thalassemia trait']\n",
      "['None' 'No' 'Yes']\n",
      "['None' 'Immune Globulin deficiency for unknown reasons'\n",
      " 'Exposed to toxoplasma in utero - born negative'\n",
      " 'juvenile rheumatoid arthritis']\n"
     ]
    }
   ],
   "source": [
    "# Find the potential negative/positive observed vaue\n",
    "from pandas.api.types import is_numeric_dtype\n",
    "filepath = os.path.join(DBGAP_DIR, '3a_dbGaP_SubjectPhenotypes_ExtracardiacFindingsDS.txt')\n",
    "\n",
    "# Read csv\n",
    "df = pd.read_csv(filepath,\n",
    "             delimiter='\\t',\n",
    "             dtype={'SUBJID': str})\n",
    "\n",
    "cols = df.columns.tolist()[1:]\n",
    "for col in cols:\n",
    "    if not is_numeric_dtype(df[col]):\n",
    "        print(df[col].unique())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['none' 'no' 'yes' 'not applicable' 'no/not checked' 'unknown'\n",
      " 'congenital hypoglycemia' 'hypothyroidism, vit d deficiency' 'hypothyroid'\n",
      " 'late onset of puberty/growth hormone deficiency'\n",
      " 'hypothyroidism (update )' 'congenital hypothyroidism' 'short stature'\n",
      " 'hypothyroidism' 'central hypothyrodism' 'hypercholesterolemia'\n",
      " 'hypothyroidism, hyperbilirubinemia' 'hypoglycemia, hypoparathyroidism'\n",
      " 'delayed puberty-no facial hair,pubic or underarm hair.'\n",
      " 'low growth factor/idiopathic short stature' 'pheochromocytoma   surgery '\n",
      " 'hypoglycemia' 'immune globulin deficiency for unknown reasons'\n",
      " 'exposed to toxoplasma in utero - born negative'\n",
      " 'juvenile rheumatoid arthritis' 'abnormal fetal calvarium'\n",
      " 'dolichocephaly with bony synostosis of sagittal structure.'\n",
      " 'dolichochocehaly' 'unilateral coronal synostosis' 'hypoplastic' 'absent'\n",
      " 'other']\n"
     ]
    }
   ],
   "source": [
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_phenotype_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read phenotype data\n",
    "    \"\"\"\n",
    "    filepath = os.path.join(\n",
    "    DBGAP_DIR,\n",
    "    '3a_dbGaP_SubjectPhenotypes_ExtracardiacFindingsDS.txt')\n",
    "\n",
    "    # Read csv\n",
    "    df = pd.read_csv(filepath,\n",
    "                     delimiter='\\t',\n",
    "                     dtype={'SUBJID': str})\n",
    "\n",
    "    # Convert age years to days\n",
    "    df['LATEST_EXAM_AGE'] = df[\"LATEST_EXAM_AGE\"].apply(\n",
    "        lambda x: float(x) * 365)\n",
    "    age_at_event_days = df[['LATEST_EXAM_AGE', 'SUBJID']]\n",
    "\n",
    "    # Select string based phenotypes\n",
    "    df = df.select_dtypes(include='object')\n",
    "\n",
    "    # Make all values lower case\n",
    "    for col in df.columns.tolist():\n",
    "        df[col] = df[col].apply(lambda x: str(x).lower())\n",
    "\n",
    "    # Reshape to build the phenotypes df\n",
    "    cols = df.columns.tolist()[2:]\n",
    "    phenotype_cols = [col for col in cols if not col.startswith('OTHER')]\n",
    "    phenotype_df = pd.melt(df, id_vars='SUBJID', value_vars=phenotype_cols,\n",
    "                           var_name='phenotype', value_name='observed')\n",
    "\n",
    "    # Merge with HPO mapping\n",
    "    mapping_filepath = os.path.join(DATA_DIR, 'phenotype_hpo_mapping.txt')\n",
    "    if os.path.isfile(mapping_filepath):\n",
    "        hpo_df = pd.read_csv(mapping_filepath)\n",
    "        new_df = pd.merge(hpo_df, phenotype_df, on='phenotype')\n",
    "        new_df.rename(columns={\"Yes\": \"hpo_id\"}, inplace=True)\n",
    "        phenotype_df = new_df[['hpo_id', 'phenotype', 'observed','SUBJID']]\n",
    "\n",
    "    # Remove unkonwns\n",
    "    unknown_values = ['none', 'unknown', 'no/not checked' 'not applicable', 'absent']\n",
    "    phenotype_df = phenotype_df[phenotype_df['observed'].apply(lambda x: x not in unknown_values)]\n",
    "\n",
    "    # Map to positive/negative\n",
    "    def func(row): \n",
    "        return 'negative' if row['observed'] == 'no' else 'positive'\n",
    "    phenotype_df['observed'] = phenotype_df.apply(func, axis=1)\n",
    "\n",
    "    # Clean up hpo_id\n",
    "    phenotype_df.loc[(phenotype_df.hpo_id == 'None') | (phenotype_df.hpo_id == '--'), 'hpo_id'] = None\n",
    "\n",
    "    # Merge back in age at event in days\n",
    "    phenotype_df = pd.merge(phenotype_df, age_at_event_days, on='SUBJID')\n",
    "\n",
    "    # Add unique col\n",
    "    def func(row): return \"_\".join(['phenotype', str(row.name)])\n",
    "    phenotype_df['phenotype_id'] = phenotype_df.apply(func, axis=1)\n",
    "\n",
    "    return phenotype_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gender\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_gender_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read gender data for all subjects\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DBGAP_DIR,\n",
    "                                '3a_dbGaP_SubjectPhenotypes_GenderDS.txt')\n",
    "    df = pd.read_csv(filepath,\n",
    "                     delimiter='\\t',\n",
    "                     dtype={'SUBJID': str})\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Demographic\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_demographic_data(filepaths=None):\n",
    "    \"\"\"\n",
    "    Read demographic data for all subjects (child, mother, father)\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filenames = ['3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt',\n",
    "                     '3a_dbGaP_SubjectPhenotypes_MaternalDemographicsDS.txt',\n",
    "                     '3a_dbGaP_SubjectPhenotypes_PaternalDemographicsDS.txt']\n",
    "\n",
    "        filepaths = [os.path.join(DBGAP_DIR, filename)\n",
    "                     for filename in filenames\n",
    "                     ]\n",
    "\n",
    "    child_demo_df = pd.read_csv(os.path.join(filepaths[0]),\n",
    "                                delimiter='\\t',\n",
    "                                dtype={'SUBJID': str})\n",
    "\n",
    "    mother_demo_df = pd.read_csv(os.path.join(filepaths[1]),\n",
    "                                 delimiter='\\t',\n",
    "                                 dtype={'SUBJID': str})\n",
    "\n",
    "    father_demo_df = pd.read_csv(os.path.join(filepaths[2]),\n",
    "                                 delimiter='\\t',\n",
    "                                 dtype={'SUBJID': str})\n",
    "\n",
    "    # Combine demographics of all subjects\n",
    "    subject_demo_df = pd.concat(\n",
    "        [child_demo_df, mother_demo_df, father_demo_df])\n",
    "\n",
    "    subject_demo_df.drop_duplicates('SUBJID', inplace=True)\n",
    "    \n",
    "    # Subset of columns\n",
    "    subject_demo_df = subject_demo_df[['RACE', 'ETHNICITY', 'SUBJID']]\n",
    "\n",
    "    def func(row): return \"_\".join(['demographic', str(row.name)])\n",
    "    subject_demo_df['demographic_id'] = subject_demo_df.apply(func, axis=1)\n",
    "\n",
    "\n",
    "    return subject_demo_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Diagnosis\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_diagnosis_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read diagnoses data for all subjects\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filename = '3a_dbGaP_SubjectPhenotypes_PatientDiagnosisDS.txt'\n",
    "        filepath = os.path.join(DBGAP_DIR, filename)\n",
    "\n",
    "    diagnosis_df = pd.read_csv(filepath,\n",
    "                               delimiter='\\t',\n",
    "                               dtype={'SUBJID': str})\n",
    "\n",
    "    def func(row): return \"_\".join(['diagnosis', str(row.name)])\n",
    "    diagnosis_df['diagnosis_id'] = diagnosis_df.apply(func, axis=1)\n",
    "\n",
    "    return diagnosis_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sample\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_subject_sample_data(filepath=None):\n",
    "    \"\"\"\n",
    "    Read sample metadata for all subjects\n",
    "    \"\"\"\n",
    "    if not filepath:\n",
    "        filename = '6a_dbGaP_SubjectSampleMappingDS.txt'\n",
    "        filepath = os.path.join(DBGAP_DIR, filename)\n",
    "\n",
    "    subject_sample_df = pd.read_csv(filepath,\n",
    "                                    delimiter='\\t',\n",
    "                                    dtype={'SUBJID': str})\n",
    "    subject_sample_df.drop_duplicates('SUBJID', inplace=True)\n",
    "\n",
    "    return subject_sample_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Aliquot\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_sample_shipping_manifest_data(*filepaths):\n",
    "    \"\"\"\n",
    "    Read shipping manifest for samples (from PI/sample source center)\n",
    "    \"\"\"\n",
    "    if not filepaths:\n",
    "        filepaths = [os.path.join(ALIQUOT_SHIP_DIR, filename)\n",
    "\n",
    "                     for filename in os.listdir(ALIQUOT_SHIP_DIR)\n",
    "                     ]\n",
    "\n",
    "    # Combine all manifest files\n",
    "    dfs = [pd.read_excel(filepath,\n",
    "                         delimiter='/t',\n",
    "                         dtype={'*barcode': str},\n",
    "                         skiprows=[0, 1],\n",
    "                         header=[6])\n",
    "\n",
    "           for filepath in filepaths\n",
    "\n",
    "           if os.path.basename(filepath).startswith(\"PCGC\")\n",
    "\n",
    "           ]\n",
    "    df = pd.concat(dfs)\n",
    "\n",
    "    # Rename columns\n",
    "    df.columns = map((lambda x: x.lower().lstrip(\"*\")), df.columns)\n",
    "    \n",
    "    # Subset of columns\n",
    "    df = df[['barcode',\n",
    "             'external_id',\n",
    "             'sample_collection_site',\n",
    "             'sample_role',\n",
    "             'concentration_ng_per_ul',\n",
    "             'initial_volume_microliters']]\n",
    "\n",
    "    # Drop rows where id cols are nan\n",
    "    id_cols = [col for col in df.columns if \"id\" in col]\n",
    "    df.dropna(subset=id_cols, inplace=True)\n",
    "\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sequencing experiment (from read group metadata)\n",
    "@reformat_column_names\n",
    "@dropna_rows_cols\n",
    "def read_seq_experiment_data(filepath=None):\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR, \"seidman_metadata.xlsx\")\n",
    "\n",
    "    df = pd.read_excel(filepath, dtype={\"date\": str})\n",
    "    # Rename some columns\n",
    "    df.rename(columns={\"library_name (in original BAM header)\":\n",
    "                       \"library_name\",\n",
    "                       \"barcode\": \"rg_barcode\"}, inplace=True)\n",
    "    df[\"read_length\"] = df[\"read_length\"].apply(\n",
    "        lambda x: int(x.split(\"x\")[0]))\n",
    "    \n",
    "    # Create new columns\n",
    "    df['max_insert_size'] = df['insert_size'].max()\n",
    "    df['mean_insert_size'] = df['insert_size'].mean()\n",
    "    df['mean_read_length'] = df['read_length'].mean()\n",
    "    df['total_reads'] = df['read_length'].count()\n",
    "    \n",
    "    # Subset of columns\n",
    "    df = df[['sample_name',\n",
    "             'library_name',\n",
    "             'rg_barcode',\n",
    "             'run_name',\n",
    "             'read_length',\n",
    "             'date',\n",
    "             'library_strategy',\n",
    "             'library_source',\n",
    "             'library_selection',\n",
    "             'insert_size',\n",
    "             'instrument',\n",
    "             'library_layout',\n",
    "             'max_insert_size',\n",
    "             'mean_insert_size',\n",
    "             'mean_read_length',\n",
    "             'total_reads']]\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extraction - Execution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Study files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CardiacAbnormalitie...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CardiacSurgeriesPro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CatheterizationDS.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CopynumberResultsDS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     study_file_name\n",
       "0  3a_dbGaP_SubjectPhenotypes_CardiacAbnormalitie...\n",
       "1  3a_dbGaP_SubjectPhenotypes_CardiacSurgeriesPro...\n",
       "2   3a_dbGaP_SubjectPhenotypes_CatheterizationDS.txt\n",
       "3  3a_dbGaP_SubjectPhenotypes_CopynumberResultsDS...\n",
       "4      3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_files_df = read_study_file_data()\n",
    "study_files_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>famid</th>\n",
       "      <th>subjid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>210</td>\n",
       "      <td>211</td>\n",
       "      <td>212</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>211</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>212</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>272</td>\n",
       "      <td>273</td>\n",
       "      <td>274</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>273</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   famid subjid  mother  father  is_proband\n",
       "0      1    210     211     212        True\n",
       "1      1    211       0       0       False\n",
       "2      1    212       0       0       False\n",
       "3      2    272     273     274        True\n",
       "4      2    273       0       0       False"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Family \n",
    "family_df = read_family_data()\n",
    "family_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/singhn4/Projects/kids_first/kf-api-dataservice/venv/lib/python3.6/site-packages/ipykernel_launcher.py:44: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hpo_id</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>observed</th>\n",
       "      <th>latest_exam_age</th>\n",
       "      <th>subjid</th>\n",
       "      <th>phenotype_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3673</td>\n",
       "      <td>phenotype_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21296</td>\n",
       "      <td>phenotype_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9653</td>\n",
       "      <td>phenotype_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15386</td>\n",
       "      <td>phenotype_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>none</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26312</td>\n",
       "      <td>phenotype_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  hpo_id                        phenotype observed  latest_exam_age subjid  \\\n",
       "0   None  ABDOMINAL_ABNORMALITIES_PRESENT     none              NaN   3673   \n",
       "1   None  ABDOMINAL_ABNORMALITIES_PRESENT       no              NaN  21296   \n",
       "2   None  ABDOMINAL_ABNORMALITIES_PRESENT       no              NaN   9653   \n",
       "3   None  ABDOMINAL_ABNORMALITIES_PRESENT     none              NaN  15386   \n",
       "4   None  ABDOMINAL_ABNORMALITIES_PRESENT     none              NaN  26312   \n",
       "\n",
       "  phenotype_id  \n",
       "0  phenotype_0  \n",
       "1  phenotype_1  \n",
       "2  phenotype_2  \n",
       "3  phenotype_3  \n",
       "4  phenotype_4  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Phenotypes\n",
    "phenotype_df = read_phenotype_data()\n",
    "phenotype_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5787</td>\n",
       "      <td>M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16252</td>\n",
       "      <td>F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26290</td>\n",
       "      <td>F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14665</td>\n",
       "      <td>F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2248</td>\n",
       "      <td>M</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex\n",
       "0   5787   M\n",
       "1  16252   F\n",
       "2  26290   F\n",
       "3  14665   F\n",
       "4   2248   M"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Gender\n",
    "gender_df = read_gender_data()\n",
    "gender_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>subjid</th>\n",
       "      <th>demographic_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>3673</td>\n",
       "      <td>demographic_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Black or African American</td>\n",
       "      <td>No</td>\n",
       "      <td>21296</td>\n",
       "      <td>demographic_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>9653</td>\n",
       "      <td>demographic_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>15386</td>\n",
       "      <td>demographic_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>White</td>\n",
       "      <td>Yes</td>\n",
       "      <td>26312</td>\n",
       "      <td>demographic_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        race ethnicity subjid demographic_id\n",
       "0                      White        No   3673  demographic_0\n",
       "1  Black or African American        No  21296  demographic_1\n",
       "2                      White        No   9653  demographic_2\n",
       "3                      White        No  15386  demographic_3\n",
       "4                      White       Yes  26312  demographic_4"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Demographic\n",
    "demographic_df = read_demographic_data()\n",
    "demographic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>diagnosis_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3673</td>\n",
       "      <td>L-loop corrected transposition of the great ar...</td>\n",
       "      <td>diagnosis_0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3673</td>\n",
       "      <td>Complete heart block</td>\n",
       "      <td>diagnosis_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>21296</td>\n",
       "      <td>Tetralogy of Fallot with pulmonary atresia</td>\n",
       "      <td>diagnosis_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9653</td>\n",
       "      <td>Right aortic arch with mirror image branching ...</td>\n",
       "      <td>diagnosis_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9653</td>\n",
       "      <td>Ventricular septal defect, membranous</td>\n",
       "      <td>diagnosis_4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid                                          diagnosis diagnosis_id\n",
       "0   3673  L-loop corrected transposition of the great ar...  diagnosis_0\n",
       "1   3673                               Complete heart block  diagnosis_1\n",
       "2  21296         Tetralogy of Fallot with pulmonary atresia  diagnosis_2\n",
       "3   9653  Right aortic arch with mirror image branching ...  diagnosis_3\n",
       "4   9653              Ventricular septal defect, membranous  diagnosis_4"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Diagnosis data\n",
    "diagnosis_df = read_diagnosis_data()\n",
    "diagnosis_df.head()\n",
    "# diagnosis_df[diagnosis_df['subjid'] == '279']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sampid</th>\n",
       "      <th>samp_source</th>\n",
       "      <th>source_sampid</th>\n",
       "      <th>sample_use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>22921</td>\n",
       "      <td>CG0019-7254</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0019-7254</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16165</td>\n",
       "      <td>CG0012-4474</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0012-4474</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9800</td>\n",
       "      <td>CG0009-6098</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0009-6098</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15386</td>\n",
       "      <td>CG0011-4770</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0011-4770</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>570</td>\n",
       "      <td>CG0000-1854</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0000-1854</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid       sampid samp_source source_sampid           sample_use\n",
       "0  22921  CG0019-7254     CORIELL   CG0019-7254  Seq_DNA_WholeGenome\n",
       "1  16165  CG0012-4474     CORIELL   CG0012-4474  Seq_DNA_WholeGenome\n",
       "2   9800  CG0009-6098     CORIELL   CG0009-6098  Seq_DNA_WholeGenome\n",
       "3  15386  CG0011-4770     CORIELL   CG0011-4770  Seq_DNA_WholeGenome\n",
       "4    570  CG0000-1854     CORIELL   CG0000-1854  Seq_DNA_WholeGenome"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sample data\n",
    "subject_sample_df = read_subject_sample_data()\n",
    "subject_sample_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>external_id</th>\n",
       "      <th>sample_collection_site</th>\n",
       "      <th>sample_role</th>\n",
       "      <th>concentration_ng_per_ul</th>\n",
       "      <th>initial_volume_microliters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1125743864</td>\n",
       "      <td>CG0000-6676</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>66.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1125743887</td>\n",
       "      <td>CG0000-6978</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Parent</td>\n",
       "      <td>50.0</td>\n",
       "      <td>65.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1125743888</td>\n",
       "      <td>CG0000-6989</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Parent</td>\n",
       "      <td>50.0</td>\n",
       "      <td>66.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1125743911</td>\n",
       "      <td>CG0000-6731</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>65.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1125743912</td>\n",
       "      <td>CG0000-6738</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Parent</td>\n",
       "      <td>50.0</td>\n",
       "      <td>66.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      barcode  external_id sample_collection_site sample_role  \\\n",
       "0  1125743864  CG0000-6676                  Blood    Affected   \n",
       "1  1125743887  CG0000-6978                  Blood      Parent   \n",
       "2  1125743888  CG0000-6989                  Blood      Parent   \n",
       "3  1125743911  CG0000-6731                  Blood    Affected   \n",
       "4  1125743912  CG0000-6738                  Blood      Parent   \n",
       "\n",
       "   concentration_ng_per_ul  initial_volume_microliters  \n",
       "0                     50.0                        66.1  \n",
       "1                     50.0                        65.7  \n",
       "2                     50.0                        66.0  \n",
       "3                     50.0                        65.7  \n",
       "4                     50.0                        66.2  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Aliquot/Sample Shipping data\n",
    "shipping_manifest_df = read_sample_shipping_manifest_data()\n",
    "shipping_manifest_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>library_name</th>\n",
       "      <th>rg_barcode</th>\n",
       "      <th>run_name</th>\n",
       "      <th>read_length</th>\n",
       "      <th>date</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>insert_size</th>\n",
       "      <th>instrument</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>max_insert_size</th>\n",
       "      <th>mean_insert_size</th>\n",
       "      <th>mean_read_length</th>\n",
       "      <th>total_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CG0023-9153</td>\n",
       "      <td>GMKFCS.ST-00000936-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-1</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_1</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>361</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CG0002-6325</td>\n",
       "      <td>GMKFCS.ST-00000953-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-2</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_2</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>369</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CG0002-6931</td>\n",
       "      <td>GMKFCS.ST-00000954-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-3</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>360</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CG0002-7519</td>\n",
       "      <td>GMKFCS.ST-00003538-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-4</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>365</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CG0002-7976</td>\n",
       "      <td>GMKFCS.ST-00003539-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-5</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>2016-05-10 00:00:00</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>352</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   sample_name               library_name   rg_barcode  \\\n",
       "0  CG0023-9153  GMKFCS.ST-00000936-1_2AMP  HMNMFCCXX-1   \n",
       "1  CG0002-6325  GMKFCS.ST-00000953-1_2AMP  HMNMFCCXX-2   \n",
       "2  CG0002-6931  GMKFCS.ST-00000954-1_2AMP  HMNMFCCXX-3   \n",
       "3  CG0002-7519  GMKFCS.ST-00003538-1_2AMP  HMNMFCCXX-4   \n",
       "4  CG0002-7976  GMKFCS.ST-00003539-1_2AMP  HMNMFCCXX-5   \n",
       "\n",
       "                             run_name  read_length                 date  \\\n",
       "0  160423_ST-E00212_0248_AHMNMFCCXX_1          150  2016-05-10 00:00:00   \n",
       "1  160423_ST-E00212_0248_AHMNMFCCXX_2          150  2016-05-10 00:00:00   \n",
       "2  160423_ST-E00212_0248_AHMNMFCCXX_3          150  2016-05-10 00:00:00   \n",
       "3  160423_ST-E00212_0248_AHMNMFCCXX_4          150  2016-05-10 00:00:00   \n",
       "4  160423_ST-E00212_0248_AHMNMFCCXX_5          150  2016-05-10 00:00:00   \n",
       "\n",
       "  library_strategy library_source library_selection  insert_size   instrument  \\\n",
       "0              WGS        GENOMIC            RANDOM          361  HiSeq X Ten   \n",
       "1              WGS        GENOMIC            RANDOM          369  HiSeq X Ten   \n",
       "2              WGS        GENOMIC            RANDOM          360  HiSeq X Ten   \n",
       "3              WGS        GENOMIC            RANDOM          365  HiSeq X Ten   \n",
       "4              WGS        GENOMIC            RANDOM          352  HiSeq X Ten   \n",
       "\n",
       "  library_layout  max_insert_size  mean_insert_size  mean_read_length  \\\n",
       "0     Paired-End              428        392.684444             150.0   \n",
       "1     Paired-End              428        392.684444             150.0   \n",
       "2     Paired-End              428        392.684444             150.0   \n",
       "3     Paired-End              428        392.684444             150.0   \n",
       "4     Paired-End              428        392.684444             150.0   \n",
       "\n",
       "   total_reads  \n",
       "0          900  \n",
       "1          900  \n",
       "2          900  \n",
       "3          900  \n",
       "4          900  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sequencing experiments\n",
    "seq_exp_df = read_seq_experiment_data()\n",
    "seq_exp_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Explore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Family\n",
      "famid          736\n",
      "subjid        2208\n",
      "mother         737\n",
      "father         737\n",
      "is_proband       2\n",
      "dtype: int64\n",
      "\n",
      "Demographics\n",
      "race                6\n",
      "ethnicity           2\n",
      "subjid            736\n",
      "demographic_id    736\n",
      "dtype: int64\n",
      "\n",
      "Gender\n",
      "race                6\n",
      "ethnicity           2\n",
      "subjid            736\n",
      "demographic_id    736\n",
      "dtype: int64\n",
      "\n",
      "Diagnosis\n",
      "race                6\n",
      "ethnicity           2\n",
      "subjid            736\n",
      "demographic_id    736\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# Participants\n",
    "print(\"Family\")\n",
    "print(family_df.nunique())\n",
    "print(\"\\nDemographics\")\n",
    "print(demographic_df.nunique())\n",
    "print(\"\\nGender\")\n",
    "print(demographic_df.nunique())\n",
    "print(\"\\nDiagnosis\")\n",
    "print(demographic_df.nunique())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unnamed:_0</th>\n",
       "      <th>institution</th>\n",
       "      <th>investigator_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>Harvard Medical School</td>\n",
       "      <td>Christine E. Seidman</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   unnamed:_0             institution     investigator_name\n",
       "0           0  Harvard Medical School  Christine E. Seidman"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Investigator\n",
    "investigator_df = read_investigator_data()\n",
    "investigator_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unnamed:_0</th>\n",
       "      <th>attribution</th>\n",
       "      <th>data_access_authority</th>\n",
       "      <th>study_id</th>\n",
       "      <th>study_name</th>\n",
       "      <th>study_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/projects/gap/cgi-...</td>\n",
       "      <td>dbGaP</td>\n",
       "      <td>phs001138</td>\n",
       "      <td>Discovery of the Genetic Basis of Structural H...</td>\n",
       "      <td>v1.p2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   unnamed:_0                                        attribution  \\\n",
       "0           0  https://www.ncbi.nlm.nih.gov/projects/gap/cgi-...   \n",
       "\n",
       "  data_access_authority   study_id  \\\n",
       "0                 dbGaP  phs001138   \n",
       "\n",
       "                                          study_name study_version  \n",
       "0  Discovery of the Genetic Basis of Structural H...         v1.p2  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Study\n",
    "study_df = read_study_data()\n",
    "study_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_file_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CardiacAbnormalitie...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CardiacSurgeriesPro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CatheterizationDS.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_CopynumberResultsDS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     study_file_name\n",
       "0  3a_dbGaP_SubjectPhenotypes_CardiacAbnormalitie...\n",
       "1  3a_dbGaP_SubjectPhenotypes_CardiacSurgeriesPro...\n",
       "2   3a_dbGaP_SubjectPhenotypes_CatheterizationDS.txt\n",
       "3  3a_dbGaP_SubjectPhenotypes_CopynumberResultsDS...\n",
       "4      3a_dbGaP_SubjectPhenotypes_DemographicsDS.txt"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Study files\n",
    "study_files_df = read_study_file_data()\n",
    "study_files_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>famid</th>\n",
       "      <th>subjid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>210</td>\n",
       "      <td>211</td>\n",
       "      <td>212</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>211</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>212</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>272</td>\n",
       "      <td>273</td>\n",
       "      <td>274</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>273</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   famid subjid  mother  father  is_proband\n",
       "0      1    210     211     212        True\n",
       "1      1    211       0       0       False\n",
       "2      1    212       0       0       False\n",
       "3      2    272     273     274        True\n",
       "4      2    273       0       0       False"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Family\n",
    "family_df = read_family_data()\n",
    "family_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>demographic_id</th>\n",
       "      <th>famid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6688</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_602</td>\n",
       "      <td>260</td>\n",
       "      <td>6689</td>\n",
       "      <td>6690</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>18844</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_591</td>\n",
       "      <td>582</td>\n",
       "      <td>18845</td>\n",
       "      <td>18846</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>25597</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_657</td>\n",
       "      <td>668</td>\n",
       "      <td>25598</td>\n",
       "      <td>25599</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2561</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_617</td>\n",
       "      <td>107</td>\n",
       "      <td>2562</td>\n",
       "      <td>2563</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1295</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_406</td>\n",
       "      <td>49</td>\n",
       "      <td>1296</td>\n",
       "      <td>1297</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2575</td>\n",
       "      <td>F</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_41</td>\n",
       "      <td>108</td>\n",
       "      <td>2576</td>\n",
       "      <td>2577</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4132</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_479</td>\n",
       "      <td>164</td>\n",
       "      <td>4133</td>\n",
       "      <td>4134</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4740</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_343</td>\n",
       "      <td>191</td>\n",
       "      <td>4741</td>\n",
       "      <td>4742</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5725</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_317</td>\n",
       "      <td>224</td>\n",
       "      <td>5726</td>\n",
       "      <td>5727</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>6347</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_464</td>\n",
       "      <td>242</td>\n",
       "      <td>6348</td>\n",
       "      <td>6349</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>6401</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_314</td>\n",
       "      <td>245</td>\n",
       "      <td>6402</td>\n",
       "      <td>6403</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>6659</td>\n",
       "      <td>F</td>\n",
       "      <td>More than one race</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_226</td>\n",
       "      <td>258</td>\n",
       "      <td>6660</td>\n",
       "      <td>6661</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>7222</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_659</td>\n",
       "      <td>283</td>\n",
       "      <td>7223</td>\n",
       "      <td>7224</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>7780</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_16</td>\n",
       "      <td>308</td>\n",
       "      <td>7781</td>\n",
       "      <td>7782</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>7913</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_182</td>\n",
       "      <td>314</td>\n",
       "      <td>7914</td>\n",
       "      <td>7915</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>7946</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_70</td>\n",
       "      <td>316</td>\n",
       "      <td>7947</td>\n",
       "      <td>7948</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>8719</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_259</td>\n",
       "      <td>340</td>\n",
       "      <td>8720</td>\n",
       "      <td>8721</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>8775</td>\n",
       "      <td>M</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_509</td>\n",
       "      <td>346</td>\n",
       "      <td>8776</td>\n",
       "      <td>8777</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>9297</td>\n",
       "      <td>M</td>\n",
       "      <td>More than one race</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_161</td>\n",
       "      <td>361</td>\n",
       "      <td>9298</td>\n",
       "      <td>9299</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>10168</td>\n",
       "      <td>M</td>\n",
       "      <td>Asian</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_722</td>\n",
       "      <td>401</td>\n",
       "      <td>11776</td>\n",
       "      <td>11777</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>15194</td>\n",
       "      <td>M</td>\n",
       "      <td>Asian</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_666</td>\n",
       "      <td>448</td>\n",
       "      <td>15921</td>\n",
       "      <td>14534</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>14885</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_519</td>\n",
       "      <td>459</td>\n",
       "      <td>14886</td>\n",
       "      <td>15029</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>15087</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_558</td>\n",
       "      <td>469</td>\n",
       "      <td>15088</td>\n",
       "      <td>15089</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>15800</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_300</td>\n",
       "      <td>492</td>\n",
       "      <td>15801</td>\n",
       "      <td>15802</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>16089</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_172</td>\n",
       "      <td>497</td>\n",
       "      <td>16090</td>\n",
       "      <td>16091</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>16362</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_663</td>\n",
       "      <td>508</td>\n",
       "      <td>16427</td>\n",
       "      <td>16428</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>16837</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_77</td>\n",
       "      <td>519</td>\n",
       "      <td>16838</td>\n",
       "      <td>16839</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>18342</td>\n",
       "      <td>M</td>\n",
       "      <td>Asian</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_304</td>\n",
       "      <td>567</td>\n",
       "      <td>18343</td>\n",
       "      <td>18344</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>18621</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_286</td>\n",
       "      <td>576</td>\n",
       "      <td>18622</td>\n",
       "      <td>18623</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706</th>\n",
       "      <td>2498</td>\n",
       "      <td>F</td>\n",
       "      <td>Asian</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_378</td>\n",
       "      <td>104</td>\n",
       "      <td>2499</td>\n",
       "      <td>2500</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>707</th>\n",
       "      <td>364</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_534</td>\n",
       "      <td>6</td>\n",
       "      <td>365</td>\n",
       "      <td>366</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>708</th>\n",
       "      <td>4410</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_600</td>\n",
       "      <td>171</td>\n",
       "      <td>4411</td>\n",
       "      <td>4412</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>709</th>\n",
       "      <td>1966</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_384</td>\n",
       "      <td>73</td>\n",
       "      <td>1967</td>\n",
       "      <td>1968</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>710</th>\n",
       "      <td>2121</td>\n",
       "      <td>F</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_395</td>\n",
       "      <td>83</td>\n",
       "      <td>2122</td>\n",
       "      <td>2123</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>711</th>\n",
       "      <td>645</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_79</td>\n",
       "      <td>22</td>\n",
       "      <td>646</td>\n",
       "      <td>647</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>712</th>\n",
       "      <td>2204</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_432</td>\n",
       "      <td>90</td>\n",
       "      <td>2205</td>\n",
       "      <td>2206</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>713</th>\n",
       "      <td>5118</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_227</td>\n",
       "      <td>201</td>\n",
       "      <td>5119</td>\n",
       "      <td>5120</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>714</th>\n",
       "      <td>442</td>\n",
       "      <td>M</td>\n",
       "      <td>More than one race</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_497</td>\n",
       "      <td>10</td>\n",
       "      <td>443</td>\n",
       "      <td>444</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>715</th>\n",
       "      <td>570</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_20</td>\n",
       "      <td>17</td>\n",
       "      <td>571</td>\n",
       "      <td>572</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>716</th>\n",
       "      <td>3009</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_28</td>\n",
       "      <td>125</td>\n",
       "      <td>3010</td>\n",
       "      <td>3011</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>717</th>\n",
       "      <td>4544</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_204</td>\n",
       "      <td>182</td>\n",
       "      <td>4545</td>\n",
       "      <td>4546</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>718</th>\n",
       "      <td>747</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_460</td>\n",
       "      <td>27</td>\n",
       "      <td>748</td>\n",
       "      <td>749</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>719</th>\n",
       "      <td>1703</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_311</td>\n",
       "      <td>64</td>\n",
       "      <td>1704</td>\n",
       "      <td>1705</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>720</th>\n",
       "      <td>3472</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_673</td>\n",
       "      <td>143</td>\n",
       "      <td>3473</td>\n",
       "      <td>3474</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>721</th>\n",
       "      <td>556</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_6</td>\n",
       "      <td>15</td>\n",
       "      <td>557</td>\n",
       "      <td>558</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>722</th>\n",
       "      <td>2195</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_340</td>\n",
       "      <td>88</td>\n",
       "      <td>2196</td>\n",
       "      <td>2197</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>723</th>\n",
       "      <td>5765</td>\n",
       "      <td>M</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_222</td>\n",
       "      <td>226</td>\n",
       "      <td>5766</td>\n",
       "      <td>5767</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>724</th>\n",
       "      <td>1062</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_601</td>\n",
       "      <td>39</td>\n",
       "      <td>1063</td>\n",
       "      <td>1064</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>725</th>\n",
       "      <td>4476</td>\n",
       "      <td>M</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_440</td>\n",
       "      <td>177</td>\n",
       "      <td>4477</td>\n",
       "      <td>4478</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>4947</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_508</td>\n",
       "      <td>197</td>\n",
       "      <td>4948</td>\n",
       "      <td>4949</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>727</th>\n",
       "      <td>2338</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_328</td>\n",
       "      <td>101</td>\n",
       "      <td>2339</td>\n",
       "      <td>2340</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>728</th>\n",
       "      <td>720</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_27</td>\n",
       "      <td>25</td>\n",
       "      <td>721</td>\n",
       "      <td>722</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729</th>\n",
       "      <td>1118</td>\n",
       "      <td>M</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_307</td>\n",
       "      <td>41</td>\n",
       "      <td>1119</td>\n",
       "      <td>1120</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>730</th>\n",
       "      <td>26418</td>\n",
       "      <td>M</td>\n",
       "      <td>More than one race</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_184</td>\n",
       "      <td>689</td>\n",
       "      <td>26419</td>\n",
       "      <td>26420</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>731</th>\n",
       "      <td>22481</td>\n",
       "      <td>F</td>\n",
       "      <td>More than one race</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_215</td>\n",
       "      <td>705</td>\n",
       "      <td>22482</td>\n",
       "      <td>22483</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>732</th>\n",
       "      <td>2118</td>\n",
       "      <td>F</td>\n",
       "      <td>Asian</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_361</td>\n",
       "      <td>82</td>\n",
       "      <td>2119</td>\n",
       "      <td>2120</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>733</th>\n",
       "      <td>1306</td>\n",
       "      <td>F</td>\n",
       "      <td>More than one race</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_556</td>\n",
       "      <td>50</td>\n",
       "      <td>1307</td>\n",
       "      <td>1308</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>734</th>\n",
       "      <td>2290</td>\n",
       "      <td>F</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>Yes</td>\n",
       "      <td>demographic_18</td>\n",
       "      <td>96</td>\n",
       "      <td>2291</td>\n",
       "      <td>2292</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>735</th>\n",
       "      <td>2655</td>\n",
       "      <td>M</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_725</td>\n",
       "      <td>114</td>\n",
       "      <td>2656</td>\n",
       "      <td>2657</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>736 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    subjid sex                       race ethnicity   demographic_id  famid  \\\n",
       "0      390   F                      White        No   demographic_64      8   \n",
       "1     6688   M                      White        No  demographic_602    260   \n",
       "2    18844   M                      White        No  demographic_591    582   \n",
       "3    25597   M                      White        No  demographic_657    668   \n",
       "4     2561   M                      White        No  demographic_617    107   \n",
       "5     1295   F                      White        No  demographic_406     49   \n",
       "6     2575   F  Black or African American        No   demographic_41    108   \n",
       "7     4132   F                      White        No  demographic_479    164   \n",
       "8     4740   M                      White        No  demographic_343    191   \n",
       "9     5725   M                      White        No  demographic_317    224   \n",
       "10    6347   M                      White        No  demographic_464    242   \n",
       "11    6401   M                      White        No  demographic_314    245   \n",
       "12    6659   F         More than one race        No  demographic_226    258   \n",
       "13    7222   F                      White        No  demographic_659    283   \n",
       "14    7780   F                      White        No   demographic_16    308   \n",
       "15    7913   M                      White        No  demographic_182    314   \n",
       "16    7946   M                      White        No   demographic_70    316   \n",
       "17    8719   F                      White        No  demographic_259    340   \n",
       "18    8775   M  Black or African American       Yes  demographic_509    346   \n",
       "19    9297   M         More than one race        No  demographic_161    361   \n",
       "20   10168   M                      Asian        No  demographic_722    401   \n",
       "21   15194   M                      Asian        No  demographic_666    448   \n",
       "22   14885   F                      White       Yes  demographic_519    459   \n",
       "23   15087   F                      White       Yes  demographic_558    469   \n",
       "24   15800   M                      White        No  demographic_300    492   \n",
       "25   16089   M                      White        No  demographic_172    497   \n",
       "26   16362   F                      White        No  demographic_663    508   \n",
       "27   16837   M                      White       Yes   demographic_77    519   \n",
       "28   18342   M                      Asian        No  demographic_304    567   \n",
       "29   18621   M                      White        No  demographic_286    576   \n",
       "..     ...  ..                        ...       ...              ...    ...   \n",
       "706   2498   F                      Asian        No  demographic_378    104   \n",
       "707    364   F                      White        No  demographic_534      6   \n",
       "708   4410   M                      White        No  demographic_600    171   \n",
       "709   1966   M                      White        No  demographic_384     73   \n",
       "710   2121   F  Black or African American        No  demographic_395     83   \n",
       "711    645   M                      White        No   demographic_79     22   \n",
       "712   2204   F                      White        No  demographic_432     90   \n",
       "713   5118   F                      White        No  demographic_227    201   \n",
       "714    442   M         More than one race        No  demographic_497     10   \n",
       "715    570   F                      White        No   demographic_20     17   \n",
       "716   3009   M                      White       Yes   demographic_28    125   \n",
       "717   4544   M                      White        No  demographic_204    182   \n",
       "718    747   F                      White        No  demographic_460     27   \n",
       "719   1703   F                      White        No  demographic_311     64   \n",
       "720   3472   F                      White        No  demographic_673    143   \n",
       "721    556   M                      White        No    demographic_6     15   \n",
       "722   2195   F                      White        No  demographic_340     88   \n",
       "723   5765   M  Black or African American        No  demographic_222    226   \n",
       "724   1062   F                      White        No  demographic_601     39   \n",
       "725   4476   M  Black or African American       Yes  demographic_440    177   \n",
       "726   4947   M                      White        No  demographic_508    197   \n",
       "727   2338   M                      White        No  demographic_328    101   \n",
       "728    720   F                      White       Yes   demographic_27     25   \n",
       "729   1118   M                      White        No  demographic_307     41   \n",
       "730  26418   M         More than one race        No  demographic_184    689   \n",
       "731  22481   F         More than one race       Yes  demographic_215    705   \n",
       "732   2118   F                      Asian        No  demographic_361     82   \n",
       "733   1306   F         More than one race       Yes  demographic_556     50   \n",
       "734   2290   F                    Unknown       Yes   demographic_18     96   \n",
       "735   2655   M                    Unknown        No  demographic_725    114   \n",
       "\n",
       "     mother  father  is_proband  \n",
       "0       391     392        True  \n",
       "1      6689    6690        True  \n",
       "2     18845   18846        True  \n",
       "3     25598   25599        True  \n",
       "4      2562    2563        True  \n",
       "5      1296    1297        True  \n",
       "6      2576    2577        True  \n",
       "7      4133    4134        True  \n",
       "8      4741    4742        True  \n",
       "9      5726    5727        True  \n",
       "10     6348    6349        True  \n",
       "11     6402    6403        True  \n",
       "12     6660    6661        True  \n",
       "13     7223    7224        True  \n",
       "14     7781    7782        True  \n",
       "15     7914    7915        True  \n",
       "16     7947    7948        True  \n",
       "17     8720    8721        True  \n",
       "18     8776    8777        True  \n",
       "19     9298    9299        True  \n",
       "20    11776   11777        True  \n",
       "21    15921   14534        True  \n",
       "22    14886   15029        True  \n",
       "23    15088   15089        True  \n",
       "24    15801   15802        True  \n",
       "25    16090   16091        True  \n",
       "26    16427   16428        True  \n",
       "27    16838   16839        True  \n",
       "28    18343   18344        True  \n",
       "29    18622   18623        True  \n",
       "..      ...     ...         ...  \n",
       "706    2499    2500        True  \n",
       "707     365     366        True  \n",
       "708    4411    4412        True  \n",
       "709    1967    1968        True  \n",
       "710    2122    2123        True  \n",
       "711     646     647        True  \n",
       "712    2205    2206        True  \n",
       "713    5119    5120        True  \n",
       "714     443     444        True  \n",
       "715     571     572        True  \n",
       "716    3010    3011        True  \n",
       "717    4545    4546        True  \n",
       "718     748     749        True  \n",
       "719    1704    1705        True  \n",
       "720    3473    3474        True  \n",
       "721     557     558        True  \n",
       "722    2196    2197        True  \n",
       "723    5766    5767        True  \n",
       "724    1063    1064        True  \n",
       "725    4477    4478        True  \n",
       "726    4948    4949        True  \n",
       "727    2339    2340        True  \n",
       "728     721     722        True  \n",
       "729    1119    1120        True  \n",
       "730   26419   26420        True  \n",
       "731   22482   22483        True  \n",
       "732    2119    2120        True  \n",
       "733    1307    1308        True  \n",
       "734    2291    2292        True  \n",
       "735    2656    2657        True  \n",
       "\n",
       "[736 rows x 9 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create participant df\n",
    "# Merge Gender + Demographics\n",
    "gender_demo_df = pd.merge(gender_df, demographic_df, on='subjid')\n",
    "# Add Family\n",
    "df1 = pd.merge(gender_demo_df, family_df, on='subjid')\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>demographic_id</th>\n",
       "      <th>famid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>diagnosis_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis, bilateral branch pulmonary...</td>\n",
       "      <td>diagnosis_232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Single ventricle comprised of mostly left vent...</td>\n",
       "      <td>diagnosis_233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>L-loop transposition of the great arteries</td>\n",
       "      <td>diagnosis_234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Double outlet left ventricle</td>\n",
       "      <td>diagnosis_235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis</td>\n",
       "      <td>diagnosis_236</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex   race ethnicity  demographic_id  famid  mother  father  \\\n",
       "0    390   F  White        No  demographic_64      8     391     392   \n",
       "1    390   F  White        No  demographic_64      8     391     392   \n",
       "2    390   F  White        No  demographic_64      8     391     392   \n",
       "3    390   F  White        No  demographic_64      8     391     392   \n",
       "4    390   F  White        No  demographic_64      8     391     392   \n",
       "\n",
       "   is_proband                                          diagnosis  \\\n",
       "0        True  Pulmonary stenosis, bilateral branch pulmonary...   \n",
       "1        True  Single ventricle comprised of mostly left vent...   \n",
       "2        True         L-loop transposition of the great arteries   \n",
       "3        True                       Double outlet left ventricle   \n",
       "4        True                                 Pulmonary stenosis   \n",
       "\n",
       "    diagnosis_id  \n",
       "0  diagnosis_232  \n",
       "1  diagnosis_233  \n",
       "2  diagnosis_234  \n",
       "3  diagnosis_235  \n",
       "4  diagnosis_236  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge Diagnosis\n",
    "df2 = pd.merge(df1, diagnosis_df, on='subjid')\n",
    "df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>demographic_id</th>\n",
       "      <th>famid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>diagnosis_id</th>\n",
       "      <th>sampid</th>\n",
       "      <th>samp_source</th>\n",
       "      <th>source_sampid</th>\n",
       "      <th>sample_use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis, bilateral branch pulmonary...</td>\n",
       "      <td>diagnosis_232</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Single ventricle comprised of mostly left vent...</td>\n",
       "      <td>diagnosis_233</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>L-loop transposition of the great arteries</td>\n",
       "      <td>diagnosis_234</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Double outlet left ventricle</td>\n",
       "      <td>diagnosis_235</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis</td>\n",
       "      <td>diagnosis_236</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex   race ethnicity  demographic_id  famid  mother  father  \\\n",
       "0    390   F  White        No  demographic_64      8     391     392   \n",
       "1    390   F  White        No  demographic_64      8     391     392   \n",
       "2    390   F  White        No  demographic_64      8     391     392   \n",
       "3    390   F  White        No  demographic_64      8     391     392   \n",
       "4    390   F  White        No  demographic_64      8     391     392   \n",
       "\n",
       "   is_proband                                          diagnosis  \\\n",
       "0        True  Pulmonary stenosis, bilateral branch pulmonary...   \n",
       "1        True  Single ventricle comprised of mostly left vent...   \n",
       "2        True         L-loop transposition of the great arteries   \n",
       "3        True                       Double outlet left ventricle   \n",
       "4        True                                 Pulmonary stenosis   \n",
       "\n",
       "    diagnosis_id       sampid samp_source source_sampid           sample_use  \n",
       "0  diagnosis_232  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  \n",
       "1  diagnosis_233  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  \n",
       "2  diagnosis_234  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  \n",
       "3  diagnosis_235  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  \n",
       "4  diagnosis_236  CG0003-5269     CORIELL   CG0003-5269  Seq_DNA_WholeGenome  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge Sample\n",
    "df3 = pd.merge(df2, subject_sample_df, on='subjid')\n",
    "df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>demographic_id</th>\n",
       "      <th>famid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>sampid</th>\n",
       "      <th>samp_source</th>\n",
       "      <th>source_sampid</th>\n",
       "      <th>sample_use</th>\n",
       "      <th>barcode</th>\n",
       "      <th>external_id</th>\n",
       "      <th>sample_collection_site</th>\n",
       "      <th>sample_role</th>\n",
       "      <th>concentration_ng_per_ul</th>\n",
       "      <th>initial_volume_microliters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis, bilateral branch pulmonary...</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Single ventricle comprised of mostly left vent...</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>L-loop transposition of the great arteries</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Double outlet left ventricle</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis</td>\n",
       "      <td>...</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>CORIELL</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Seq_DNA_WholeGenome</td>\n",
       "      <td>1125743958</td>\n",
       "      <td>CG0003-5269</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Affected</td>\n",
       "      <td>50.0</td>\n",
       "      <td>60.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex   race ethnicity  demographic_id  famid  mother  father  \\\n",
       "0    390   F  White        No  demographic_64      8     391     392   \n",
       "1    390   F  White        No  demographic_64      8     391     392   \n",
       "2    390   F  White        No  demographic_64      8     391     392   \n",
       "3    390   F  White        No  demographic_64      8     391     392   \n",
       "4    390   F  White        No  demographic_64      8     391     392   \n",
       "\n",
       "   is_proband                                          diagnosis  \\\n",
       "0        True  Pulmonary stenosis, bilateral branch pulmonary...   \n",
       "1        True  Single ventricle comprised of mostly left vent...   \n",
       "2        True         L-loop transposition of the great arteries   \n",
       "3        True                       Double outlet left ventricle   \n",
       "4        True                                 Pulmonary stenosis   \n",
       "\n",
       "              ...                   sampid samp_source source_sampid  \\\n",
       "0             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "1             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "2             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "3             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "4             ...              CG0003-5269     CORIELL   CG0003-5269   \n",
       "\n",
       "            sample_use     barcode  external_id sample_collection_site  \\\n",
       "0  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "1  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "2  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "3  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "4  Seq_DNA_WholeGenome  1125743958  CG0003-5269                  Blood   \n",
       "\n",
       "  sample_role concentration_ng_per_ul  initial_volume_microliters  \n",
       "0    Affected                    50.0                        60.1  \n",
       "1    Affected                    50.0                        60.1  \n",
       "2    Affected                    50.0                        60.1  \n",
       "3    Affected                    50.0                        60.1  \n",
       "4    Affected                    50.0                        60.1  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge Aliquot\n",
    "df4 = pd.merge(df3, shipping_manifest_df, left_on='sampid', right_on='external_id')\n",
    "df4.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subjid</th>\n",
       "      <th>sex</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>demographic_id</th>\n",
       "      <th>famid</th>\n",
       "      <th>mother</th>\n",
       "      <th>father</th>\n",
       "      <th>is_proband</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>insert_size</th>\n",
       "      <th>instrument</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>max_insert_size</th>\n",
       "      <th>mean_insert_size</th>\n",
       "      <th>mean_read_length</th>\n",
       "      <th>total_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis, bilateral branch pulmonary...</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Single ventricle comprised of mostly left vent...</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>L-loop transposition of the great arteries</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Double outlet left ventricle</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>390</td>\n",
       "      <td>F</td>\n",
       "      <td>White</td>\n",
       "      <td>No</td>\n",
       "      <td>demographic_64</td>\n",
       "      <td>8</td>\n",
       "      <td>391</td>\n",
       "      <td>392</td>\n",
       "      <td>True</td>\n",
       "      <td>Pulmonary stenosis</td>\n",
       "      <td>...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>387</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>428</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  subjid sex   race ethnicity  demographic_id  famid  mother  father  \\\n",
       "0    390   F  White        No  demographic_64      8     391     392   \n",
       "1    390   F  White        No  demographic_64      8     391     392   \n",
       "2    390   F  White        No  demographic_64      8     391     392   \n",
       "3    390   F  White        No  demographic_64      8     391     392   \n",
       "4    390   F  White        No  demographic_64      8     391     392   \n",
       "\n",
       "   is_proband                                          diagnosis     ...       \\\n",
       "0        True  Pulmonary stenosis, bilateral branch pulmonary...     ...        \n",
       "1        True  Single ventricle comprised of mostly left vent...     ...        \n",
       "2        True         L-loop transposition of the great arteries     ...        \n",
       "3        True                       Double outlet left ventricle     ...        \n",
       "4        True                                 Pulmonary stenosis     ...        \n",
       "\n",
       "  library_strategy library_source library_selection insert_size   instrument  \\\n",
       "0              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "1              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "2              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "3              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "4              WGS        GENOMIC            RANDOM         387  HiSeq X Ten   \n",
       "\n",
       "  library_layout max_insert_size mean_insert_size mean_read_length  \\\n",
       "0     Paired-End             428       392.684444            150.0   \n",
       "1     Paired-End             428       392.684444            150.0   \n",
       "2     Paired-End             428       392.684444            150.0   \n",
       "3     Paired-End             428       392.684444            150.0   \n",
       "4     Paired-End             428       392.684444            150.0   \n",
       "\n",
       "   total_reads  \n",
       "0          900  \n",
       "1          900  \n",
       "2          900  \n",
       "3          900  \n",
       "4          900  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge Sequencing Experiment\n",
    "full_participant_df = pd.merge(df4, seq_exp_df, left_on='external_id', right_on='sample_name')\n",
    "full_participant_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create study\n",
    "study = {\n",
    "    'data_access_authority': 'dbGaP',\n",
    "    'study_id': 'phs001138',\n",
    "    'study_version': 'v1.p2',\n",
    "    'study_name': 'Discovery of the Genetic Basis of Structural Heart'\n",
    "    'and Other Birth Defects',\n",
    "    'attribution': 'https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/'\n",
    "    'GetAcknowledgementStatement.cgi?study_id=phs001138.v1.p2'\n",
    "}\n",
    "study_df = pd.DataFrame([study])\n",
    "study_df.to_csv(os.path.join(DATA_DIR, 'study.txt'))\n",
    "\n",
    "# Create investigator\n",
    "invest = {\n",
    "    'investigator_name': 'Christine E. Seidman',\n",
    "    'institution': 'Harvard Medical School'\n",
    "}\n",
    "inv_df = pd.DataFrame([invest])\n",
    "inv_df.to_csv(os.path.join(DATA_DIR, 'investigator.txt'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "def _add_study_cols(study_df, df):\n",
    "    # Add study cols to a df\n",
    "    cols = study_df.columns.tolist()\n",
    "    row = study_df.iloc[0]\n",
    "    for col in cols:\n",
    "        df[col] = row[col]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add study to full participant df\n",
    "_add_study_cols(study_df, full_participant_df)\n",
    "\n",
    "# Add study to basic participant df\n",
    "participant_df = _add_study_cols(study_df, family_df)\n",
    "\n",
    "# Add study to investigator df\n",
    "study_investigator_df =_add_study_cols(study_df, investigator_df)\n",
    "\n",
    "# Add study to study files df\n",
    "study_study_files_df = _add_study_cols(study_df, study_files_df)\n",
    "\n",
    "# Phenotype df\n",
    "phenotype_participant_df = pd.merge(phenotype_df, participant_df,\n",
    "                                    on='subjid')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adding 1:10 participants to session\n",
      "['participant_1',\n",
      " 'participant_2',\n",
      " 'participant_3',\n",
      " 'participant_4',\n",
      " 'participant_5',\n",
      " 'participant_6',\n",
      " 'participant_7',\n",
      " 'participant_8',\n",
      " 'participant_9']\n",
      "Flushing 10 participant\n",
      "s\n",
      "Adding 11:20 participants to session\n",
      "['participant_11',\n",
      " 'participant_12',\n",
      " 'participant_13',\n",
      " 'participant_14',\n",
      " 'participant_15',\n",
      " 'participant_16',\n",
      " 'participant_17',\n",
      " 'participant_18',\n",
      " 'participant_19']\n",
      "Flushing 10 participant\n",
      "s\n",
      "Flushing remaining 3 participants to session\n",
      "\n",
      "['participant_0', 'participant_20', 'participant_21']\n"
     ]
    }
   ],
   "source": [
    "from pprint import pprint\n",
    "n = 22\n",
    "chunk_size = 10\n",
    "entity_type ='participant'\n",
    "entities = ['{}_{}'.format(entity_type, j) for j in range(n)]\n",
    "for i in range(0, n, chunk_size):\n",
    "    chunk = entities[i - chunk_size:i]\n",
    "    if chunk:\n",
    "        start = i - chunk_size + 1\n",
    "        print('Adding {}:{} {}s to session'.format(start, i,\n",
    "                                                   entity_type))\n",
    "        pprint([e for e in entities[start:i]])\n",
    "        print('Flushing {} {}\\ns'.format(chunk_size, entity_type))\n",
    "\n",
    "print('Flushing remaining {} {}s to session\\n'.format(\n",
    "    len(entities[i:]) + 1, entity_type))\n",
    "\n",
    "remaining = entities[0:1] + entities[i:]\n",
    "\n",
    "pprint([e for e in remaining])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "def read_json(filepath):\n",
    "    with open(filepath, 'r') as json_file:\n",
    "        return json.load(json_file)\n",
    "\n",
    "\n",
    "def write_json(data, filepath):\n",
    "    with open(filepath, 'w') as json_file:\n",
    "        json.dump(data, json_file, sort_keys=True, indent=4, separators=(',', ':'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "fp = os.path.join(DATA_DIR, 'genomic_file_uuid.json')\n",
    "file_json = read_json(fp)\n",
    "write_json(file_json, fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Genomic file info df\n",
    "def read_genomic_file_info(filepath=None):\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR, 'genomic_file_uuid.json')\n",
    "        \n",
    "    def get_ext(fp):\n",
    "        filename = os.path.basename(fp)\n",
    "        parts = filename.split('.')\n",
    "        if len(parts) > 2:\n",
    "            ext = '.'.join(parts[1:])\n",
    "        else:\n",
    "            ext = parts[-1]\n",
    "        return ext\n",
    "\n",
    "    with open(filepath, 'r') as json_file:\n",
    "        uuid_dict = json.load(json_file)\n",
    "\n",
    "    gf_dicts = []\n",
    "    for k, v in uuid_dict.items():\n",
    "        file_info = {\n",
    "            'uuid': v['did'],\n",
    "            'md5sum': v['hashes']['md5'],\n",
    "            'file_url': v['urls'][0],\n",
    "            'file_size': v['size'],\n",
    "            'data_type': 'submitted aligned reads',\n",
    "            'file_format': get_ext(v['urls'][0]),\n",
    "            'file_name': os.path.basename(v['urls'][0])\n",
    "        }\n",
    "        gf_dicts.append(file_info)\n",
    "        \n",
    "    return pd.DataFrame(gf_dicts)\n",
    "\n",
    "def read_sample_gf_data(filepath=None):\n",
    "    if not filepath:\n",
    "        filepath = os.path.join(DATA_DIR, 'manifests', 'GMKF_BAMsampleIDs.xlsx')\n",
    "    df = pd.read_excel(filepath)\n",
    "    df = df.loc[df['Cohort'] == 'GMKF-Seidman']\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cohort</th>\n",
       "      <th>BAM sample ID</th>\n",
       "      <th>Library</th>\n",
       "      <th>HGSC Sample ID</th>\n",
       "      <th>dbgap_subject_id</th>\n",
       "      <th>sample_name</th>\n",
       "      <th>library_name</th>\n",
       "      <th>rg_barcode</th>\n",
       "      <th>run_name</th>\n",
       "      <th>read_length</th>\n",
       "      <th>...</th>\n",
       "      <th>mean_insert_size</th>\n",
       "      <th>mean_read_length</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>data_type</th>\n",
       "      <th>file_format</th>\n",
       "      <th>file_name</th>\n",
       "      <th>file_size</th>\n",
       "      <th>file_url</th>\n",
       "      <th>md5sum</th>\n",
       "      <th>uuid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNMFCCXX-1.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000936-1_2pA</td>\n",
       "      <td>ST-00000936</td>\n",
       "      <td>CG0023-9153</td>\n",
       "      <td>CG0023-9153</td>\n",
       "      <td>GMKFCS.ST-00000936-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-1</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_1</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNMFCCXX-1.hgv.bam</td>\n",
       "      <td>64365517822</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNMFCCXX-1.hgv.bam</td>\n",
       "      <td>6f111ed7b35e421fce94571eba8e5be8</td>\n",
       "      <td>e88c2d14-0b7e-43e8-9f7e-7020793b6ebe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNMFCCXX-2.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000953-1_2pA</td>\n",
       "      <td>ST-00000953</td>\n",
       "      <td>CG0002-6325</td>\n",
       "      <td>CG0002-6325</td>\n",
       "      <td>GMKFCS.ST-00000953-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-2</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_2</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNMFCCXX-2.hgv.bam</td>\n",
       "      <td>63435348733</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNMFCCXX-2.hgv.bam</td>\n",
       "      <td>fb82871da4cc68c70a971dd55677d933</td>\n",
       "      <td>40fb29d7-71e1-426e-9678-a54f22a4f80e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNMFCCXX-3.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000954-1_2pA</td>\n",
       "      <td>ST-00000954</td>\n",
       "      <td>CG0002-6931</td>\n",
       "      <td>CG0002-6931</td>\n",
       "      <td>GMKFCS.ST-00000954-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-3</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNMFCCXX-3.hgv.bam</td>\n",
       "      <td>63739965918</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNMFCCXX-3.hgv.bam</td>\n",
       "      <td>748932ce3d9edfc9fba57291c6f632a5</td>\n",
       "      <td>d19eeeae-6acd-4f08-b5c8-6f200160ebed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNMFCCXX-4.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00003538-1_2pA</td>\n",
       "      <td>ST-00003538</td>\n",
       "      <td>CG0002-7519</td>\n",
       "      <td>CG0002-7519</td>\n",
       "      <td>GMKFCS.ST-00003538-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-4</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNMFCCXX-4.hgv.bam</td>\n",
       "      <td>62635938381</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNMFCCXX-4.hgv.bam</td>\n",
       "      <td>b60d5db835db5544e43fa2113a0cc08a</td>\n",
       "      <td>c5d91600-e286-4ddd-bbc0-e111b079f8bf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNMFCCXX-5.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00003539-1_2pA</td>\n",
       "      <td>ST-00003539</td>\n",
       "      <td>CG0002-7976</td>\n",
       "      <td>CG0002-7976</td>\n",
       "      <td>GMKFCS.ST-00003539-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-5</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNMFCCXX-5.hgv.bam</td>\n",
       "      <td>64626450198</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNMFCCXX-5.hgv.bam</td>\n",
       "      <td>068b52d7685a5e1916dbb784da850f3b</td>\n",
       "      <td>6904e887-901c-4df6-8579-0ae55cc11125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNMFCCXX-6.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00003561-1_2pA</td>\n",
       "      <td>ST-00003561</td>\n",
       "      <td>CG0002-7848</td>\n",
       "      <td>CG0002-7848</td>\n",
       "      <td>GMKFCS.ST-00003561-1_2AMP</td>\n",
       "      <td>HMNMFCCXX-6</td>\n",
       "      <td>160423_ST-E00212_0248_AHMNMFCCXX_6</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNMFCCXX-6.hgv.bam</td>\n",
       "      <td>62979984038</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNMFCCXX-6.hgv.bam</td>\n",
       "      <td>e36e9551a23d5065cbfd9eea8069ee73</td>\n",
       "      <td>6b60464f-31cd-411c-91b4-c33d535cb01f</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNVCCCXX-1.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000768-1_2pA</td>\n",
       "      <td>ST-00000768</td>\n",
       "      <td>CG0000-3840</td>\n",
       "      <td>CG0000-3840</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000768-1_2pA</td>\n",
       "      <td>HMNVCCCXX-1</td>\n",
       "      <td>160402_E00380_0096_BHMNVCCCXX_1</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNVCCCXX-1.hgv.bam</td>\n",
       "      <td>68933421158</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNVCCCXX-1.hgv.bam</td>\n",
       "      <td>be2c10950413714413e35743187248c7</td>\n",
       "      <td>1ab280b9-e054-4e25-a88a-d7566b02510a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNVCCCXX-2.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000580-1_2pA</td>\n",
       "      <td>ST-00000580</td>\n",
       "      <td>CG0000-2758</td>\n",
       "      <td>CG0000-2758</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000580-1_2pA</td>\n",
       "      <td>HMNVCCCXX-2</td>\n",
       "      <td>160402_E00380_0096_BHMNVCCCXX_2</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNVCCCXX-2.hgv.bam</td>\n",
       "      <td>70510006515</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNVCCCXX-2.hgv.bam</td>\n",
       "      <td>794bb59888d85fd092d45c1e7da9dc2e</td>\n",
       "      <td>a79803fc-372e-40fa-b62d-29bb5b5fbaf0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNVCCCXX-3.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000589-1_2pA</td>\n",
       "      <td>ST-00000589</td>\n",
       "      <td>CG0000-0344</td>\n",
       "      <td>CG0000-0344</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000589-1_2pA</td>\n",
       "      <td>HMNVCCCXX-3</td>\n",
       "      <td>160402_E00380_0096_BHMNVCCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNVCCCXX-3.hgv.bam</td>\n",
       "      <td>70748511577</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNVCCCXX-3.hgv.bam</td>\n",
       "      <td>02f5e1d8fc3cf41a1993ea369a88411e</td>\n",
       "      <td>5cea8a2f-cfd6-45da-9cf0-0aff9bc3185c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNVCCCXX-4.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000991-1_2pA</td>\n",
       "      <td>ST-00000991</td>\n",
       "      <td>CG0000-4744</td>\n",
       "      <td>CG0000-4744</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000991-1_2pA</td>\n",
       "      <td>HMNVCCCXX-4</td>\n",
       "      <td>160402_E00380_0096_BHMNVCCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNVCCCXX-4.hgv.bam</td>\n",
       "      <td>70496953371</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNVCCCXX-4.hgv.bam</td>\n",
       "      <td>abcdeb907c26445979b5ef79b327835e</td>\n",
       "      <td>1d6a481c-a86a-4cc2-acc1-45829d0dbb8e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNVCCCXX-5.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000992-1_2pA</td>\n",
       "      <td>ST-00000992</td>\n",
       "      <td>CG0000-4752</td>\n",
       "      <td>CG0000-4752</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000992-1_2pA</td>\n",
       "      <td>HMNVCCCXX-5</td>\n",
       "      <td>160402_E00380_0096_BHMNVCCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNVCCCXX-5.hgv.bam</td>\n",
       "      <td>69942859118</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNVCCCXX-5.hgv.bam</td>\n",
       "      <td>5bfe2294575b6ddd11bf285ca802b6b8</td>\n",
       "      <td>dd743abd-b615-400e-a295-937bc5af13be</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNVCCCXX-6.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000603-1_2pA</td>\n",
       "      <td>ST-00000603</td>\n",
       "      <td>CG0000-6252</td>\n",
       "      <td>CG0000-6252</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000603-1_2pA</td>\n",
       "      <td>HMNVCCCXX-6</td>\n",
       "      <td>160402_E00380_0096_BHMNVCCCXX_6</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNVCCCXX-6.hgv.bam</td>\n",
       "      <td>72853808794</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNVCCCXX-6.hgv.bam</td>\n",
       "      <td>db829e78b533409b5540c2c6ac0afdb9</td>\n",
       "      <td>7639d110-3c0d-4a73-8cde-ab42b9bbc4ce</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNVCCCXX-7.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000604-1_2pA</td>\n",
       "      <td>ST-00000604</td>\n",
       "      <td>CG0000-6259</td>\n",
       "      <td>CG0000-6259</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000604-1_2pA</td>\n",
       "      <td>HMNVCCCXX-7</td>\n",
       "      <td>160402_E00380_0096_BHMNVCCCXX_7</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNVCCCXX-7.hgv.bam</td>\n",
       "      <td>72605537636</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNVCCCXX-7.hgv.bam</td>\n",
       "      <td>a277e75e3da743579816a49c88bafde9</td>\n",
       "      <td>60582705-673d-4a60-9648-1e8fa71e0aaf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNVCCCXX-8.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000611-1_2pA</td>\n",
       "      <td>ST-00000611</td>\n",
       "      <td>CG0000-6267</td>\n",
       "      <td>CG0000-6267</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000611-1_2pA</td>\n",
       "      <td>HMNVCCCXX-8</td>\n",
       "      <td>160402_E00380_0096_BHMNVCCCXX_8</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNVCCCXX-8.hgv.bam</td>\n",
       "      <td>70275800752</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNVCCCXX-8.hgv.bam</td>\n",
       "      <td>0433ba0e658ec4ac9d0a71d4e2e507c5</td>\n",
       "      <td>2fa09a20-ce15-464d-a326-ffabd8fb7dde</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNWGCCXX-1.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000818-1_2pA</td>\n",
       "      <td>ST-00000818</td>\n",
       "      <td>CG0001-4817</td>\n",
       "      <td>CG0001-4817</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000818-1_2pA</td>\n",
       "      <td>HMNWGCCXX-1</td>\n",
       "      <td>160404_ST-E00270_0172_AHMNWGCCXX_1</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNWGCCXX-1.hgv.bam</td>\n",
       "      <td>63701059861</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNWGCCXX-1.hgv.bam</td>\n",
       "      <td>5b669a9e5ab932f88c7988a92c9a0fb2</td>\n",
       "      <td>da029205-5dd0-4d73-bd85-3ebefdbbd0c2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNWGCCXX-2.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000872-1_2pA</td>\n",
       "      <td>ST-00000872</td>\n",
       "      <td>CG0002-0380</td>\n",
       "      <td>CG0002-0380</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000872-1_2pA</td>\n",
       "      <td>HMNWGCCXX-2</td>\n",
       "      <td>160404_ST-E00270_0172_AHMNWGCCXX_2</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNWGCCXX-2.hgv.bam</td>\n",
       "      <td>63130114606</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNWGCCXX-2.hgv.bam</td>\n",
       "      <td>cb674e8810d8b1b3b10839c42d5e458e</td>\n",
       "      <td>6a3d20c1-e8ff-4d3c-88b7-94d3a8235414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNWGCCXX-3.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000890-1_2pA</td>\n",
       "      <td>ST-00000890</td>\n",
       "      <td>CG0002-0677</td>\n",
       "      <td>CG0002-0677</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000890-1_2pA</td>\n",
       "      <td>HMNWGCCXX-3</td>\n",
       "      <td>160404_ST-E00270_0172_AHMNWGCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNWGCCXX-3.hgv.bam</td>\n",
       "      <td>66284290234</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNWGCCXX-3.hgv.bam</td>\n",
       "      <td>e646f30bbc3cab5464d35cb09c0b7222</td>\n",
       "      <td>a3f5d36e-98c6-4de4-82b9-7b2262454b0e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNWGCCXX-4.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000891-1_2pA</td>\n",
       "      <td>ST-00000891</td>\n",
       "      <td>CG0002-0688</td>\n",
       "      <td>CG0002-0688</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000891-1_2pA</td>\n",
       "      <td>HMNWGCCXX-4</td>\n",
       "      <td>160404_ST-E00270_0172_AHMNWGCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNWGCCXX-4.hgv.bam</td>\n",
       "      <td>63887749384</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNWGCCXX-4.hgv.bam</td>\n",
       "      <td>eb166bc368340b9a3993be91894efe6a</td>\n",
       "      <td>0b96ae40-1eac-4e4a-b5c0-b13d1521e4da</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNWGCCXX-5.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000881-1_2pA</td>\n",
       "      <td>ST-00000881</td>\n",
       "      <td>CG0001-7000</td>\n",
       "      <td>CG0001-7000</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000881-1_2pA</td>\n",
       "      <td>HMNWGCCXX-5</td>\n",
       "      <td>160404_ST-E00270_0172_AHMNWGCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNWGCCXX-5.hgv.bam</td>\n",
       "      <td>62551790031</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNWGCCXX-5.hgv.bam</td>\n",
       "      <td>22e7e1579815d43cb795a72bd513fd5e</td>\n",
       "      <td>43a6dd91-6c1b-4a2d-a1c4-edfd661978f3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNWGCCXX-6.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000882-1_2pA</td>\n",
       "      <td>ST-00000882</td>\n",
       "      <td>CG0001-7009</td>\n",
       "      <td>CG0001-7009</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000882-1_2pA</td>\n",
       "      <td>HMNWGCCXX-6</td>\n",
       "      <td>160404_ST-E00270_0172_AHMNWGCCXX_6</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNWGCCXX-6.hgv.bam</td>\n",
       "      <td>63419244156</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNWGCCXX-6.hgv.bam</td>\n",
       "      <td>d1d1075323c24986b9c0f61459bddbfa</td>\n",
       "      <td>ae170ddb-792c-4c1c-84d4-3e43d65c0400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNWGCCXX-7.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00002913-1_2pA</td>\n",
       "      <td>ST-00002913</td>\n",
       "      <td>CG0001-7755</td>\n",
       "      <td>CG0001-7755</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00002913-1_2pA</td>\n",
       "      <td>HMNWGCCXX-7</td>\n",
       "      <td>160404_ST-E00270_0172_AHMNWGCCXX_7</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNWGCCXX-7.hgv.bam</td>\n",
       "      <td>63390037001</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNWGCCXX-7.hgv.bam</td>\n",
       "      <td>f25b67087deb3475e113a941df86c37d</td>\n",
       "      <td>30402b37-d1eb-40fb-bdd1-b217644ec722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNWGCCXX-8.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000896-1_2pA</td>\n",
       "      <td>ST-00000896</td>\n",
       "      <td>CG0001-7101</td>\n",
       "      <td>CG0001-7101</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000896-1_2pA</td>\n",
       "      <td>HMNWGCCXX-8</td>\n",
       "      <td>160404_ST-E00270_0172_AHMNWGCCXX_8</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNWGCCXX-8.hgv.bam</td>\n",
       "      <td>65003434003</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNWGCCXX-8.hgv.bam</td>\n",
       "      <td>22f7b98ccef9ed1c3f0f44de46d84f33</td>\n",
       "      <td>ca0a0b90-b9e6-437a-809a-2a7a5d3395ff</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNYTCCXX-1.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00035489-1_2pA</td>\n",
       "      <td>ST-00035489</td>\n",
       "      <td>CG0024-1638</td>\n",
       "      <td>CG0024-1638</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00035489-1_2pA</td>\n",
       "      <td>HMNYTCCXX-1</td>\n",
       "      <td>160404_E00366_0102_AHMNYTCCXX_1</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNYTCCXX-1.hgv.bam</td>\n",
       "      <td>68731807689</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNYTCCXX-1.hgv.bam</td>\n",
       "      <td>b574a8d1eb18eaf65ab067aa4d6716db</td>\n",
       "      <td>18faf620-7f2a-4ade-af93-4f44a8cf8eea</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNYTCCXX-2.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000897-1_2pA</td>\n",
       "      <td>ST-00000897</td>\n",
       "      <td>CG0001-7091</td>\n",
       "      <td>CG0001-7091</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000897-1_2pA</td>\n",
       "      <td>HMNYTCCXX-2</td>\n",
       "      <td>160404_E00366_0102_AHMNYTCCXX_2</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNYTCCXX-2.hgv.bam</td>\n",
       "      <td>71209716845</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNYTCCXX-2.hgv.bam</td>\n",
       "      <td>c36e9cdf57d1b9f3d149bf237e7c230a</td>\n",
       "      <td>2b2ba140-c15b-4680-a8df-335b4442c4a7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNYTCCXX-3.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000915-1_2pA</td>\n",
       "      <td>ST-00000915</td>\n",
       "      <td>CG0002-2667</td>\n",
       "      <td>CG0002-2667</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000915-1_2pA</td>\n",
       "      <td>HMNYTCCXX-3</td>\n",
       "      <td>160404_E00366_0102_AHMNYTCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNYTCCXX-3.hgv.bam</td>\n",
       "      <td>72340921082</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNYTCCXX-3.hgv.bam</td>\n",
       "      <td>20d8715c78be4035171aa01517128e45</td>\n",
       "      <td>250fea48-57e5-430a-8131-7935c6e51ac4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNYTCCXX-4.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000916-1_2pA</td>\n",
       "      <td>ST-00000916</td>\n",
       "      <td>CG0002-2679</td>\n",
       "      <td>CG0002-2679</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000916-1_2pA</td>\n",
       "      <td>HMNYTCCXX-4</td>\n",
       "      <td>160404_E00366_0102_AHMNYTCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNYTCCXX-4.hgv.bam</td>\n",
       "      <td>73775241806</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNYTCCXX-4.hgv.bam</td>\n",
       "      <td>f93bdc0b155364e608af43d432912622</td>\n",
       "      <td>724f015b-82c5-4349-96b2-ef354e04b918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNYTCCXX-5.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00006135-1_2pA</td>\n",
       "      <td>ST-00006135</td>\n",
       "      <td>CG0005-0087</td>\n",
       "      <td>CG0005-0087</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00006135-1_2pA</td>\n",
       "      <td>HMNYTCCXX-5</td>\n",
       "      <td>160404_E00366_0102_AHMNYTCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNYTCCXX-5.hgv.bam</td>\n",
       "      <td>69182413410</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNYTCCXX-5.hgv.bam</td>\n",
       "      <td>938aa2542c0757060fe5dd559233cfce</td>\n",
       "      <td>014ffc3c-27e8-4357-b23b-1885c0b286f3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNYTCCXX-6.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001160-1_2pA</td>\n",
       "      <td>ST-00001160</td>\n",
       "      <td>CG0001-6036</td>\n",
       "      <td>CG0001-6036</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001160-1_2pA</td>\n",
       "      <td>HMNYTCCXX-6</td>\n",
       "      <td>160404_E00366_0102_AHMNYTCCXX_6</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNYTCCXX-6.hgv.bam</td>\n",
       "      <td>71554102371</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNYTCCXX-6.hgv.bam</td>\n",
       "      <td>2ca3a0716b017d37993734fe8269fd8b</td>\n",
       "      <td>77087e9c-b6c1-44bb-a628-c1558592c090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNYTCCXX-7.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001161-1_2pA</td>\n",
       "      <td>ST-00001161</td>\n",
       "      <td>CG0001-6043</td>\n",
       "      <td>CG0001-6043</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001161-1_2pA</td>\n",
       "      <td>HMNYTCCXX-7</td>\n",
       "      <td>160404_E00366_0102_AHMNYTCCXX_7</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNYTCCXX-7.hgv.bam</td>\n",
       "      <td>71182790908</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNYTCCXX-7.hgv.bam</td>\n",
       "      <td>d99b0805a24e77788ac711621b28cb2c</td>\n",
       "      <td>51516d9c-ddf5-42a3-99cf-6766ffc989f1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HMNYTCCXX-8.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00003609-1_2pA</td>\n",
       "      <td>ST-00003609</td>\n",
       "      <td>CG0002-9521</td>\n",
       "      <td>CG0002-9521</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00003609-1_2pA</td>\n",
       "      <td>HMNYTCCXX-8</td>\n",
       "      <td>160404_E00366_0102_AHMNYTCCXX_8</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HMNYTCCXX-8.hgv.bam</td>\n",
       "      <td>73017218859</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HMNYTCCXX-8.hgv.bam</td>\n",
       "      <td>cee527f56a1d87faeaf0ef456bb23327</td>\n",
       "      <td>35e7b6c7-68e4-4b4d-9f25-402675a5f4eb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>870</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLGYCCXX-1.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000847-1_2pA</td>\n",
       "      <td>ST-00000847</td>\n",
       "      <td>CG0001-6447</td>\n",
       "      <td>CG0001-6447</td>\n",
       "      <td>GMKFCS.ST-00000847-1_2AMP</td>\n",
       "      <td>HNLGYCCXX-1</td>\n",
       "      <td>160503_E00385_0138_BHNLGYCCXX_1</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLGYCCXX-1.hgv.bam</td>\n",
       "      <td>63045174117</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLGYCCXX-1.hgv.bam</td>\n",
       "      <td>4e09c16d079b79add9c01afa97d2efe4</td>\n",
       "      <td>13160506-4ad6-4310-b7c1-839327584193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>871</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLGYCCXX-2.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000848-1_3pA</td>\n",
       "      <td>ST-00000848</td>\n",
       "      <td>CG0001-6438</td>\n",
       "      <td>CG0001-6438</td>\n",
       "      <td>GMKFCS.ST-00000848-1_3AMP</td>\n",
       "      <td>HNLGYCCXX-2</td>\n",
       "      <td>160503_E00385_0138_BHNLGYCCXX_2</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLGYCCXX-2.hgv.bam</td>\n",
       "      <td>62451333230</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLGYCCXX-2.hgv.bam</td>\n",
       "      <td>450e1fe98480416e36ac7be03d4c2c54</td>\n",
       "      <td>be75c603-a0c0-4a28-8c9a-00c2a59380ac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>872</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLGYCCXX-3.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000849-1_2pA</td>\n",
       "      <td>ST-00000849</td>\n",
       "      <td>CG0001-6431</td>\n",
       "      <td>CG0001-6431</td>\n",
       "      <td>GMKFCS.ST-00000849-1_2AMP</td>\n",
       "      <td>HNLGYCCXX-3</td>\n",
       "      <td>160503_E00385_0138_BHNLGYCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLGYCCXX-3.hgv.bam</td>\n",
       "      <td>66394383540</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLGYCCXX-3.hgv.bam</td>\n",
       "      <td>608cc692b1d487a111e3bee9fc0b39d4</td>\n",
       "      <td>77c02f21-42c0-4b38-b03a-80650f1559bb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>873</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLGYCCXX-4.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001397-1_2pA</td>\n",
       "      <td>ST-00001397</td>\n",
       "      <td>CG0000-8011</td>\n",
       "      <td>CG0000-8011</td>\n",
       "      <td>GMKFCS.ST-00001397-1_2AMP</td>\n",
       "      <td>HNLGYCCXX-4</td>\n",
       "      <td>160503_E00385_0138_BHNLGYCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLGYCCXX-4.hgv.bam</td>\n",
       "      <td>65400960140</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLGYCCXX-4.hgv.bam</td>\n",
       "      <td>7301fb467e9ed731fcea25a600794049</td>\n",
       "      <td>37564660-0c05-494d-8016-a7cbd1ae6397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLGYCCXX-5.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001398-1_3pA</td>\n",
       "      <td>ST-00001398</td>\n",
       "      <td>CG0000-8018</td>\n",
       "      <td>CG0000-8018</td>\n",
       "      <td>GMKFCS.ST-00001398-1_3AMP</td>\n",
       "      <td>HNLGYCCXX-5</td>\n",
       "      <td>160503_E00385_0138_BHNLGYCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLGYCCXX-5.hgv.bam</td>\n",
       "      <td>64656711688</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLGYCCXX-5.hgv.bam</td>\n",
       "      <td>5357d4c950971594895faea2ccfe22aa</td>\n",
       "      <td>9a3f376f-2aa3-4f04-9dd4-472516f766b9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>875</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLGYCCXX-6.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001399-1_2pA</td>\n",
       "      <td>ST-00001399</td>\n",
       "      <td>CG0000-8025</td>\n",
       "      <td>CG0000-8025</td>\n",
       "      <td>GMKFCS.ST-00001399-1_2AMP</td>\n",
       "      <td>HNLGYCCXX-6</td>\n",
       "      <td>160503_E00385_0138_BHNLGYCCXX_6</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLGYCCXX-6.hgv.bam</td>\n",
       "      <td>65028281477</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLGYCCXX-6.hgv.bam</td>\n",
       "      <td>a542c2bc1ff3f6b2481d62179038033c</td>\n",
       "      <td>7f6a770f-44ce-4437-83dd-ce95ee211375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>876</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLLGCCXX-1.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000097-1_2pA</td>\n",
       "      <td>ST-00000097</td>\n",
       "      <td>CG0001-6466</td>\n",
       "      <td>CG0001-6466</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000097-1_2pA</td>\n",
       "      <td>HNLLGCCXX-1</td>\n",
       "      <td>160503_E00380_0118_BHNLLGCCXX_1</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLLGCCXX-1.hgv.bam</td>\n",
       "      <td>67558188737</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLLGCCXX-1.hgv.bam</td>\n",
       "      <td>1b0ceb7104efef3815a56239b9946bbe</td>\n",
       "      <td>d7395c41-eeac-4901-b034-76fb6ca334e0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>877</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLLGCCXX-2.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001517-1_3pA</td>\n",
       "      <td>ST-00001517</td>\n",
       "      <td>CG0001-7987</td>\n",
       "      <td>CG0001-7987</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001517-1_3pA</td>\n",
       "      <td>HNLLGCCXX-2</td>\n",
       "      <td>160503_E00380_0118_BHNLLGCCXX_2</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLLGCCXX-2.hgv.bam</td>\n",
       "      <td>64770159106</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLLGCCXX-2.hgv.bam</td>\n",
       "      <td>4db9c33019180df0ea5a7640393a98b2</td>\n",
       "      <td>10843f7e-3b91-471c-a993-ee61e75d9c37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>878</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLLGCCXX-3.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001518-1_2pA</td>\n",
       "      <td>ST-00001518</td>\n",
       "      <td>CG0001-7978</td>\n",
       "      <td>CG0001-7978</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001518-1_2pA</td>\n",
       "      <td>HNLLGCCXX-3</td>\n",
       "      <td>160503_E00380_0118_BHNLLGCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLLGCCXX-3.hgv.bam</td>\n",
       "      <td>70198446428</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLLGCCXX-3.hgv.bam</td>\n",
       "      <td>170fe0b6ff7fcff6e54332e4cf34ae6f</td>\n",
       "      <td>7c161065-01ce-449f-a619-229f33cb485d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLLGCCXX-4.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000315-1_2pA</td>\n",
       "      <td>ST-00000315</td>\n",
       "      <td>CG0001-5411</td>\n",
       "      <td>CG0001-5411</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000315-1_2pA</td>\n",
       "      <td>HNLLGCCXX-4</td>\n",
       "      <td>160503_E00380_0118_BHNLLGCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLLGCCXX-4.hgv.bam</td>\n",
       "      <td>68526123888</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLLGCCXX-4.hgv.bam</td>\n",
       "      <td>64b78adc81a019448e710d3b663d9ecf</td>\n",
       "      <td>28a71fcd-7b64-4cd4-bb11-b5e93a792efe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>880</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLLGCCXX-5.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000319-1_3pA</td>\n",
       "      <td>ST-00000319</td>\n",
       "      <td>CG0001-5227</td>\n",
       "      <td>CG0001-5227</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00000319-1_3pA</td>\n",
       "      <td>HNLLGCCXX-5</td>\n",
       "      <td>160503_E00380_0118_BHNLLGCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLLGCCXX-5.hgv.bam</td>\n",
       "      <td>70615280661</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLLGCCXX-5.hgv.bam</td>\n",
       "      <td>2dd28b16ebf9e80fcd02ce8b95d877e0</td>\n",
       "      <td>9436e470-aed2-4346-af1f-37c4dc3f1700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>881</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLLGCCXX-6.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00004955-1_2pA</td>\n",
       "      <td>ST-00004955</td>\n",
       "      <td>CG0004-2218</td>\n",
       "      <td>CG0004-2218</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00004955-1_2pA</td>\n",
       "      <td>HNLLGCCXX-6</td>\n",
       "      <td>160503_E00380_0118_BHNLLGCCXX_6</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLLGCCXX-6.hgv.bam</td>\n",
       "      <td>68833299936</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLLGCCXX-6.hgv.bam</td>\n",
       "      <td>023501d1533a29bf8cf8ab4b08a182d2</td>\n",
       "      <td>d3d2eb9e-86eb-4187-9948-2d14f9893d6f</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>882</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLTMCCXX-3.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00004736-1_3pA</td>\n",
       "      <td>ST-00004736</td>\n",
       "      <td>CG0004-5192</td>\n",
       "      <td>CG0004-5192</td>\n",
       "      <td>GMKFCS.ST-00004736-1_3AMP</td>\n",
       "      <td>HNLTMCCXX-3</td>\n",
       "      <td>160505_E00329_0217_BHNLTMCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLTMCCXX-3.hgv.bam</td>\n",
       "      <td>66561442449</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLTMCCXX-3.hgv.bam</td>\n",
       "      <td>22cdd4595cd1831f5bb20f82b956e4dd</td>\n",
       "      <td>a46a84b2-951c-4838-95c9-ddc78074f3f3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>883</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLTMCCXX-4.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00004743-1_2pA</td>\n",
       "      <td>ST-00004743</td>\n",
       "      <td>CG0004-5231</td>\n",
       "      <td>CG0004-5231</td>\n",
       "      <td>GMKFCS.ST-00004743-1_2AMP</td>\n",
       "      <td>HNLTMCCXX-4</td>\n",
       "      <td>160505_E00329_0217_BHNLTMCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLTMCCXX-4.hgv.bam</td>\n",
       "      <td>67423179381</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLTMCCXX-4.hgv.bam</td>\n",
       "      <td>a94e17951b8e6f198a3d6710e16e8dae</td>\n",
       "      <td>d197c7d7-1899-45bc-9ee7-f17fde386526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>884</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLTMCCXX-5.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00004745-1_2pA</td>\n",
       "      <td>ST-00004745</td>\n",
       "      <td>CG0004-5250</td>\n",
       "      <td>CG0004-5250</td>\n",
       "      <td>GMKFCS.ST-00004745-1_2AMP</td>\n",
       "      <td>HNLTMCCXX-5</td>\n",
       "      <td>160505_E00329_0217_BHNLTMCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLTMCCXX-5.hgv.bam</td>\n",
       "      <td>66903856843</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLTMCCXX-5.hgv.bam</td>\n",
       "      <td>a05423c9c966bfd2a17cfafc8961f5e3</td>\n",
       "      <td>85e6b54e-ca93-4990-bdcd-284744267f8e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>885</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLTMCCXX-6.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00005127-1_3pA</td>\n",
       "      <td>ST-00005127</td>\n",
       "      <td>CG0004-9944</td>\n",
       "      <td>CG0004-9944</td>\n",
       "      <td>GMKFCS.ST-00005127-1_3AMP</td>\n",
       "      <td>HNLTMCCXX-6</td>\n",
       "      <td>160505_E00329_0217_BHNLTMCCXX_6</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLTMCCXX-6.hgv.bam</td>\n",
       "      <td>68643834876</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLTMCCXX-6.hgv.bam</td>\n",
       "      <td>01bb3e5d2a49761400afbcf7ced243fc</td>\n",
       "      <td>dc697d71-0a64-4d82-a671-7c0b114444b0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>886</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLTMCCXX-7.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00005128-1_2pA</td>\n",
       "      <td>ST-00005128</td>\n",
       "      <td>CG0004-9953</td>\n",
       "      <td>CG0004-9953</td>\n",
       "      <td>GMKFCS.ST-00005128-1_2AMP</td>\n",
       "      <td>HNLTMCCXX-7</td>\n",
       "      <td>160505_E00329_0217_BHNLTMCCXX_7</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLTMCCXX-7.hgv.bam</td>\n",
       "      <td>62853583002</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLTMCCXX-7.hgv.bam</td>\n",
       "      <td>fefa5fe465cf5deb7f0d32553c231cc0</td>\n",
       "      <td>dcde4f71-29a6-489a-8785-370b9d6386d0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>887</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLTMCCXX-8.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00005129-1_2pA</td>\n",
       "      <td>ST-00005129</td>\n",
       "      <td>CG0023-9179</td>\n",
       "      <td>CG0023-9179</td>\n",
       "      <td>GMKFCS.ST-00005129-1_2AMP</td>\n",
       "      <td>HNLTMCCXX-8</td>\n",
       "      <td>160505_E00329_0217_BHNLTMCCXX_8</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLTMCCXX-8.hgv.bam</td>\n",
       "      <td>63770026098</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLTMCCXX-8.hgv.bam</td>\n",
       "      <td>1afca9e0c32a5267a593b6b84562e9be</td>\n",
       "      <td>0adb51dc-d15a-4898-a6f4-66c25ae37028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>888</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLWWCCXX-3.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001875-1_3pA</td>\n",
       "      <td>ST-00001875</td>\n",
       "      <td>CG0000-4060</td>\n",
       "      <td>CG0000-4060</td>\n",
       "      <td>GMKFCS.ST-00001875-1_3AMP</td>\n",
       "      <td>HNLWWCCXX-3</td>\n",
       "      <td>160504_ST-E00238_0221_BHNLWWCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLWWCCXX-3.hgv.bam</td>\n",
       "      <td>65574536013</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLWWCCXX-3.hgv.bam</td>\n",
       "      <td>fbf5990c65fb570eaff851f754078326</td>\n",
       "      <td>42d783cf-a91f-40ea-a6da-462608d3dd15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>889</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLWWCCXX-4.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001876-1_3pA</td>\n",
       "      <td>ST-00001876</td>\n",
       "      <td>CG0023-9135</td>\n",
       "      <td>CG0023-9135</td>\n",
       "      <td>GMKFCS.ST-00001876-1_3AMP</td>\n",
       "      <td>HNLWWCCXX-4</td>\n",
       "      <td>160504_ST-E00238_0221_BHNLWWCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLWWCCXX-4.hgv.bam</td>\n",
       "      <td>63403425870</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLWWCCXX-4.hgv.bam</td>\n",
       "      <td>286e7eeb97b24c6c77609af176a8ad80</td>\n",
       "      <td>6889ba10-d670-4bd4-b25f-18ef9be454c8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLWWCCXX-5.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00001877-1_2pA</td>\n",
       "      <td>ST-00001877</td>\n",
       "      <td>CG0023-9151</td>\n",
       "      <td>CG0023-9151</td>\n",
       "      <td>GMKFCS.ST-00001877-1_2AMP</td>\n",
       "      <td>HNLWWCCXX-5</td>\n",
       "      <td>160504_ST-E00238_0221_BHNLWWCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLWWCCXX-5.hgv.bam</td>\n",
       "      <td>66024331506</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLWWCCXX-5.hgv.bam</td>\n",
       "      <td>dfa2c3c212aa85dfc7aa48a091e437e3</td>\n",
       "      <td>1b50fde7-6b44-4c7d-b8d4-88e65627d4bc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>891</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLWWCCXX-6.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00002220-1_3pA</td>\n",
       "      <td>ST-00002220</td>\n",
       "      <td>CG0023-9136</td>\n",
       "      <td>CG0023-9136</td>\n",
       "      <td>GMKFCS.ST-00002220-1_3AMP</td>\n",
       "      <td>HNLWWCCXX-6</td>\n",
       "      <td>160504_ST-E00238_0221_BHNLWWCCXX_6</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLWWCCXX-6.hgv.bam</td>\n",
       "      <td>66062654349</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLWWCCXX-6.hgv.bam</td>\n",
       "      <td>2a365584395a733f32539a98d035f7bb</td>\n",
       "      <td>3c207f39-78c0-4cd0-978e-3c94d8056e18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLWWCCXX-7.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00002221-1_2pA</td>\n",
       "      <td>ST-00002221</td>\n",
       "      <td>CG0023-9142</td>\n",
       "      <td>CG0023-9142</td>\n",
       "      <td>GMKFCS.ST-00002221-1_2AMP</td>\n",
       "      <td>HNLWWCCXX-7</td>\n",
       "      <td>160504_ST-E00238_0221_BHNLWWCCXX_7</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLWWCCXX-7.hgv.bam</td>\n",
       "      <td>64825822393</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLWWCCXX-7.hgv.bam</td>\n",
       "      <td>fb1ef6c7bb954c7a6b8ce256b911c0bb</td>\n",
       "      <td>61084a2e-d0ad-4a2d-ac3a-127925eeef78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>893</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLWWCCXX-8.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00002222-1_2pA</td>\n",
       "      <td>ST-00002222</td>\n",
       "      <td>CG0023-9143</td>\n",
       "      <td>CG0023-9143</td>\n",
       "      <td>GMKFCS.ST-00002222-1_2AMP</td>\n",
       "      <td>HNLWWCCXX-8</td>\n",
       "      <td>160504_ST-E00238_0221_BHNLWWCCXX_8</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLWWCCXX-8.hgv.bam</td>\n",
       "      <td>65530524355</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLWWCCXX-8.hgv.bam</td>\n",
       "      <td>b1c11f1f2a1e1979bd8d67ddc7eaf33d</td>\n",
       "      <td>366464c9-290b-40de-91c4-e9c8a5b2fb92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>894</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLYFCCXX-1.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00006214-1_3pA</td>\n",
       "      <td>ST-00006214</td>\n",
       "      <td>CG0005-0760</td>\n",
       "      <td>CG0005-0760</td>\n",
       "      <td>GMKFCS.ST-00006214-1_3AMP</td>\n",
       "      <td>HNLYFCCXX-1</td>\n",
       "      <td>160507_ST-E00186_0258_AHNLYFCCXX_1</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLYFCCXX-1.hgv.bam</td>\n",
       "      <td>65960227146</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLYFCCXX-1.hgv.bam</td>\n",
       "      <td>798b6b4f949629f370c681d701e9dfc2</td>\n",
       "      <td>fe4aed3b-2429-4ceb-b64f-90360c71fc2d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>895</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLYFCCXX-2.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00006215-1_2pA</td>\n",
       "      <td>ST-00006215</td>\n",
       "      <td>CG0005-0769</td>\n",
       "      <td>CG0005-0769</td>\n",
       "      <td>GMKFCS.ST-00006215-1_2AMP</td>\n",
       "      <td>HNLYFCCXX-2</td>\n",
       "      <td>160507_ST-E00186_0258_AHNLYFCCXX_2</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLYFCCXX-2.hgv.bam</td>\n",
       "      <td>67502614833</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLYFCCXX-2.hgv.bam</td>\n",
       "      <td>6ff605480071139a2c90e3f06893601b</td>\n",
       "      <td>44a4ddd8-2a0d-46dd-91d5-a262cef630c5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>896</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLYFCCXX-3.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00006216-1_2pA</td>\n",
       "      <td>ST-00006216</td>\n",
       "      <td>CG0005-0777</td>\n",
       "      <td>CG0005-0777</td>\n",
       "      <td>GMKFCS.ST-00006216-1_2AMP</td>\n",
       "      <td>HNLYFCCXX-3</td>\n",
       "      <td>160507_ST-E00186_0258_AHNLYFCCXX_3</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLYFCCXX-3.hgv.bam</td>\n",
       "      <td>65801483473</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLYFCCXX-3.hgv.bam</td>\n",
       "      <td>76636b8ac9fa9e4fe5fad431a88b1dc8</td>\n",
       "      <td>2faf8164-925a-4d83-a4e3-b0334b2fedef</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>897</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLYFCCXX-4.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00006246-1_2pA</td>\n",
       "      <td>ST-00006246</td>\n",
       "      <td>CG0005-2446</td>\n",
       "      <td>CG0005-2446</td>\n",
       "      <td>GMKFCS.ST-00006246-1_2AMP</td>\n",
       "      <td>HNLYFCCXX-4</td>\n",
       "      <td>160507_ST-E00186_0258_AHNLYFCCXX_4</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLYFCCXX-4.hgv.bam</td>\n",
       "      <td>66789458840</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLYFCCXX-4.hgv.bam</td>\n",
       "      <td>6533372f0751391ba22cbb846e779d7f</td>\n",
       "      <td>60811aff-e630-490a-9d29-f2dd376f7b3b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>898</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLYFCCXX-5.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00006249-1_2pA</td>\n",
       "      <td>ST-00006249</td>\n",
       "      <td>CG0005-2464</td>\n",
       "      <td>CG0005-2464</td>\n",
       "      <td>GMKFCS.ST-00006249-1_2AMP</td>\n",
       "      <td>HNLYFCCXX-5</td>\n",
       "      <td>160507_ST-E00186_0258_AHNLYFCCXX_5</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLYFCCXX-5.hgv.bam</td>\n",
       "      <td>65925371957</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLYFCCXX-5.hgv.bam</td>\n",
       "      <td>16f2f3f1fd99e3ff2942fe2ba5a0f653</td>\n",
       "      <td>c028a1e1-3fe3-481e-b98b-b60e67f83c6b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>899</th>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>HNLYFCCXX-6.hgv.bam</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00006250-1_3pA</td>\n",
       "      <td>ST-00006250</td>\n",
       "      <td>CG0005-2473</td>\n",
       "      <td>CG0005-2473</td>\n",
       "      <td>GMKFCS.ST-00006250-1_3AMP</td>\n",
       "      <td>HNLYFCCXX-6</td>\n",
       "      <td>160507_ST-E00186_0258_AHNLYFCCXX_6</td>\n",
       "      <td>150</td>\n",
       "      <td>...</td>\n",
       "      <td>392.684444</td>\n",
       "      <td>150.0</td>\n",
       "      <td>900</td>\n",
       "      <td>submitted aligned reads</td>\n",
       "      <td>hgv.bam</td>\n",
       "      <td>HNLYFCCXX-6.hgv.bam</td>\n",
       "      <td>65541019973</td>\n",
       "      <td>s3://kf-seq-data-bcm/seidman/HNLYFCCXX-6.hgv.bam</td>\n",
       "      <td>760439a8531adf583547be4c667c0b84</td>\n",
       "      <td>10352b13-62d1-4a09-853e-cbfcdbf96ab5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>900 rows  28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Cohort        BAM sample ID                           Library  \\\n",
       "0    GMKF-Seidman  HMNMFCCXX-1.hgv.bam  IWG_IND-GMKFCS.ST-00000936-1_2pA   \n",
       "1    GMKF-Seidman  HMNMFCCXX-2.hgv.bam  IWG_IND-GMKFCS.ST-00000953-1_2pA   \n",
       "2    GMKF-Seidman  HMNMFCCXX-3.hgv.bam  IWG_IND-GMKFCS.ST-00000954-1_2pA   \n",
       "3    GMKF-Seidman  HMNMFCCXX-4.hgv.bam  IWG_IND-GMKFCS.ST-00003538-1_2pA   \n",
       "4    GMKF-Seidman  HMNMFCCXX-5.hgv.bam  IWG_IND-GMKFCS.ST-00003539-1_2pA   \n",
       "5    GMKF-Seidman  HMNMFCCXX-6.hgv.bam  IWG_IND-GMKFCS.ST-00003561-1_2pA   \n",
       "6    GMKF-Seidman  HMNVCCCXX-1.hgv.bam  IWG_IND-GMKFCS.ST-00000768-1_2pA   \n",
       "7    GMKF-Seidman  HMNVCCCXX-2.hgv.bam  IWG_IND-GMKFCS.ST-00000580-1_2pA   \n",
       "8    GMKF-Seidman  HMNVCCCXX-3.hgv.bam  IWG_IND-GMKFCS.ST-00000589-1_2pA   \n",
       "9    GMKF-Seidman  HMNVCCCXX-4.hgv.bam  IWG_IND-GMKFCS.ST-00000991-1_2pA   \n",
       "10   GMKF-Seidman  HMNVCCCXX-5.hgv.bam  IWG_IND-GMKFCS.ST-00000992-1_2pA   \n",
       "11   GMKF-Seidman  HMNVCCCXX-6.hgv.bam  IWG_IND-GMKFCS.ST-00000603-1_2pA   \n",
       "12   GMKF-Seidman  HMNVCCCXX-7.hgv.bam  IWG_IND-GMKFCS.ST-00000604-1_2pA   \n",
       "13   GMKF-Seidman  HMNVCCCXX-8.hgv.bam  IWG_IND-GMKFCS.ST-00000611-1_2pA   \n",
       "14   GMKF-Seidman  HMNWGCCXX-1.hgv.bam  IWG_IND-GMKFCS.ST-00000818-1_2pA   \n",
       "15   GMKF-Seidman  HMNWGCCXX-2.hgv.bam  IWG_IND-GMKFCS.ST-00000872-1_2pA   \n",
       "16   GMKF-Seidman  HMNWGCCXX-3.hgv.bam  IWG_IND-GMKFCS.ST-00000890-1_2pA   \n",
       "17   GMKF-Seidman  HMNWGCCXX-4.hgv.bam  IWG_IND-GMKFCS.ST-00000891-1_2pA   \n",
       "18   GMKF-Seidman  HMNWGCCXX-5.hgv.bam  IWG_IND-GMKFCS.ST-00000881-1_2pA   \n",
       "19   GMKF-Seidman  HMNWGCCXX-6.hgv.bam  IWG_IND-GMKFCS.ST-00000882-1_2pA   \n",
       "20   GMKF-Seidman  HMNWGCCXX-7.hgv.bam  IWG_IND-GMKFCS.ST-00002913-1_2pA   \n",
       "21   GMKF-Seidman  HMNWGCCXX-8.hgv.bam  IWG_IND-GMKFCS.ST-00000896-1_2pA   \n",
       "22   GMKF-Seidman  HMNYTCCXX-1.hgv.bam  IWG_IND-GMKFCS.ST-00035489-1_2pA   \n",
       "23   GMKF-Seidman  HMNYTCCXX-2.hgv.bam  IWG_IND-GMKFCS.ST-00000897-1_2pA   \n",
       "24   GMKF-Seidman  HMNYTCCXX-3.hgv.bam  IWG_IND-GMKFCS.ST-00000915-1_2pA   \n",
       "25   GMKF-Seidman  HMNYTCCXX-4.hgv.bam  IWG_IND-GMKFCS.ST-00000916-1_2pA   \n",
       "26   GMKF-Seidman  HMNYTCCXX-5.hgv.bam  IWG_IND-GMKFCS.ST-00006135-1_2pA   \n",
       "27   GMKF-Seidman  HMNYTCCXX-6.hgv.bam  IWG_IND-GMKFCS.ST-00001160-1_2pA   \n",
       "28   GMKF-Seidman  HMNYTCCXX-7.hgv.bam  IWG_IND-GMKFCS.ST-00001161-1_2pA   \n",
       "29   GMKF-Seidman  HMNYTCCXX-8.hgv.bam  IWG_IND-GMKFCS.ST-00003609-1_2pA   \n",
       "..            ...                  ...                               ...   \n",
       "870  GMKF-Seidman  HNLGYCCXX-1.hgv.bam  IWG_IND-GMKFCS.ST-00000847-1_2pA   \n",
       "871  GMKF-Seidman  HNLGYCCXX-2.hgv.bam  IWG_IND-GMKFCS.ST-00000848-1_3pA   \n",
       "872  GMKF-Seidman  HNLGYCCXX-3.hgv.bam  IWG_IND-GMKFCS.ST-00000849-1_2pA   \n",
       "873  GMKF-Seidman  HNLGYCCXX-4.hgv.bam  IWG_IND-GMKFCS.ST-00001397-1_2pA   \n",
       "874  GMKF-Seidman  HNLGYCCXX-5.hgv.bam  IWG_IND-GMKFCS.ST-00001398-1_3pA   \n",
       "875  GMKF-Seidman  HNLGYCCXX-6.hgv.bam  IWG_IND-GMKFCS.ST-00001399-1_2pA   \n",
       "876  GMKF-Seidman  HNLLGCCXX-1.hgv.bam  IWG_IND-GMKFCS.ST-00000097-1_2pA   \n",
       "877  GMKF-Seidman  HNLLGCCXX-2.hgv.bam  IWG_IND-GMKFCS.ST-00001517-1_3pA   \n",
       "878  GMKF-Seidman  HNLLGCCXX-3.hgv.bam  IWG_IND-GMKFCS.ST-00001518-1_2pA   \n",
       "879  GMKF-Seidman  HNLLGCCXX-4.hgv.bam  IWG_IND-GMKFCS.ST-00000315-1_2pA   \n",
       "880  GMKF-Seidman  HNLLGCCXX-5.hgv.bam  IWG_IND-GMKFCS.ST-00000319-1_3pA   \n",
       "881  GMKF-Seidman  HNLLGCCXX-6.hgv.bam  IWG_IND-GMKFCS.ST-00004955-1_2pA   \n",
       "882  GMKF-Seidman  HNLTMCCXX-3.hgv.bam  IWG_IND-GMKFCS.ST-00004736-1_3pA   \n",
       "883  GMKF-Seidman  HNLTMCCXX-4.hgv.bam  IWG_IND-GMKFCS.ST-00004743-1_2pA   \n",
       "884  GMKF-Seidman  HNLTMCCXX-5.hgv.bam  IWG_IND-GMKFCS.ST-00004745-1_2pA   \n",
       "885  GMKF-Seidman  HNLTMCCXX-6.hgv.bam  IWG_IND-GMKFCS.ST-00005127-1_3pA   \n",
       "886  GMKF-Seidman  HNLTMCCXX-7.hgv.bam  IWG_IND-GMKFCS.ST-00005128-1_2pA   \n",
       "887  GMKF-Seidman  HNLTMCCXX-8.hgv.bam  IWG_IND-GMKFCS.ST-00005129-1_2pA   \n",
       "888  GMKF-Seidman  HNLWWCCXX-3.hgv.bam  IWG_IND-GMKFCS.ST-00001875-1_3pA   \n",
       "889  GMKF-Seidman  HNLWWCCXX-4.hgv.bam  IWG_IND-GMKFCS.ST-00001876-1_3pA   \n",
       "890  GMKF-Seidman  HNLWWCCXX-5.hgv.bam  IWG_IND-GMKFCS.ST-00001877-1_2pA   \n",
       "891  GMKF-Seidman  HNLWWCCXX-6.hgv.bam  IWG_IND-GMKFCS.ST-00002220-1_3pA   \n",
       "892  GMKF-Seidman  HNLWWCCXX-7.hgv.bam  IWG_IND-GMKFCS.ST-00002221-1_2pA   \n",
       "893  GMKF-Seidman  HNLWWCCXX-8.hgv.bam  IWG_IND-GMKFCS.ST-00002222-1_2pA   \n",
       "894  GMKF-Seidman  HNLYFCCXX-1.hgv.bam  IWG_IND-GMKFCS.ST-00006214-1_3pA   \n",
       "895  GMKF-Seidman  HNLYFCCXX-2.hgv.bam  IWG_IND-GMKFCS.ST-00006215-1_2pA   \n",
       "896  GMKF-Seidman  HNLYFCCXX-3.hgv.bam  IWG_IND-GMKFCS.ST-00006216-1_2pA   \n",
       "897  GMKF-Seidman  HNLYFCCXX-4.hgv.bam  IWG_IND-GMKFCS.ST-00006246-1_2pA   \n",
       "898  GMKF-Seidman  HNLYFCCXX-5.hgv.bam  IWG_IND-GMKFCS.ST-00006249-1_2pA   \n",
       "899  GMKF-Seidman  HNLYFCCXX-6.hgv.bam  IWG_IND-GMKFCS.ST-00006250-1_3pA   \n",
       "\n",
       "    HGSC Sample ID dbgap_subject_id  sample_name  \\\n",
       "0      ST-00000936      CG0023-9153  CG0023-9153   \n",
       "1      ST-00000953      CG0002-6325  CG0002-6325   \n",
       "2      ST-00000954      CG0002-6931  CG0002-6931   \n",
       "3      ST-00003538      CG0002-7519  CG0002-7519   \n",
       "4      ST-00003539      CG0002-7976  CG0002-7976   \n",
       "5      ST-00003561      CG0002-7848  CG0002-7848   \n",
       "6      ST-00000768      CG0000-3840  CG0000-3840   \n",
       "7      ST-00000580      CG0000-2758  CG0000-2758   \n",
       "8      ST-00000589      CG0000-0344  CG0000-0344   \n",
       "9      ST-00000991      CG0000-4744  CG0000-4744   \n",
       "10     ST-00000992      CG0000-4752  CG0000-4752   \n",
       "11     ST-00000603      CG0000-6252  CG0000-6252   \n",
       "12     ST-00000604      CG0000-6259  CG0000-6259   \n",
       "13     ST-00000611      CG0000-6267  CG0000-6267   \n",
       "14     ST-00000818      CG0001-4817  CG0001-4817   \n",
       "15     ST-00000872      CG0002-0380  CG0002-0380   \n",
       "16     ST-00000890      CG0002-0677  CG0002-0677   \n",
       "17     ST-00000891      CG0002-0688  CG0002-0688   \n",
       "18     ST-00000881      CG0001-7000  CG0001-7000   \n",
       "19     ST-00000882      CG0001-7009  CG0001-7009   \n",
       "20     ST-00002913      CG0001-7755  CG0001-7755   \n",
       "21     ST-00000896      CG0001-7101  CG0001-7101   \n",
       "22     ST-00035489      CG0024-1638  CG0024-1638   \n",
       "23     ST-00000897      CG0001-7091  CG0001-7091   \n",
       "24     ST-00000915      CG0002-2667  CG0002-2667   \n",
       "25     ST-00000916      CG0002-2679  CG0002-2679   \n",
       "26     ST-00006135      CG0005-0087  CG0005-0087   \n",
       "27     ST-00001160      CG0001-6036  CG0001-6036   \n",
       "28     ST-00001161      CG0001-6043  CG0001-6043   \n",
       "29     ST-00003609      CG0002-9521  CG0002-9521   \n",
       "..             ...              ...          ...   \n",
       "870    ST-00000847      CG0001-6447  CG0001-6447   \n",
       "871    ST-00000848      CG0001-6438  CG0001-6438   \n",
       "872    ST-00000849      CG0001-6431  CG0001-6431   \n",
       "873    ST-00001397      CG0000-8011  CG0000-8011   \n",
       "874    ST-00001398      CG0000-8018  CG0000-8018   \n",
       "875    ST-00001399      CG0000-8025  CG0000-8025   \n",
       "876    ST-00000097      CG0001-6466  CG0001-6466   \n",
       "877    ST-00001517      CG0001-7987  CG0001-7987   \n",
       "878    ST-00001518      CG0001-7978  CG0001-7978   \n",
       "879    ST-00000315      CG0001-5411  CG0001-5411   \n",
       "880    ST-00000319      CG0001-5227  CG0001-5227   \n",
       "881    ST-00004955      CG0004-2218  CG0004-2218   \n",
       "882    ST-00004736      CG0004-5192  CG0004-5192   \n",
       "883    ST-00004743      CG0004-5231  CG0004-5231   \n",
       "884    ST-00004745      CG0004-5250  CG0004-5250   \n",
       "885    ST-00005127      CG0004-9944  CG0004-9944   \n",
       "886    ST-00005128      CG0004-9953  CG0004-9953   \n",
       "887    ST-00005129      CG0023-9179  CG0023-9179   \n",
       "888    ST-00001875      CG0000-4060  CG0000-4060   \n",
       "889    ST-00001876      CG0023-9135  CG0023-9135   \n",
       "890    ST-00001877      CG0023-9151  CG0023-9151   \n",
       "891    ST-00002220      CG0023-9136  CG0023-9136   \n",
       "892    ST-00002221      CG0023-9142  CG0023-9142   \n",
       "893    ST-00002222      CG0023-9143  CG0023-9143   \n",
       "894    ST-00006214      CG0005-0760  CG0005-0760   \n",
       "895    ST-00006215      CG0005-0769  CG0005-0769   \n",
       "896    ST-00006216      CG0005-0777  CG0005-0777   \n",
       "897    ST-00006246      CG0005-2446  CG0005-2446   \n",
       "898    ST-00006249      CG0005-2464  CG0005-2464   \n",
       "899    ST-00006250      CG0005-2473  CG0005-2473   \n",
       "\n",
       "                         library_name   rg_barcode  \\\n",
       "0           GMKFCS.ST-00000936-1_2AMP  HMNMFCCXX-1   \n",
       "1           GMKFCS.ST-00000953-1_2AMP  HMNMFCCXX-2   \n",
       "2           GMKFCS.ST-00000954-1_2AMP  HMNMFCCXX-3   \n",
       "3           GMKFCS.ST-00003538-1_2AMP  HMNMFCCXX-4   \n",
       "4           GMKFCS.ST-00003539-1_2AMP  HMNMFCCXX-5   \n",
       "5           GMKFCS.ST-00003561-1_2AMP  HMNMFCCXX-6   \n",
       "6    IWG_IND-GMKFCS.ST-00000768-1_2pA  HMNVCCCXX-1   \n",
       "7    IWG_IND-GMKFCS.ST-00000580-1_2pA  HMNVCCCXX-2   \n",
       "8    IWG_IND-GMKFCS.ST-00000589-1_2pA  HMNVCCCXX-3   \n",
       "9    IWG_IND-GMKFCS.ST-00000991-1_2pA  HMNVCCCXX-4   \n",
       "10   IWG_IND-GMKFCS.ST-00000992-1_2pA  HMNVCCCXX-5   \n",
       "11   IWG_IND-GMKFCS.ST-00000603-1_2pA  HMNVCCCXX-6   \n",
       "12   IWG_IND-GMKFCS.ST-00000604-1_2pA  HMNVCCCXX-7   \n",
       "13   IWG_IND-GMKFCS.ST-00000611-1_2pA  HMNVCCCXX-8   \n",
       "14   IWG_IND-GMKFCS.ST-00000818-1_2pA  HMNWGCCXX-1   \n",
       "15   IWG_IND-GMKFCS.ST-00000872-1_2pA  HMNWGCCXX-2   \n",
       "16   IWG_IND-GMKFCS.ST-00000890-1_2pA  HMNWGCCXX-3   \n",
       "17   IWG_IND-GMKFCS.ST-00000891-1_2pA  HMNWGCCXX-4   \n",
       "18   IWG_IND-GMKFCS.ST-00000881-1_2pA  HMNWGCCXX-5   \n",
       "19   IWG_IND-GMKFCS.ST-00000882-1_2pA  HMNWGCCXX-6   \n",
       "20   IWG_IND-GMKFCS.ST-00002913-1_2pA  HMNWGCCXX-7   \n",
       "21   IWG_IND-GMKFCS.ST-00000896-1_2pA  HMNWGCCXX-8   \n",
       "22   IWG_IND-GMKFCS.ST-00035489-1_2pA  HMNYTCCXX-1   \n",
       "23   IWG_IND-GMKFCS.ST-00000897-1_2pA  HMNYTCCXX-2   \n",
       "24   IWG_IND-GMKFCS.ST-00000915-1_2pA  HMNYTCCXX-3   \n",
       "25   IWG_IND-GMKFCS.ST-00000916-1_2pA  HMNYTCCXX-4   \n",
       "26   IWG_IND-GMKFCS.ST-00006135-1_2pA  HMNYTCCXX-5   \n",
       "27   IWG_IND-GMKFCS.ST-00001160-1_2pA  HMNYTCCXX-6   \n",
       "28   IWG_IND-GMKFCS.ST-00001161-1_2pA  HMNYTCCXX-7   \n",
       "29   IWG_IND-GMKFCS.ST-00003609-1_2pA  HMNYTCCXX-8   \n",
       "..                                ...          ...   \n",
       "870         GMKFCS.ST-00000847-1_2AMP  HNLGYCCXX-1   \n",
       "871         GMKFCS.ST-00000848-1_3AMP  HNLGYCCXX-2   \n",
       "872         GMKFCS.ST-00000849-1_2AMP  HNLGYCCXX-3   \n",
       "873         GMKFCS.ST-00001397-1_2AMP  HNLGYCCXX-4   \n",
       "874         GMKFCS.ST-00001398-1_3AMP  HNLGYCCXX-5   \n",
       "875         GMKFCS.ST-00001399-1_2AMP  HNLGYCCXX-6   \n",
       "876  IWG_IND-GMKFCS.ST-00000097-1_2pA  HNLLGCCXX-1   \n",
       "877  IWG_IND-GMKFCS.ST-00001517-1_3pA  HNLLGCCXX-2   \n",
       "878  IWG_IND-GMKFCS.ST-00001518-1_2pA  HNLLGCCXX-3   \n",
       "879  IWG_IND-GMKFCS.ST-00000315-1_2pA  HNLLGCCXX-4   \n",
       "880  IWG_IND-GMKFCS.ST-00000319-1_3pA  HNLLGCCXX-5   \n",
       "881  IWG_IND-GMKFCS.ST-00004955-1_2pA  HNLLGCCXX-6   \n",
       "882         GMKFCS.ST-00004736-1_3AMP  HNLTMCCXX-3   \n",
       "883         GMKFCS.ST-00004743-1_2AMP  HNLTMCCXX-4   \n",
       "884         GMKFCS.ST-00004745-1_2AMP  HNLTMCCXX-5   \n",
       "885         GMKFCS.ST-00005127-1_3AMP  HNLTMCCXX-6   \n",
       "886         GMKFCS.ST-00005128-1_2AMP  HNLTMCCXX-7   \n",
       "887         GMKFCS.ST-00005129-1_2AMP  HNLTMCCXX-8   \n",
       "888         GMKFCS.ST-00001875-1_3AMP  HNLWWCCXX-3   \n",
       "889         GMKFCS.ST-00001876-1_3AMP  HNLWWCCXX-4   \n",
       "890         GMKFCS.ST-00001877-1_2AMP  HNLWWCCXX-5   \n",
       "891         GMKFCS.ST-00002220-1_3AMP  HNLWWCCXX-6   \n",
       "892         GMKFCS.ST-00002221-1_2AMP  HNLWWCCXX-7   \n",
       "893         GMKFCS.ST-00002222-1_2AMP  HNLWWCCXX-8   \n",
       "894         GMKFCS.ST-00006214-1_3AMP  HNLYFCCXX-1   \n",
       "895         GMKFCS.ST-00006215-1_2AMP  HNLYFCCXX-2   \n",
       "896         GMKFCS.ST-00006216-1_2AMP  HNLYFCCXX-3   \n",
       "897         GMKFCS.ST-00006246-1_2AMP  HNLYFCCXX-4   \n",
       "898         GMKFCS.ST-00006249-1_2AMP  HNLYFCCXX-5   \n",
       "899         GMKFCS.ST-00006250-1_3AMP  HNLYFCCXX-6   \n",
       "\n",
       "                               run_name  read_length  \\\n",
       "0    160423_ST-E00212_0248_AHMNMFCCXX_1          150   \n",
       "1    160423_ST-E00212_0248_AHMNMFCCXX_2          150   \n",
       "2    160423_ST-E00212_0248_AHMNMFCCXX_3          150   \n",
       "3    160423_ST-E00212_0248_AHMNMFCCXX_4          150   \n",
       "4    160423_ST-E00212_0248_AHMNMFCCXX_5          150   \n",
       "5    160423_ST-E00212_0248_AHMNMFCCXX_6          150   \n",
       "6       160402_E00380_0096_BHMNVCCCXX_1          150   \n",
       "7       160402_E00380_0096_BHMNVCCCXX_2          150   \n",
       "8       160402_E00380_0096_BHMNVCCCXX_3          150   \n",
       "9       160402_E00380_0096_BHMNVCCCXX_4          150   \n",
       "10      160402_E00380_0096_BHMNVCCCXX_5          150   \n",
       "11      160402_E00380_0096_BHMNVCCCXX_6          150   \n",
       "12      160402_E00380_0096_BHMNVCCCXX_7          150   \n",
       "13      160402_E00380_0096_BHMNVCCCXX_8          150   \n",
       "14   160404_ST-E00270_0172_AHMNWGCCXX_1          150   \n",
       "15   160404_ST-E00270_0172_AHMNWGCCXX_2          150   \n",
       "16   160404_ST-E00270_0172_AHMNWGCCXX_3          150   \n",
       "17   160404_ST-E00270_0172_AHMNWGCCXX_4          150   \n",
       "18   160404_ST-E00270_0172_AHMNWGCCXX_5          150   \n",
       "19   160404_ST-E00270_0172_AHMNWGCCXX_6          150   \n",
       "20   160404_ST-E00270_0172_AHMNWGCCXX_7          150   \n",
       "21   160404_ST-E00270_0172_AHMNWGCCXX_8          150   \n",
       "22      160404_E00366_0102_AHMNYTCCXX_1          150   \n",
       "23      160404_E00366_0102_AHMNYTCCXX_2          150   \n",
       "24      160404_E00366_0102_AHMNYTCCXX_3          150   \n",
       "25      160404_E00366_0102_AHMNYTCCXX_4          150   \n",
       "26      160404_E00366_0102_AHMNYTCCXX_5          150   \n",
       "27      160404_E00366_0102_AHMNYTCCXX_6          150   \n",
       "28      160404_E00366_0102_AHMNYTCCXX_7          150   \n",
       "29      160404_E00366_0102_AHMNYTCCXX_8          150   \n",
       "..                                  ...          ...   \n",
       "870     160503_E00385_0138_BHNLGYCCXX_1          150   \n",
       "871     160503_E00385_0138_BHNLGYCCXX_2          150   \n",
       "872     160503_E00385_0138_BHNLGYCCXX_3          150   \n",
       "873     160503_E00385_0138_BHNLGYCCXX_4          150   \n",
       "874     160503_E00385_0138_BHNLGYCCXX_5          150   \n",
       "875     160503_E00385_0138_BHNLGYCCXX_6          150   \n",
       "876     160503_E00380_0118_BHNLLGCCXX_1          150   \n",
       "877     160503_E00380_0118_BHNLLGCCXX_2          150   \n",
       "878     160503_E00380_0118_BHNLLGCCXX_3          150   \n",
       "879     160503_E00380_0118_BHNLLGCCXX_4          150   \n",
       "880     160503_E00380_0118_BHNLLGCCXX_5          150   \n",
       "881     160503_E00380_0118_BHNLLGCCXX_6          150   \n",
       "882     160505_E00329_0217_BHNLTMCCXX_3          150   \n",
       "883     160505_E00329_0217_BHNLTMCCXX_4          150   \n",
       "884     160505_E00329_0217_BHNLTMCCXX_5          150   \n",
       "885     160505_E00329_0217_BHNLTMCCXX_6          150   \n",
       "886     160505_E00329_0217_BHNLTMCCXX_7          150   \n",
       "887     160505_E00329_0217_BHNLTMCCXX_8          150   \n",
       "888  160504_ST-E00238_0221_BHNLWWCCXX_3          150   \n",
       "889  160504_ST-E00238_0221_BHNLWWCCXX_4          150   \n",
       "890  160504_ST-E00238_0221_BHNLWWCCXX_5          150   \n",
       "891  160504_ST-E00238_0221_BHNLWWCCXX_6          150   \n",
       "892  160504_ST-E00238_0221_BHNLWWCCXX_7          150   \n",
       "893  160504_ST-E00238_0221_BHNLWWCCXX_8          150   \n",
       "894  160507_ST-E00186_0258_AHNLYFCCXX_1          150   \n",
       "895  160507_ST-E00186_0258_AHNLYFCCXX_2          150   \n",
       "896  160507_ST-E00186_0258_AHNLYFCCXX_3          150   \n",
       "897  160507_ST-E00186_0258_AHNLYFCCXX_4          150   \n",
       "898  160507_ST-E00186_0258_AHNLYFCCXX_5          150   \n",
       "899  160507_ST-E00186_0258_AHNLYFCCXX_6          150   \n",
       "\n",
       "                     ...                  mean_insert_size mean_read_length  \\\n",
       "0                    ...                        392.684444            150.0   \n",
       "1                    ...                        392.684444            150.0   \n",
       "2                    ...                        392.684444            150.0   \n",
       "3                    ...                        392.684444            150.0   \n",
       "4                    ...                        392.684444            150.0   \n",
       "5                    ...                        392.684444            150.0   \n",
       "6                    ...                        392.684444            150.0   \n",
       "7                    ...                        392.684444            150.0   \n",
       "8                    ...                        392.684444            150.0   \n",
       "9                    ...                        392.684444            150.0   \n",
       "10                   ...                        392.684444            150.0   \n",
       "11                   ...                        392.684444            150.0   \n",
       "12                   ...                        392.684444            150.0   \n",
       "13                   ...                        392.684444            150.0   \n",
       "14                   ...                        392.684444            150.0   \n",
       "15                   ...                        392.684444            150.0   \n",
       "16                   ...                        392.684444            150.0   \n",
       "17                   ...                        392.684444            150.0   \n",
       "18                   ...                        392.684444            150.0   \n",
       "19                   ...                        392.684444            150.0   \n",
       "20                   ...                        392.684444            150.0   \n",
       "21                   ...                        392.684444            150.0   \n",
       "22                   ...                        392.684444            150.0   \n",
       "23                   ...                        392.684444            150.0   \n",
       "24                   ...                        392.684444            150.0   \n",
       "25                   ...                        392.684444            150.0   \n",
       "26                   ...                        392.684444            150.0   \n",
       "27                   ...                        392.684444            150.0   \n",
       "28                   ...                        392.684444            150.0   \n",
       "29                   ...                        392.684444            150.0   \n",
       "..                   ...                               ...              ...   \n",
       "870                  ...                        392.684444            150.0   \n",
       "871                  ...                        392.684444            150.0   \n",
       "872                  ...                        392.684444            150.0   \n",
       "873                  ...                        392.684444            150.0   \n",
       "874                  ...                        392.684444            150.0   \n",
       "875                  ...                        392.684444            150.0   \n",
       "876                  ...                        392.684444            150.0   \n",
       "877                  ...                        392.684444            150.0   \n",
       "878                  ...                        392.684444            150.0   \n",
       "879                  ...                        392.684444            150.0   \n",
       "880                  ...                        392.684444            150.0   \n",
       "881                  ...                        392.684444            150.0   \n",
       "882                  ...                        392.684444            150.0   \n",
       "883                  ...                        392.684444            150.0   \n",
       "884                  ...                        392.684444            150.0   \n",
       "885                  ...                        392.684444            150.0   \n",
       "886                  ...                        392.684444            150.0   \n",
       "887                  ...                        392.684444            150.0   \n",
       "888                  ...                        392.684444            150.0   \n",
       "889                  ...                        392.684444            150.0   \n",
       "890                  ...                        392.684444            150.0   \n",
       "891                  ...                        392.684444            150.0   \n",
       "892                  ...                        392.684444            150.0   \n",
       "893                  ...                        392.684444            150.0   \n",
       "894                  ...                        392.684444            150.0   \n",
       "895                  ...                        392.684444            150.0   \n",
       "896                  ...                        392.684444            150.0   \n",
       "897                  ...                        392.684444            150.0   \n",
       "898                  ...                        392.684444            150.0   \n",
       "899                  ...                        392.684444            150.0   \n",
       "\n",
       "    total_reads                data_type  file_format            file_name  \\\n",
       "0           900  submitted aligned reads      hgv.bam  HMNMFCCXX-1.hgv.bam   \n",
       "1           900  submitted aligned reads      hgv.bam  HMNMFCCXX-2.hgv.bam   \n",
       "2           900  submitted aligned reads      hgv.bam  HMNMFCCXX-3.hgv.bam   \n",
       "3           900  submitted aligned reads      hgv.bam  HMNMFCCXX-4.hgv.bam   \n",
       "4           900  submitted aligned reads      hgv.bam  HMNMFCCXX-5.hgv.bam   \n",
       "5           900  submitted aligned reads      hgv.bam  HMNMFCCXX-6.hgv.bam   \n",
       "6           900  submitted aligned reads      hgv.bam  HMNVCCCXX-1.hgv.bam   \n",
       "7           900  submitted aligned reads      hgv.bam  HMNVCCCXX-2.hgv.bam   \n",
       "8           900  submitted aligned reads      hgv.bam  HMNVCCCXX-3.hgv.bam   \n",
       "9           900  submitted aligned reads      hgv.bam  HMNVCCCXX-4.hgv.bam   \n",
       "10          900  submitted aligned reads      hgv.bam  HMNVCCCXX-5.hgv.bam   \n",
       "11          900  submitted aligned reads      hgv.bam  HMNVCCCXX-6.hgv.bam   \n",
       "12          900  submitted aligned reads      hgv.bam  HMNVCCCXX-7.hgv.bam   \n",
       "13          900  submitted aligned reads      hgv.bam  HMNVCCCXX-8.hgv.bam   \n",
       "14          900  submitted aligned reads      hgv.bam  HMNWGCCXX-1.hgv.bam   \n",
       "15          900  submitted aligned reads      hgv.bam  HMNWGCCXX-2.hgv.bam   \n",
       "16          900  submitted aligned reads      hgv.bam  HMNWGCCXX-3.hgv.bam   \n",
       "17          900  submitted aligned reads      hgv.bam  HMNWGCCXX-4.hgv.bam   \n",
       "18          900  submitted aligned reads      hgv.bam  HMNWGCCXX-5.hgv.bam   \n",
       "19          900  submitted aligned reads      hgv.bam  HMNWGCCXX-6.hgv.bam   \n",
       "20          900  submitted aligned reads      hgv.bam  HMNWGCCXX-7.hgv.bam   \n",
       "21          900  submitted aligned reads      hgv.bam  HMNWGCCXX-8.hgv.bam   \n",
       "22          900  submitted aligned reads      hgv.bam  HMNYTCCXX-1.hgv.bam   \n",
       "23          900  submitted aligned reads      hgv.bam  HMNYTCCXX-2.hgv.bam   \n",
       "24          900  submitted aligned reads      hgv.bam  HMNYTCCXX-3.hgv.bam   \n",
       "25          900  submitted aligned reads      hgv.bam  HMNYTCCXX-4.hgv.bam   \n",
       "26          900  submitted aligned reads      hgv.bam  HMNYTCCXX-5.hgv.bam   \n",
       "27          900  submitted aligned reads      hgv.bam  HMNYTCCXX-6.hgv.bam   \n",
       "28          900  submitted aligned reads      hgv.bam  HMNYTCCXX-7.hgv.bam   \n",
       "29          900  submitted aligned reads      hgv.bam  HMNYTCCXX-8.hgv.bam   \n",
       "..          ...                      ...          ...                  ...   \n",
       "870         900  submitted aligned reads      hgv.bam  HNLGYCCXX-1.hgv.bam   \n",
       "871         900  submitted aligned reads      hgv.bam  HNLGYCCXX-2.hgv.bam   \n",
       "872         900  submitted aligned reads      hgv.bam  HNLGYCCXX-3.hgv.bam   \n",
       "873         900  submitted aligned reads      hgv.bam  HNLGYCCXX-4.hgv.bam   \n",
       "874         900  submitted aligned reads      hgv.bam  HNLGYCCXX-5.hgv.bam   \n",
       "875         900  submitted aligned reads      hgv.bam  HNLGYCCXX-6.hgv.bam   \n",
       "876         900  submitted aligned reads      hgv.bam  HNLLGCCXX-1.hgv.bam   \n",
       "877         900  submitted aligned reads      hgv.bam  HNLLGCCXX-2.hgv.bam   \n",
       "878         900  submitted aligned reads      hgv.bam  HNLLGCCXX-3.hgv.bam   \n",
       "879         900  submitted aligned reads      hgv.bam  HNLLGCCXX-4.hgv.bam   \n",
       "880         900  submitted aligned reads      hgv.bam  HNLLGCCXX-5.hgv.bam   \n",
       "881         900  submitted aligned reads      hgv.bam  HNLLGCCXX-6.hgv.bam   \n",
       "882         900  submitted aligned reads      hgv.bam  HNLTMCCXX-3.hgv.bam   \n",
       "883         900  submitted aligned reads      hgv.bam  HNLTMCCXX-4.hgv.bam   \n",
       "884         900  submitted aligned reads      hgv.bam  HNLTMCCXX-5.hgv.bam   \n",
       "885         900  submitted aligned reads      hgv.bam  HNLTMCCXX-6.hgv.bam   \n",
       "886         900  submitted aligned reads      hgv.bam  HNLTMCCXX-7.hgv.bam   \n",
       "887         900  submitted aligned reads      hgv.bam  HNLTMCCXX-8.hgv.bam   \n",
       "888         900  submitted aligned reads      hgv.bam  HNLWWCCXX-3.hgv.bam   \n",
       "889         900  submitted aligned reads      hgv.bam  HNLWWCCXX-4.hgv.bam   \n",
       "890         900  submitted aligned reads      hgv.bam  HNLWWCCXX-5.hgv.bam   \n",
       "891         900  submitted aligned reads      hgv.bam  HNLWWCCXX-6.hgv.bam   \n",
       "892         900  submitted aligned reads      hgv.bam  HNLWWCCXX-7.hgv.bam   \n",
       "893         900  submitted aligned reads      hgv.bam  HNLWWCCXX-8.hgv.bam   \n",
       "894         900  submitted aligned reads      hgv.bam  HNLYFCCXX-1.hgv.bam   \n",
       "895         900  submitted aligned reads      hgv.bam  HNLYFCCXX-2.hgv.bam   \n",
       "896         900  submitted aligned reads      hgv.bam  HNLYFCCXX-3.hgv.bam   \n",
       "897         900  submitted aligned reads      hgv.bam  HNLYFCCXX-4.hgv.bam   \n",
       "898         900  submitted aligned reads      hgv.bam  HNLYFCCXX-5.hgv.bam   \n",
       "899         900  submitted aligned reads      hgv.bam  HNLYFCCXX-6.hgv.bam   \n",
       "\n",
       "       file_size                                          file_url  \\\n",
       "0    64365517822  s3://kf-seq-data-bcm/seidman/HMNMFCCXX-1.hgv.bam   \n",
       "1    63435348733  s3://kf-seq-data-bcm/seidman/HMNMFCCXX-2.hgv.bam   \n",
       "2    63739965918  s3://kf-seq-data-bcm/seidman/HMNMFCCXX-3.hgv.bam   \n",
       "3    62635938381  s3://kf-seq-data-bcm/seidman/HMNMFCCXX-4.hgv.bam   \n",
       "4    64626450198  s3://kf-seq-data-bcm/seidman/HMNMFCCXX-5.hgv.bam   \n",
       "5    62979984038  s3://kf-seq-data-bcm/seidman/HMNMFCCXX-6.hgv.bam   \n",
       "6    68933421158  s3://kf-seq-data-bcm/seidman/HMNVCCCXX-1.hgv.bam   \n",
       "7    70510006515  s3://kf-seq-data-bcm/seidman/HMNVCCCXX-2.hgv.bam   \n",
       "8    70748511577  s3://kf-seq-data-bcm/seidman/HMNVCCCXX-3.hgv.bam   \n",
       "9    70496953371  s3://kf-seq-data-bcm/seidman/HMNVCCCXX-4.hgv.bam   \n",
       "10   69942859118  s3://kf-seq-data-bcm/seidman/HMNVCCCXX-5.hgv.bam   \n",
       "11   72853808794  s3://kf-seq-data-bcm/seidman/HMNVCCCXX-6.hgv.bam   \n",
       "12   72605537636  s3://kf-seq-data-bcm/seidman/HMNVCCCXX-7.hgv.bam   \n",
       "13   70275800752  s3://kf-seq-data-bcm/seidman/HMNVCCCXX-8.hgv.bam   \n",
       "14   63701059861  s3://kf-seq-data-bcm/seidman/HMNWGCCXX-1.hgv.bam   \n",
       "15   63130114606  s3://kf-seq-data-bcm/seidman/HMNWGCCXX-2.hgv.bam   \n",
       "16   66284290234  s3://kf-seq-data-bcm/seidman/HMNWGCCXX-3.hgv.bam   \n",
       "17   63887749384  s3://kf-seq-data-bcm/seidman/HMNWGCCXX-4.hgv.bam   \n",
       "18   62551790031  s3://kf-seq-data-bcm/seidman/HMNWGCCXX-5.hgv.bam   \n",
       "19   63419244156  s3://kf-seq-data-bcm/seidman/HMNWGCCXX-6.hgv.bam   \n",
       "20   63390037001  s3://kf-seq-data-bcm/seidman/HMNWGCCXX-7.hgv.bam   \n",
       "21   65003434003  s3://kf-seq-data-bcm/seidman/HMNWGCCXX-8.hgv.bam   \n",
       "22   68731807689  s3://kf-seq-data-bcm/seidman/HMNYTCCXX-1.hgv.bam   \n",
       "23   71209716845  s3://kf-seq-data-bcm/seidman/HMNYTCCXX-2.hgv.bam   \n",
       "24   72340921082  s3://kf-seq-data-bcm/seidman/HMNYTCCXX-3.hgv.bam   \n",
       "25   73775241806  s3://kf-seq-data-bcm/seidman/HMNYTCCXX-4.hgv.bam   \n",
       "26   69182413410  s3://kf-seq-data-bcm/seidman/HMNYTCCXX-5.hgv.bam   \n",
       "27   71554102371  s3://kf-seq-data-bcm/seidman/HMNYTCCXX-6.hgv.bam   \n",
       "28   71182790908  s3://kf-seq-data-bcm/seidman/HMNYTCCXX-7.hgv.bam   \n",
       "29   73017218859  s3://kf-seq-data-bcm/seidman/HMNYTCCXX-8.hgv.bam   \n",
       "..           ...                                               ...   \n",
       "870  63045174117  s3://kf-seq-data-bcm/seidman/HNLGYCCXX-1.hgv.bam   \n",
       "871  62451333230  s3://kf-seq-data-bcm/seidman/HNLGYCCXX-2.hgv.bam   \n",
       "872  66394383540  s3://kf-seq-data-bcm/seidman/HNLGYCCXX-3.hgv.bam   \n",
       "873  65400960140  s3://kf-seq-data-bcm/seidman/HNLGYCCXX-4.hgv.bam   \n",
       "874  64656711688  s3://kf-seq-data-bcm/seidman/HNLGYCCXX-5.hgv.bam   \n",
       "875  65028281477  s3://kf-seq-data-bcm/seidman/HNLGYCCXX-6.hgv.bam   \n",
       "876  67558188737  s3://kf-seq-data-bcm/seidman/HNLLGCCXX-1.hgv.bam   \n",
       "877  64770159106  s3://kf-seq-data-bcm/seidman/HNLLGCCXX-2.hgv.bam   \n",
       "878  70198446428  s3://kf-seq-data-bcm/seidman/HNLLGCCXX-3.hgv.bam   \n",
       "879  68526123888  s3://kf-seq-data-bcm/seidman/HNLLGCCXX-4.hgv.bam   \n",
       "880  70615280661  s3://kf-seq-data-bcm/seidman/HNLLGCCXX-5.hgv.bam   \n",
       "881  68833299936  s3://kf-seq-data-bcm/seidman/HNLLGCCXX-6.hgv.bam   \n",
       "882  66561442449  s3://kf-seq-data-bcm/seidman/HNLTMCCXX-3.hgv.bam   \n",
       "883  67423179381  s3://kf-seq-data-bcm/seidman/HNLTMCCXX-4.hgv.bam   \n",
       "884  66903856843  s3://kf-seq-data-bcm/seidman/HNLTMCCXX-5.hgv.bam   \n",
       "885  68643834876  s3://kf-seq-data-bcm/seidman/HNLTMCCXX-6.hgv.bam   \n",
       "886  62853583002  s3://kf-seq-data-bcm/seidman/HNLTMCCXX-7.hgv.bam   \n",
       "887  63770026098  s3://kf-seq-data-bcm/seidman/HNLTMCCXX-8.hgv.bam   \n",
       "888  65574536013  s3://kf-seq-data-bcm/seidman/HNLWWCCXX-3.hgv.bam   \n",
       "889  63403425870  s3://kf-seq-data-bcm/seidman/HNLWWCCXX-4.hgv.bam   \n",
       "890  66024331506  s3://kf-seq-data-bcm/seidman/HNLWWCCXX-5.hgv.bam   \n",
       "891  66062654349  s3://kf-seq-data-bcm/seidman/HNLWWCCXX-6.hgv.bam   \n",
       "892  64825822393  s3://kf-seq-data-bcm/seidman/HNLWWCCXX-7.hgv.bam   \n",
       "893  65530524355  s3://kf-seq-data-bcm/seidman/HNLWWCCXX-8.hgv.bam   \n",
       "894  65960227146  s3://kf-seq-data-bcm/seidman/HNLYFCCXX-1.hgv.bam   \n",
       "895  67502614833  s3://kf-seq-data-bcm/seidman/HNLYFCCXX-2.hgv.bam   \n",
       "896  65801483473  s3://kf-seq-data-bcm/seidman/HNLYFCCXX-3.hgv.bam   \n",
       "897  66789458840  s3://kf-seq-data-bcm/seidman/HNLYFCCXX-4.hgv.bam   \n",
       "898  65925371957  s3://kf-seq-data-bcm/seidman/HNLYFCCXX-5.hgv.bam   \n",
       "899  65541019973  s3://kf-seq-data-bcm/seidman/HNLYFCCXX-6.hgv.bam   \n",
       "\n",
       "                               md5sum                                  uuid  \n",
       "0    6f111ed7b35e421fce94571eba8e5be8  e88c2d14-0b7e-43e8-9f7e-7020793b6ebe  \n",
       "1    fb82871da4cc68c70a971dd55677d933  40fb29d7-71e1-426e-9678-a54f22a4f80e  \n",
       "2    748932ce3d9edfc9fba57291c6f632a5  d19eeeae-6acd-4f08-b5c8-6f200160ebed  \n",
       "3    b60d5db835db5544e43fa2113a0cc08a  c5d91600-e286-4ddd-bbc0-e111b079f8bf  \n",
       "4    068b52d7685a5e1916dbb784da850f3b  6904e887-901c-4df6-8579-0ae55cc11125  \n",
       "5    e36e9551a23d5065cbfd9eea8069ee73  6b60464f-31cd-411c-91b4-c33d535cb01f  \n",
       "6    be2c10950413714413e35743187248c7  1ab280b9-e054-4e25-a88a-d7566b02510a  \n",
       "7    794bb59888d85fd092d45c1e7da9dc2e  a79803fc-372e-40fa-b62d-29bb5b5fbaf0  \n",
       "8    02f5e1d8fc3cf41a1993ea369a88411e  5cea8a2f-cfd6-45da-9cf0-0aff9bc3185c  \n",
       "9    abcdeb907c26445979b5ef79b327835e  1d6a481c-a86a-4cc2-acc1-45829d0dbb8e  \n",
       "10   5bfe2294575b6ddd11bf285ca802b6b8  dd743abd-b615-400e-a295-937bc5af13be  \n",
       "11   db829e78b533409b5540c2c6ac0afdb9  7639d110-3c0d-4a73-8cde-ab42b9bbc4ce  \n",
       "12   a277e75e3da743579816a49c88bafde9  60582705-673d-4a60-9648-1e8fa71e0aaf  \n",
       "13   0433ba0e658ec4ac9d0a71d4e2e507c5  2fa09a20-ce15-464d-a326-ffabd8fb7dde  \n",
       "14   5b669a9e5ab932f88c7988a92c9a0fb2  da029205-5dd0-4d73-bd85-3ebefdbbd0c2  \n",
       "15   cb674e8810d8b1b3b10839c42d5e458e  6a3d20c1-e8ff-4d3c-88b7-94d3a8235414  \n",
       "16   e646f30bbc3cab5464d35cb09c0b7222  a3f5d36e-98c6-4de4-82b9-7b2262454b0e  \n",
       "17   eb166bc368340b9a3993be91894efe6a  0b96ae40-1eac-4e4a-b5c0-b13d1521e4da  \n",
       "18   22e7e1579815d43cb795a72bd513fd5e  43a6dd91-6c1b-4a2d-a1c4-edfd661978f3  \n",
       "19   d1d1075323c24986b9c0f61459bddbfa  ae170ddb-792c-4c1c-84d4-3e43d65c0400  \n",
       "20   f25b67087deb3475e113a941df86c37d  30402b37-d1eb-40fb-bdd1-b217644ec722  \n",
       "21   22f7b98ccef9ed1c3f0f44de46d84f33  ca0a0b90-b9e6-437a-809a-2a7a5d3395ff  \n",
       "22   b574a8d1eb18eaf65ab067aa4d6716db  18faf620-7f2a-4ade-af93-4f44a8cf8eea  \n",
       "23   c36e9cdf57d1b9f3d149bf237e7c230a  2b2ba140-c15b-4680-a8df-335b4442c4a7  \n",
       "24   20d8715c78be4035171aa01517128e45  250fea48-57e5-430a-8131-7935c6e51ac4  \n",
       "25   f93bdc0b155364e608af43d432912622  724f015b-82c5-4349-96b2-ef354e04b918  \n",
       "26   938aa2542c0757060fe5dd559233cfce  014ffc3c-27e8-4357-b23b-1885c0b286f3  \n",
       "27   2ca3a0716b017d37993734fe8269fd8b  77087e9c-b6c1-44bb-a628-c1558592c090  \n",
       "28   d99b0805a24e77788ac711621b28cb2c  51516d9c-ddf5-42a3-99cf-6766ffc989f1  \n",
       "29   cee527f56a1d87faeaf0ef456bb23327  35e7b6c7-68e4-4b4d-9f25-402675a5f4eb  \n",
       "..                                ...                                   ...  \n",
       "870  4e09c16d079b79add9c01afa97d2efe4  13160506-4ad6-4310-b7c1-839327584193  \n",
       "871  450e1fe98480416e36ac7be03d4c2c54  be75c603-a0c0-4a28-8c9a-00c2a59380ac  \n",
       "872  608cc692b1d487a111e3bee9fc0b39d4  77c02f21-42c0-4b38-b03a-80650f1559bb  \n",
       "873  7301fb467e9ed731fcea25a600794049  37564660-0c05-494d-8016-a7cbd1ae6397  \n",
       "874  5357d4c950971594895faea2ccfe22aa  9a3f376f-2aa3-4f04-9dd4-472516f766b9  \n",
       "875  a542c2bc1ff3f6b2481d62179038033c  7f6a770f-44ce-4437-83dd-ce95ee211375  \n",
       "876  1b0ceb7104efef3815a56239b9946bbe  d7395c41-eeac-4901-b034-76fb6ca334e0  \n",
       "877  4db9c33019180df0ea5a7640393a98b2  10843f7e-3b91-471c-a993-ee61e75d9c37  \n",
       "878  170fe0b6ff7fcff6e54332e4cf34ae6f  7c161065-01ce-449f-a619-229f33cb485d  \n",
       "879  64b78adc81a019448e710d3b663d9ecf  28a71fcd-7b64-4cd4-bb11-b5e93a792efe  \n",
       "880  2dd28b16ebf9e80fcd02ce8b95d877e0  9436e470-aed2-4346-af1f-37c4dc3f1700  \n",
       "881  023501d1533a29bf8cf8ab4b08a182d2  d3d2eb9e-86eb-4187-9948-2d14f9893d6f  \n",
       "882  22cdd4595cd1831f5bb20f82b956e4dd  a46a84b2-951c-4838-95c9-ddc78074f3f3  \n",
       "883  a94e17951b8e6f198a3d6710e16e8dae  d197c7d7-1899-45bc-9ee7-f17fde386526  \n",
       "884  a05423c9c966bfd2a17cfafc8961f5e3  85e6b54e-ca93-4990-bdcd-284744267f8e  \n",
       "885  01bb3e5d2a49761400afbcf7ced243fc  dc697d71-0a64-4d82-a671-7c0b114444b0  \n",
       "886  fefa5fe465cf5deb7f0d32553c231cc0  dcde4f71-29a6-489a-8785-370b9d6386d0  \n",
       "887  1afca9e0c32a5267a593b6b84562e9be  0adb51dc-d15a-4898-a6f4-66c25ae37028  \n",
       "888  fbf5990c65fb570eaff851f754078326  42d783cf-a91f-40ea-a6da-462608d3dd15  \n",
       "889  286e7eeb97b24c6c77609af176a8ad80  6889ba10-d670-4bd4-b25f-18ef9be454c8  \n",
       "890  dfa2c3c212aa85dfc7aa48a091e437e3  1b50fde7-6b44-4c7d-b8d4-88e65627d4bc  \n",
       "891  2a365584395a733f32539a98d035f7bb  3c207f39-78c0-4cd0-978e-3c94d8056e18  \n",
       "892  fb1ef6c7bb954c7a6b8ce256b911c0bb  61084a2e-d0ad-4a2d-ac3a-127925eeef78  \n",
       "893  b1c11f1f2a1e1979bd8d67ddc7eaf33d  366464c9-290b-40de-91c4-e9c8a5b2fb92  \n",
       "894  798b6b4f949629f370c681d701e9dfc2  fe4aed3b-2429-4ceb-b64f-90360c71fc2d  \n",
       "895  6ff605480071139a2c90e3f06893601b  44a4ddd8-2a0d-46dd-91d5-a262cef630c5  \n",
       "896  76636b8ac9fa9e4fe5fad431a88b1dc8  2faf8164-925a-4d83-a4e3-b0334b2fedef  \n",
       "897  6533372f0751391ba22cbb846e779d7f  60811aff-e630-490a-9d29-f2dd376f7b3b  \n",
       "898  16f2f3f1fd99e3ff2942fe2ba5a0f653  c028a1e1-3fe3-481e-b98b-b60e67f83c6b  \n",
       "899  760439a8531adf583547be4c667c0b84  10352b13-62d1-4a09-853e-cbfcdbf96ab5  \n",
       "\n",
       "[900 rows x 28 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read genomic file info\n",
    "gf_file_info_df = read_genomic_file_info()\n",
    "# Sample and BAM File df\n",
    "sample_gf_df = read_sample_gf_data()\n",
    "# Merge with sequencing experiment df\n",
    "df1 = pd.merge(sample_gf_df, seq_exp_df, left_on='dbgap_subject_id', right_on='sample_name')\n",
    "# Merge with genomic file info df\n",
    "df2 = pd.merge(df1, gf_file_info_df, left_on='BAM sample ID', right_on='file_name')\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>unique</th>\n",
       "      <th>top</th>\n",
       "      <th>freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BAM sample ID</th>\n",
       "      <td>900</td>\n",
       "      <td>900</td>\n",
       "      <td>HN7NVCCXX-1.hgv.bam</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Library</th>\n",
       "      <td>900</td>\n",
       "      <td>900</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00020970-1_2pA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HGSC Sample ID</th>\n",
       "      <td>900</td>\n",
       "      <td>900</td>\n",
       "      <td>ST-00019241</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbgap_subject_id</th>\n",
       "      <td>900</td>\n",
       "      <td>900</td>\n",
       "      <td>CG0008-8428</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <td>900</td>\n",
       "      <td>900</td>\n",
       "      <td>CG0008-8428</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>library_name</th>\n",
       "      <td>900</td>\n",
       "      <td>900</td>\n",
       "      <td>IWG_IND-GMKFCS.ST-00020970-1_2pA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rg_barcode</th>\n",
       "      <td>900</td>\n",
       "      <td>900</td>\n",
       "      <td>HN7GNCCXX-3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_name</th>\n",
       "      <td>900</td>\n",
       "      <td>900</td>\n",
       "      <td>160412_ST-E00238_0206_BHN7VCCCXX_3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>date</th>\n",
       "      <td>900</td>\n",
       "      <td>18</td>\n",
       "      <td>2016-05-05 00:00:00</td>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cohort</th>\n",
       "      <td>900</td>\n",
       "      <td>1</td>\n",
       "      <td>GMKF-Seidman</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>library_strategy</th>\n",
       "      <td>900</td>\n",
       "      <td>1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>library_source</th>\n",
       "      <td>900</td>\n",
       "      <td>1</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>library_selection</th>\n",
       "      <td>900</td>\n",
       "      <td>1</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>instrument</th>\n",
       "      <td>900</td>\n",
       "      <td>1</td>\n",
       "      <td>HiSeq X Ten</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>library_layout</th>\n",
       "      <td>900</td>\n",
       "      <td>1</td>\n",
       "      <td>Paired-End</td>\n",
       "      <td>900</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  count unique                                 top freq\n",
       "BAM sample ID       900    900                 HN7NVCCXX-1.hgv.bam    1\n",
       "Library             900    900    IWG_IND-GMKFCS.ST-00020970-1_2pA    1\n",
       "HGSC Sample ID      900    900                         ST-00019241    1\n",
       "dbgap_subject_id    900    900                         CG0008-8428    1\n",
       "sample_name         900    900                         CG0008-8428    1\n",
       "library_name        900    900    IWG_IND-GMKFCS.ST-00020970-1_2pA    1\n",
       "rg_barcode          900    900                         HN7GNCCXX-3    1\n",
       "run_name            900    900  160412_ST-E00238_0206_BHN7VCCCXX_3    1\n",
       "date                900     18                 2016-05-05 00:00:00  206\n",
       "Cohort              900      1                        GMKF-Seidman  900\n",
       "library_strategy    900      1                                 WGS  900\n",
       "library_source      900      1                             GENOMIC  900\n",
       "library_selection   900      1                              RANDOM  900\n",
       "instrument          900      1                         HiSeq X Ten  900\n",
       "library_layout      900      1                          Paired-End  900"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.describe(include=['O']).T.sort_values('unique', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "73.775241806\n",
      "47.377275825\n",
      "65.6578559692\n"
     ]
    }
   ],
   "source": [
    "df = read_genomic_file_info()\n",
    "print(df['file_size'].max()/1000000000)\n",
    "print(df['file_size'].min()/1000000000)\n",
    "print(df['file_size'].mean()/1000000000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['negative', 'positive'], dtype=object)"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = read_phenotype_data()\n",
    "df['observed'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uuid</th>\n",
       "      <th>created_at</th>\n",
       "      <th>modified_at</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>hpo_id</th>\n",
       "      <th>observed</th>\n",
       "      <th>age_at_event_days</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>kf_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>f77d6506-70d8-4075-b4ff-e54fff9a452e</td>\n",
       "      <td>2018-02-27 09:35:00.03638</td>\n",
       "      <td>2018-02-27 09:35:00.036388</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_8JWM84CR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5cdea204-b001-4469-9934-5dfd8ce64f01</td>\n",
       "      <td>2018-02-27 09:35:00.038478</td>\n",
       "      <td>2018-02-27 09:35:00.038484</td>\n",
       "      <td>AIRWAY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_4WAKYZ20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>61c9f931-b47a-4849-bd40-09fc0eb614e6</td>\n",
       "      <td>2018-02-27 09:35:00.039441</td>\n",
       "      <td>2018-02-27 09:35:00.039446</td>\n",
       "      <td>CHEST_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_Z5KN5SN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>51cdd902-d508-48a0-a1c1-77a91d5e57b0</td>\n",
       "      <td>2018-02-27 09:35:00.040381</td>\n",
       "      <td>2018-02-27 09:35:00.040386</td>\n",
       "      <td>CLUB_FOOT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>positive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_H17JQKYG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>a1690ee3-3565-4d2c-8f04-6d903ceda423</td>\n",
       "      <td>2018-02-27 09:35:00.041276</td>\n",
       "      <td>2018-02-27 09:35:00.041281</td>\n",
       "      <td>DERMATOLOGICAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_PFVG3BAZ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>b9f1735e-9de3-41d3-a4f1-3cfc16796ec2</td>\n",
       "      <td>2018-02-27 09:35:00.042194</td>\n",
       "      <td>2018-02-27 09:35:00.042199</td>\n",
       "      <td>DYSMORPHIC_FACIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_BY35ZYB5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ec15cd29-5631-4c38-8795-0bd68ec905cc</td>\n",
       "      <td>2018-02-27 09:35:00.043137</td>\n",
       "      <td>2018-02-27 09:35:00.043176</td>\n",
       "      <td>EAR_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_V672HJ6C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6b241509-9a2a-4bb9-bfbe-0c0b6844579a</td>\n",
       "      <td>2018-02-27 09:35:00.043983</td>\n",
       "      <td>2018-02-27 09:35:00.043989</td>\n",
       "      <td>ENDOCRINOLOGIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_JBK8BD5J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>b7c951dd-989a-4e1a-b721-d55d22024b1d</td>\n",
       "      <td>2018-02-27 09:35:00.044848</td>\n",
       "      <td>2018-02-27 09:35:00.044854</td>\n",
       "      <td>EYE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_E4YDYAEJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>14b79e40-a409-4c4c-a4ee-09da7f67aca7</td>\n",
       "      <td>2018-02-27 09:35:00.045662</td>\n",
       "      <td>2018-02-27 09:35:00.045667</td>\n",
       "      <td>FEET_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>positive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_BCRWHNSK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>fb168208-b5bb-45c2-8da7-8ea509f770e6</td>\n",
       "      <td>2018-02-27 09:35:00.046524</td>\n",
       "      <td>2018-02-27 09:35:00.046529</td>\n",
       "      <td>GENERAL_SKELETAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_2HBS7DFV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>81455ec6-1636-47e8-b015-99e3f77d8dde</td>\n",
       "      <td>2018-02-27 09:35:00.047324</td>\n",
       "      <td>2018-02-27 09:35:00.047329</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_8HMK4B5E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>3d00c9ab-8350-4ed7-81d7-97992cd88253</td>\n",
       "      <td>2018-02-27 09:35:00.048117</td>\n",
       "      <td>2018-02-27 09:35:00.048121</td>\n",
       "      <td>GENITOURINARY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_WCR9MGYP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>09340bec-d7bd-4f8d-bf5b-4230d8b072f5</td>\n",
       "      <td>2018-02-27 09:35:00.04887</td>\n",
       "      <td>2018-02-27 09:35:00.048875</td>\n",
       "      <td>HAND_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>positive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_209WTDM8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>9a798967-4593-4dbc-bcf1-1b9573651469</td>\n",
       "      <td>2018-02-27 09:35:00.049611</td>\n",
       "      <td>2018-02-27 09:35:00.049616</td>\n",
       "      <td>HAND_POLYDACTYLY</td>\n",
       "      <td>HP:0001161</td>\n",
       "      <td>positive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_47E0BVBD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>46b91b44-7f36-4018-8c75-416ef5bcb5e7</td>\n",
       "      <td>2018-02-27 09:35:00.050342</td>\n",
       "      <td>2018-02-27 09:35:00.050347</td>\n",
       "      <td>HEAD_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_SQ5PN54S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>602d9808-266a-47ec-8bb1-13c11f0f65a8</td>\n",
       "      <td>2018-02-27 09:35:00.051045</td>\n",
       "      <td>2018-02-27 09:35:00.05105</td>\n",
       "      <td>HEMATOLOGIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_Z1FYH8TT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6197d021-5f5c-4e98-a588-dc9efd04ad74</td>\n",
       "      <td>2018-02-27 09:35:00.051811</td>\n",
       "      <td>2018-02-27 09:35:00.051816</td>\n",
       "      <td>HEPATIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_VAKJZKD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>26771f5e-0854-4847-bf22-14c7a6b7f56c</td>\n",
       "      <td>2018-02-27 09:35:00.052606</td>\n",
       "      <td>2018-02-27 09:35:00.05261</td>\n",
       "      <td>IMMUNOLOGIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_E8D9V59V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>c5d79c10-b183-45ff-aecf-68f4d3b8cfe1</td>\n",
       "      <td>2018-02-27 09:35:00.053359</td>\n",
       "      <td>2018-02-27 09:35:00.053363</td>\n",
       "      <td>MOUTH_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_NDQ0KD82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>53fff4fb-8ccb-4e9e-95d9-22c797095422</td>\n",
       "      <td>2018-02-27 09:35:00.054133</td>\n",
       "      <td>2018-02-27 09:35:00.054137</td>\n",
       "      <td>NECK_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_JBHMRFVV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>91108812-ba98-4630-84e6-a6772aebef49</td>\n",
       "      <td>2018-02-27 09:35:00.054941</td>\n",
       "      <td>2018-02-27 09:35:00.054946</td>\n",
       "      <td>NEUROLOGICAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_MGR948NV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>7114c05d-ce92-4d2d-8f25-bc2739d2ca24</td>\n",
       "      <td>2018-02-27 09:35:00.055658</td>\n",
       "      <td>2018-02-27 09:35:00.055663</td>\n",
       "      <td>NOSE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_B1M29E8S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>f032d7e8-1ae5-48b4-8737-d9cb503e48ea</td>\n",
       "      <td>2018-02-27 09:35:00.056404</td>\n",
       "      <td>2018-02-27 09:35:00.056409</td>\n",
       "      <td>PALATE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_VHHFH6JN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>46c4c1f0-1766-48e7-a4d6-4c56c3dcc30f</td>\n",
       "      <td>2018-02-27 09:35:00.057155</td>\n",
       "      <td>2018-02-27 09:35:00.057159</td>\n",
       "      <td>PULMONARY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_08QV8GQY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>0677688c-6106-43ec-804a-244ad4e8d5e1</td>\n",
       "      <td>2018-02-27 09:35:00.057969</td>\n",
       "      <td>2018-02-27 09:35:00.057973</td>\n",
       "      <td>RENAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_J12SB1R5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>7bd0d0ec-dc7e-448f-b8ce-a0792a0ba941</td>\n",
       "      <td>2018-02-27 09:35:00.05874</td>\n",
       "      <td>2018-02-27 09:35:00.058744</td>\n",
       "      <td>THYMUS_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_N260Y9FG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>f6b6b95d-ba10-4474-962b-829907772552</td>\n",
       "      <td>2018-02-27 09:35:00.059443</td>\n",
       "      <td>2018-02-27 09:35:00.059448</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_9YVMV8HV</td>\n",
       "      <td>PH_G7VDY692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>45884f70-f8a5-4bf0-921e-30314e17f5be</td>\n",
       "      <td>2018-02-27 09:35:00.060206</td>\n",
       "      <td>2018-02-27 09:35:00.06021</td>\n",
       "      <td>AIRWAY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_9YVMV8HV</td>\n",
       "      <td>PH_VD250RZE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>0825ed0e-b593-468d-beb6-9750c6d1dc6c</td>\n",
       "      <td>2018-02-27 09:35:00.060895</td>\n",
       "      <td>2018-02-27 09:35:00.0609</td>\n",
       "      <td>CHEST_ABNORMALITIES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_9YVMV8HV</td>\n",
       "      <td>PH_VPHNFZ9Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8306</th>\n",
       "      <td>9a1dd373-21e5-4544-84fd-650b615042ac</td>\n",
       "      <td>2018-02-27 09:35:05.863413</td>\n",
       "      <td>2018-02-27 09:35:05.863418</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_SKK2XWHX</td>\n",
       "      <td>PH_87W0X369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8307</th>\n",
       "      <td>3523e046-29f7-4dbd-aeb9-0beb7c3aaece</td>\n",
       "      <td>2018-02-27 09:35:05.864054</td>\n",
       "      <td>2018-02-27 09:35:05.864059</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_3R09E42A</td>\n",
       "      <td>PH_HVV7MJZ1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8308</th>\n",
       "      <td>b583f43e-477d-4019-95cb-cd27e19220e7</td>\n",
       "      <td>2018-02-27 09:35:05.864671</td>\n",
       "      <td>2018-02-27 09:35:05.864675</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_SZSVF28E</td>\n",
       "      <td>PH_437PSFDX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8309</th>\n",
       "      <td>29548995-418f-4a44-a1df-1df54ab9cad5</td>\n",
       "      <td>2018-02-27 09:35:05.865283</td>\n",
       "      <td>2018-02-27 09:35:05.865287</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_DZF6WZTB</td>\n",
       "      <td>PH_W8X0Q8Q1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8310</th>\n",
       "      <td>ffb0caaa-149e-4f08-a882-c25f84d591ad</td>\n",
       "      <td>2018-02-27 09:35:05.865914</td>\n",
       "      <td>2018-02-27 09:35:05.865921</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_9JMZSWZM</td>\n",
       "      <td>PH_A0XF8JPS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8311</th>\n",
       "      <td>4eefea6b-3cb7-473b-b7ca-61368103ae53</td>\n",
       "      <td>2018-02-27 09:35:05.866713</td>\n",
       "      <td>2018-02-27 09:35:05.866719</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_N4XY02RM</td>\n",
       "      <td>PH_4RM7QPNH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8312</th>\n",
       "      <td>036ef30c-3c4a-41f0-93f6-0395918c3d48</td>\n",
       "      <td>2018-02-27 09:35:05.867373</td>\n",
       "      <td>2018-02-27 09:35:05.867378</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_CZVKMYJ5</td>\n",
       "      <td>PH_WNSTES3P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8313</th>\n",
       "      <td>a22e4b42-5447-4564-bc0a-185e3020e2df</td>\n",
       "      <td>2018-02-27 09:35:05.867984</td>\n",
       "      <td>2018-02-27 09:35:05.867989</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_ZKVVK7YB</td>\n",
       "      <td>PH_YEJBZ3FY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8314</th>\n",
       "      <td>d3bf6a0b-7cc9-4bea-a81f-28e4e39f6b41</td>\n",
       "      <td>2018-02-27 09:35:05.868581</td>\n",
       "      <td>2018-02-27 09:35:05.868585</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_NT2QGE62</td>\n",
       "      <td>PH_YHN3SKAY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8315</th>\n",
       "      <td>7de9874b-f952-4bdd-aa70-d5d493921148</td>\n",
       "      <td>2018-02-27 09:35:05.869171</td>\n",
       "      <td>2018-02-27 09:35:05.869175</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_NDNGQ5A9</td>\n",
       "      <td>PH_JGHS1JQ4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8316</th>\n",
       "      <td>6b769d1a-362f-44ac-aac7-11e4ca87ab78</td>\n",
       "      <td>2018-02-27 09:35:05.869758</td>\n",
       "      <td>2018-02-27 09:35:05.869762</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_GV191660</td>\n",
       "      <td>PH_54XJ9AWH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8317</th>\n",
       "      <td>52956a71-1f83-4877-b2e0-48076451fec5</td>\n",
       "      <td>2018-02-27 09:35:05.870441</td>\n",
       "      <td>2018-02-27 09:35:05.870445</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_1718C9HN</td>\n",
       "      <td>PH_WW9GG285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8318</th>\n",
       "      <td>9ba23215-2638-41c3-88ab-240d06e623b0</td>\n",
       "      <td>2018-02-27 09:35:05.871029</td>\n",
       "      <td>2018-02-27 09:35:05.871033</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_PDAY4P1E</td>\n",
       "      <td>PH_0JHW0EGR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8319</th>\n",
       "      <td>c35fcf3d-e74f-4ec0-accf-1e1b5dfb0c5d</td>\n",
       "      <td>2018-02-27 09:35:05.871615</td>\n",
       "      <td>2018-02-27 09:35:05.871619</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_CCR3SXD1</td>\n",
       "      <td>PH_623J09Q4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8320</th>\n",
       "      <td>fdc400be-95a5-42a1-9a71-273c5a0597e0</td>\n",
       "      <td>2018-02-27 09:35:05.872198</td>\n",
       "      <td>2018-02-27 09:35:05.872202</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_FEYRKGW8</td>\n",
       "      <td>PH_WHS8NR0P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8321</th>\n",
       "      <td>ad151aec-07fc-47f0-b7a9-2892fc54967a</td>\n",
       "      <td>2018-02-27 09:35:05.872809</td>\n",
       "      <td>2018-02-27 09:35:05.872813</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_VMW3ESD6</td>\n",
       "      <td>PH_BNEBK2VY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8322</th>\n",
       "      <td>789f40bd-464f-4321-881c-969365a614e5</td>\n",
       "      <td>2018-02-27 09:35:05.873414</td>\n",
       "      <td>2018-02-27 09:35:05.873419</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_XW4D7HBG</td>\n",
       "      <td>PH_SX2BSD03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8323</th>\n",
       "      <td>31e23a7d-f040-4134-af7a-ea3b917e0e1a</td>\n",
       "      <td>2018-02-27 09:35:05.874037</td>\n",
       "      <td>2018-02-27 09:35:05.874041</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_3P4K5J8N</td>\n",
       "      <td>PH_TCT6PQVX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8324</th>\n",
       "      <td>4db4f340-8b07-4147-a12b-2ee9a61b2424</td>\n",
       "      <td>2018-02-27 09:35:05.874627</td>\n",
       "      <td>2018-02-27 09:35:05.874631</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_W3W87FY9</td>\n",
       "      <td>PH_SP6PYSA6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8325</th>\n",
       "      <td>d1cd9026-dec2-43d9-9457-c9fe95bcd464</td>\n",
       "      <td>2018-02-27 09:35:05.875229</td>\n",
       "      <td>2018-02-27 09:35:05.875234</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_9E6GW4YG</td>\n",
       "      <td>PH_GX1C3MAJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8326</th>\n",
       "      <td>5e855ef3-d585-48a2-bb3d-2c1e741dfc5f</td>\n",
       "      <td>2018-02-27 09:35:05.875827</td>\n",
       "      <td>2018-02-27 09:35:05.875832</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_BVCGQ7D8</td>\n",
       "      <td>PH_74HE7FT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8327</th>\n",
       "      <td>f0429bd0-9adc-4169-9ec0-b6e4bfec145e</td>\n",
       "      <td>2018-02-27 09:35:05.876416</td>\n",
       "      <td>2018-02-27 09:35:05.87642</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_16R6B2T7</td>\n",
       "      <td>PH_BV9QVWFV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8328</th>\n",
       "      <td>d34511a6-052f-4ecd-9bca-36c9ba6b15e7</td>\n",
       "      <td>2018-02-27 09:35:05.877007</td>\n",
       "      <td>2018-02-27 09:35:05.877011</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_26NG1DS8</td>\n",
       "      <td>PH_3V3W64CH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8329</th>\n",
       "      <td>2b42ab77-4f24-4424-9c32-f52c7f41f3e6</td>\n",
       "      <td>2018-02-27 09:35:05.877587</td>\n",
       "      <td>2018-02-27 09:35:05.877591</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_0BGHDRZ5</td>\n",
       "      <td>PH_2A77VAS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8330</th>\n",
       "      <td>61f7a594-4e22-4755-929b-cc32d25207ad</td>\n",
       "      <td>2018-02-27 09:35:05.878171</td>\n",
       "      <td>2018-02-27 09:35:05.878175</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_GKH2YJTV</td>\n",
       "      <td>PH_B68G824Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8331</th>\n",
       "      <td>6d1c2b97-13bb-4753-97b9-cfdf29425e35</td>\n",
       "      <td>2018-02-27 09:35:05.878762</td>\n",
       "      <td>2018-02-27 09:35:05.878766</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_5YNTF06A</td>\n",
       "      <td>PH_AHY3TNC0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8332</th>\n",
       "      <td>b9d33cf4-9fb3-43df-85f8-f887da88617c</td>\n",
       "      <td>2018-02-27 09:35:05.87934</td>\n",
       "      <td>2018-02-27 09:35:05.879345</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_WSJ8H57D</td>\n",
       "      <td>PH_ZVEETSYC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8333</th>\n",
       "      <td>b7df02a0-f5a4-4a06-8d0c-f90cf4aac5ef</td>\n",
       "      <td>2018-02-27 09:35:05.880003</td>\n",
       "      <td>2018-02-27 09:35:05.880008</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_WJ2F5KBT</td>\n",
       "      <td>PH_WQA0VP5X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8334</th>\n",
       "      <td>c000b4de-2d18-4484-af5d-4a551fa37011</td>\n",
       "      <td>2018-02-27 09:35:05.880601</td>\n",
       "      <td>2018-02-27 09:35:05.880605</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_E14YEG3F</td>\n",
       "      <td>PH_A246ZWNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8335</th>\n",
       "      <td>5357c1db-6224-4a99-b2ef-4674419b4563</td>\n",
       "      <td>2018-02-27 09:35:05.88124</td>\n",
       "      <td>2018-02-27 09:35:05.881246</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>negative</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PT_G5GKEW54</td>\n",
       "      <td>PH_WNDB15KT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8336 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      uuid                  created_at  \\\n",
       "0     f77d6506-70d8-4075-b4ff-e54fff9a452e   2018-02-27 09:35:00.03638   \n",
       "1     5cdea204-b001-4469-9934-5dfd8ce64f01  2018-02-27 09:35:00.038478   \n",
       "2     61c9f931-b47a-4849-bd40-09fc0eb614e6  2018-02-27 09:35:00.039441   \n",
       "3     51cdd902-d508-48a0-a1c1-77a91d5e57b0  2018-02-27 09:35:00.040381   \n",
       "4     a1690ee3-3565-4d2c-8f04-6d903ceda423  2018-02-27 09:35:00.041276   \n",
       "5     b9f1735e-9de3-41d3-a4f1-3cfc16796ec2  2018-02-27 09:35:00.042194   \n",
       "6     ec15cd29-5631-4c38-8795-0bd68ec905cc  2018-02-27 09:35:00.043137   \n",
       "7     6b241509-9a2a-4bb9-bfbe-0c0b6844579a  2018-02-27 09:35:00.043983   \n",
       "8     b7c951dd-989a-4e1a-b721-d55d22024b1d  2018-02-27 09:35:00.044848   \n",
       "9     14b79e40-a409-4c4c-a4ee-09da7f67aca7  2018-02-27 09:35:00.045662   \n",
       "10    fb168208-b5bb-45c2-8da7-8ea509f770e6  2018-02-27 09:35:00.046524   \n",
       "11    81455ec6-1636-47e8-b015-99e3f77d8dde  2018-02-27 09:35:00.047324   \n",
       "12    3d00c9ab-8350-4ed7-81d7-97992cd88253  2018-02-27 09:35:00.048117   \n",
       "13    09340bec-d7bd-4f8d-bf5b-4230d8b072f5   2018-02-27 09:35:00.04887   \n",
       "14    9a798967-4593-4dbc-bcf1-1b9573651469  2018-02-27 09:35:00.049611   \n",
       "15    46b91b44-7f36-4018-8c75-416ef5bcb5e7  2018-02-27 09:35:00.050342   \n",
       "16    602d9808-266a-47ec-8bb1-13c11f0f65a8  2018-02-27 09:35:00.051045   \n",
       "17    6197d021-5f5c-4e98-a588-dc9efd04ad74  2018-02-27 09:35:00.051811   \n",
       "18    26771f5e-0854-4847-bf22-14c7a6b7f56c  2018-02-27 09:35:00.052606   \n",
       "19    c5d79c10-b183-45ff-aecf-68f4d3b8cfe1  2018-02-27 09:35:00.053359   \n",
       "20    53fff4fb-8ccb-4e9e-95d9-22c797095422  2018-02-27 09:35:00.054133   \n",
       "21    91108812-ba98-4630-84e6-a6772aebef49  2018-02-27 09:35:00.054941   \n",
       "22    7114c05d-ce92-4d2d-8f25-bc2739d2ca24  2018-02-27 09:35:00.055658   \n",
       "23    f032d7e8-1ae5-48b4-8737-d9cb503e48ea  2018-02-27 09:35:00.056404   \n",
       "24    46c4c1f0-1766-48e7-a4d6-4c56c3dcc30f  2018-02-27 09:35:00.057155   \n",
       "25    0677688c-6106-43ec-804a-244ad4e8d5e1  2018-02-27 09:35:00.057969   \n",
       "26    7bd0d0ec-dc7e-448f-b8ce-a0792a0ba941   2018-02-27 09:35:00.05874   \n",
       "27    f6b6b95d-ba10-4474-962b-829907772552  2018-02-27 09:35:00.059443   \n",
       "28    45884f70-f8a5-4bf0-921e-30314e17f5be  2018-02-27 09:35:00.060206   \n",
       "29    0825ed0e-b593-468d-beb6-9750c6d1dc6c  2018-02-27 09:35:00.060895   \n",
       "...                                    ...                         ...   \n",
       "8306  9a1dd373-21e5-4544-84fd-650b615042ac  2018-02-27 09:35:05.863413   \n",
       "8307  3523e046-29f7-4dbd-aeb9-0beb7c3aaece  2018-02-27 09:35:05.864054   \n",
       "8308  b583f43e-477d-4019-95cb-cd27e19220e7  2018-02-27 09:35:05.864671   \n",
       "8309  29548995-418f-4a44-a1df-1df54ab9cad5  2018-02-27 09:35:05.865283   \n",
       "8310  ffb0caaa-149e-4f08-a882-c25f84d591ad  2018-02-27 09:35:05.865914   \n",
       "8311  4eefea6b-3cb7-473b-b7ca-61368103ae53  2018-02-27 09:35:05.866713   \n",
       "8312  036ef30c-3c4a-41f0-93f6-0395918c3d48  2018-02-27 09:35:05.867373   \n",
       "8313  a22e4b42-5447-4564-bc0a-185e3020e2df  2018-02-27 09:35:05.867984   \n",
       "8314  d3bf6a0b-7cc9-4bea-a81f-28e4e39f6b41  2018-02-27 09:35:05.868581   \n",
       "8315  7de9874b-f952-4bdd-aa70-d5d493921148  2018-02-27 09:35:05.869171   \n",
       "8316  6b769d1a-362f-44ac-aac7-11e4ca87ab78  2018-02-27 09:35:05.869758   \n",
       "8317  52956a71-1f83-4877-b2e0-48076451fec5  2018-02-27 09:35:05.870441   \n",
       "8318  9ba23215-2638-41c3-88ab-240d06e623b0  2018-02-27 09:35:05.871029   \n",
       "8319  c35fcf3d-e74f-4ec0-accf-1e1b5dfb0c5d  2018-02-27 09:35:05.871615   \n",
       "8320  fdc400be-95a5-42a1-9a71-273c5a0597e0  2018-02-27 09:35:05.872198   \n",
       "8321  ad151aec-07fc-47f0-b7a9-2892fc54967a  2018-02-27 09:35:05.872809   \n",
       "8322  789f40bd-464f-4321-881c-969365a614e5  2018-02-27 09:35:05.873414   \n",
       "8323  31e23a7d-f040-4134-af7a-ea3b917e0e1a  2018-02-27 09:35:05.874037   \n",
       "8324  4db4f340-8b07-4147-a12b-2ee9a61b2424  2018-02-27 09:35:05.874627   \n",
       "8325  d1cd9026-dec2-43d9-9457-c9fe95bcd464  2018-02-27 09:35:05.875229   \n",
       "8326  5e855ef3-d585-48a2-bb3d-2c1e741dfc5f  2018-02-27 09:35:05.875827   \n",
       "8327  f0429bd0-9adc-4169-9ec0-b6e4bfec145e  2018-02-27 09:35:05.876416   \n",
       "8328  d34511a6-052f-4ecd-9bca-36c9ba6b15e7  2018-02-27 09:35:05.877007   \n",
       "8329  2b42ab77-4f24-4424-9c32-f52c7f41f3e6  2018-02-27 09:35:05.877587   \n",
       "8330  61f7a594-4e22-4755-929b-cc32d25207ad  2018-02-27 09:35:05.878171   \n",
       "8331  6d1c2b97-13bb-4753-97b9-cfdf29425e35  2018-02-27 09:35:05.878762   \n",
       "8332  b9d33cf4-9fb3-43df-85f8-f887da88617c   2018-02-27 09:35:05.87934   \n",
       "8333  b7df02a0-f5a4-4a06-8d0c-f90cf4aac5ef  2018-02-27 09:35:05.880003   \n",
       "8334  c000b4de-2d18-4484-af5d-4a551fa37011  2018-02-27 09:35:05.880601   \n",
       "8335  5357c1db-6224-4a99-b2ef-4674419b4563   2018-02-27 09:35:05.88124   \n",
       "\n",
       "                     modified_at                               phenotype  \\\n",
       "0     2018-02-27 09:35:00.036388         ABDOMINAL_ABNORMALITIES_PRESENT   \n",
       "1     2018-02-27 09:35:00.038484            AIRWAY_ABNORMALITIES_PRESENT   \n",
       "2     2018-02-27 09:35:00.039446             CHEST_ABNORMALITIES_PRESENT   \n",
       "3     2018-02-27 09:35:00.040386                               CLUB_FOOT   \n",
       "4     2018-02-27 09:35:00.041281    DERMATOLOGICAL_ABNORMALITIES_PRESENT   \n",
       "5     2018-02-27 09:35:00.042199               DYSMORPHIC_FACIES_PRESENT   \n",
       "6     2018-02-27 09:35:00.043176               EAR_ABNORMALITIES_PRESENT   \n",
       "7     2018-02-27 09:35:00.043989    ENDOCRINOLOGIC_ABNORMALITIES_PRESENT   \n",
       "8     2018-02-27 09:35:00.044854               EYE_ABNORMALITIES_PRESENT   \n",
       "9     2018-02-27 09:35:00.045667              FEET_ABNORMALITIES_PRESENT   \n",
       "10    2018-02-27 09:35:00.046529  GENERAL_SKELETAL_ABNORMALITIES_PRESENT   \n",
       "11    2018-02-27 09:35:00.047329               GENETIC_SYNDROMES_PRESENT   \n",
       "12    2018-02-27 09:35:00.048121     GENITOURINARY_ABNORMALITIES_PRESENT   \n",
       "13    2018-02-27 09:35:00.048875              HAND_ABNORMALITIES_PRESENT   \n",
       "14    2018-02-27 09:35:00.049616                        HAND_POLYDACTYLY   \n",
       "15    2018-02-27 09:35:00.050347              HEAD_ABNORMALITIES_PRESENT   \n",
       "16     2018-02-27 09:35:00.05105       HEMATOLOGIC_ABNORMALITIES_PRESENT   \n",
       "17    2018-02-27 09:35:00.051816           HEPATIC_ABNORMALITIES_PRESENT   \n",
       "18     2018-02-27 09:35:00.05261       IMMUNOLOGIC_ABNORMALITIES_PRESENT   \n",
       "19    2018-02-27 09:35:00.053363             MOUTH_ABNORMALITIES_PRESENT   \n",
       "20    2018-02-27 09:35:00.054137              NECK_ABNORMALITIES_PRESENT   \n",
       "21    2018-02-27 09:35:00.054946      NEUROLOGICAL_ABNORMALITIES_PRESENT   \n",
       "22    2018-02-27 09:35:00.055663              NOSE_ABNORMALITIES_PRESENT   \n",
       "23    2018-02-27 09:35:00.056409            PALATE_ABNORMALITIES_PRESENT   \n",
       "24    2018-02-27 09:35:00.057159         PULMONARY_ABNORMALITIES_PRESENT   \n",
       "25    2018-02-27 09:35:00.057973             RENAL_ABNORMALITIES_PRESENT   \n",
       "26    2018-02-27 09:35:00.058744            THYMUS_ABNORMALITIES_PRESENT   \n",
       "27    2018-02-27 09:35:00.059448         ABDOMINAL_ABNORMALITIES_PRESENT   \n",
       "28     2018-02-27 09:35:00.06021            AIRWAY_ABNORMALITIES_PRESENT   \n",
       "29      2018-02-27 09:35:00.0609             CHEST_ABNORMALITIES_PRESENT   \n",
       "...                          ...                                     ...   \n",
       "8306  2018-02-27 09:35:05.863418               GENETIC_SYNDROMES_PRESENT   \n",
       "8307  2018-02-27 09:35:05.864059               GENETIC_SYNDROMES_PRESENT   \n",
       "8308  2018-02-27 09:35:05.864675               GENETIC_SYNDROMES_PRESENT   \n",
       "8309  2018-02-27 09:35:05.865287               GENETIC_SYNDROMES_PRESENT   \n",
       "8310  2018-02-27 09:35:05.865921               GENETIC_SYNDROMES_PRESENT   \n",
       "8311  2018-02-27 09:35:05.866719               GENETIC_SYNDROMES_PRESENT   \n",
       "8312  2018-02-27 09:35:05.867378               GENETIC_SYNDROMES_PRESENT   \n",
       "8313  2018-02-27 09:35:05.867989               GENETIC_SYNDROMES_PRESENT   \n",
       "8314  2018-02-27 09:35:05.868585               GENETIC_SYNDROMES_PRESENT   \n",
       "8315  2018-02-27 09:35:05.869175               GENETIC_SYNDROMES_PRESENT   \n",
       "8316  2018-02-27 09:35:05.869762               GENETIC_SYNDROMES_PRESENT   \n",
       "8317  2018-02-27 09:35:05.870445               GENETIC_SYNDROMES_PRESENT   \n",
       "8318  2018-02-27 09:35:05.871033               GENETIC_SYNDROMES_PRESENT   \n",
       "8319  2018-02-27 09:35:05.871619               GENETIC_SYNDROMES_PRESENT   \n",
       "8320  2018-02-27 09:35:05.872202               GENETIC_SYNDROMES_PRESENT   \n",
       "8321  2018-02-27 09:35:05.872813               GENETIC_SYNDROMES_PRESENT   \n",
       "8322  2018-02-27 09:35:05.873419               GENETIC_SYNDROMES_PRESENT   \n",
       "8323  2018-02-27 09:35:05.874041               GENETIC_SYNDROMES_PRESENT   \n",
       "8324  2018-02-27 09:35:05.874631               GENETIC_SYNDROMES_PRESENT   \n",
       "8325  2018-02-27 09:35:05.875234               GENETIC_SYNDROMES_PRESENT   \n",
       "8326  2018-02-27 09:35:05.875832               GENETIC_SYNDROMES_PRESENT   \n",
       "8327   2018-02-27 09:35:05.87642               GENETIC_SYNDROMES_PRESENT   \n",
       "8328  2018-02-27 09:35:05.877011               GENETIC_SYNDROMES_PRESENT   \n",
       "8329  2018-02-27 09:35:05.877591               GENETIC_SYNDROMES_PRESENT   \n",
       "8330  2018-02-27 09:35:05.878175               GENETIC_SYNDROMES_PRESENT   \n",
       "8331  2018-02-27 09:35:05.878766               GENETIC_SYNDROMES_PRESENT   \n",
       "8332  2018-02-27 09:35:05.879345               GENETIC_SYNDROMES_PRESENT   \n",
       "8333  2018-02-27 09:35:05.880008               GENETIC_SYNDROMES_PRESENT   \n",
       "8334  2018-02-27 09:35:05.880605               GENETIC_SYNDROMES_PRESENT   \n",
       "8335  2018-02-27 09:35:05.881246               GENETIC_SYNDROMES_PRESENT   \n",
       "\n",
       "          hpo_id  observed  age_at_event_days participant_id        kf_id  \n",
       "0            NaN  negative                NaN    PT_YJEZ38PS  PH_8JWM84CR  \n",
       "1            NaN  negative                NaN    PT_YJEZ38PS  PH_4WAKYZ20  \n",
       "2            NaN  negative                NaN    PT_YJEZ38PS  PH_Z5KN5SN2  \n",
       "3            NaN  positive                NaN    PT_YJEZ38PS  PH_H17JQKYG  \n",
       "4            NaN  negative                NaN    PT_YJEZ38PS  PH_PFVG3BAZ  \n",
       "5            NaN  negative                NaN    PT_YJEZ38PS  PH_BY35ZYB5  \n",
       "6            NaN  negative                NaN    PT_YJEZ38PS  PH_V672HJ6C  \n",
       "7            NaN  negative                NaN    PT_YJEZ38PS  PH_JBK8BD5J  \n",
       "8            NaN  negative                NaN    PT_YJEZ38PS  PH_E4YDYAEJ  \n",
       "9            NaN  positive                NaN    PT_YJEZ38PS  PH_BCRWHNSK  \n",
       "10           NaN  negative                NaN    PT_YJEZ38PS  PH_2HBS7DFV  \n",
       "11           NaN  negative                NaN    PT_YJEZ38PS  PH_8HMK4B5E  \n",
       "12           NaN  negative                NaN    PT_YJEZ38PS  PH_WCR9MGYP  \n",
       "13           NaN  positive                NaN    PT_YJEZ38PS  PH_209WTDM8  \n",
       "14    HP:0001161  positive                NaN    PT_YJEZ38PS  PH_47E0BVBD  \n",
       "15           NaN  negative                NaN    PT_YJEZ38PS  PH_SQ5PN54S  \n",
       "16           NaN  negative                NaN    PT_YJEZ38PS  PH_Z1FYH8TT  \n",
       "17           NaN  negative                NaN    PT_YJEZ38PS  PH_VAKJZKD4  \n",
       "18           NaN  negative                NaN    PT_YJEZ38PS  PH_E8D9V59V  \n",
       "19           NaN  negative                NaN    PT_YJEZ38PS  PH_NDQ0KD82  \n",
       "20           NaN  negative                NaN    PT_YJEZ38PS  PH_JBHMRFVV  \n",
       "21           NaN  negative                NaN    PT_YJEZ38PS  PH_MGR948NV  \n",
       "22           NaN  negative                NaN    PT_YJEZ38PS  PH_B1M29E8S  \n",
       "23           NaN  negative                NaN    PT_YJEZ38PS  PH_VHHFH6JN  \n",
       "24           NaN  negative                NaN    PT_YJEZ38PS  PH_08QV8GQY  \n",
       "25           NaN  negative                NaN    PT_YJEZ38PS  PH_J12SB1R5  \n",
       "26           NaN  negative                NaN    PT_YJEZ38PS  PH_N260Y9FG  \n",
       "27           NaN  negative                NaN    PT_9YVMV8HV  PH_G7VDY692  \n",
       "28           NaN  negative                NaN    PT_9YVMV8HV  PH_VD250RZE  \n",
       "29           NaN  negative                NaN    PT_9YVMV8HV  PH_VPHNFZ9Q  \n",
       "...          ...       ...                ...            ...          ...  \n",
       "8306         NaN  negative                NaN    PT_SKK2XWHX  PH_87W0X369  \n",
       "8307         NaN  negative                NaN    PT_3R09E42A  PH_HVV7MJZ1  \n",
       "8308         NaN  negative                NaN    PT_SZSVF28E  PH_437PSFDX  \n",
       "8309         NaN  negative                NaN    PT_DZF6WZTB  PH_W8X0Q8Q1  \n",
       "8310         NaN  negative                NaN    PT_9JMZSWZM  PH_A0XF8JPS  \n",
       "8311         NaN  negative                NaN    PT_N4XY02RM  PH_4RM7QPNH  \n",
       "8312         NaN  negative                NaN    PT_CZVKMYJ5  PH_WNSTES3P  \n",
       "8313         NaN  negative                NaN    PT_ZKVVK7YB  PH_YEJBZ3FY  \n",
       "8314         NaN  negative                NaN    PT_NT2QGE62  PH_YHN3SKAY  \n",
       "8315         NaN  negative                NaN    PT_NDNGQ5A9  PH_JGHS1JQ4  \n",
       "8316         NaN  negative                NaN    PT_GV191660  PH_54XJ9AWH  \n",
       "8317         NaN  negative                NaN    PT_1718C9HN  PH_WW9GG285  \n",
       "8318         NaN  negative                NaN    PT_PDAY4P1E  PH_0JHW0EGR  \n",
       "8319         NaN  negative                NaN    PT_CCR3SXD1  PH_623J09Q4  \n",
       "8320         NaN  negative                NaN    PT_FEYRKGW8  PH_WHS8NR0P  \n",
       "8321         NaN  negative                NaN    PT_VMW3ESD6  PH_BNEBK2VY  \n",
       "8322         NaN  negative                NaN    PT_XW4D7HBG  PH_SX2BSD03  \n",
       "8323         NaN  negative                NaN    PT_3P4K5J8N  PH_TCT6PQVX  \n",
       "8324         NaN  negative                NaN    PT_W3W87FY9  PH_SP6PYSA6  \n",
       "8325         NaN  negative                NaN    PT_9E6GW4YG  PH_GX1C3MAJ  \n",
       "8326         NaN  negative                NaN    PT_BVCGQ7D8  PH_74HE7FT2  \n",
       "8327         NaN  negative                NaN    PT_16R6B2T7  PH_BV9QVWFV  \n",
       "8328         NaN  negative                NaN    PT_26NG1DS8  PH_3V3W64CH  \n",
       "8329         NaN  negative                NaN    PT_0BGHDRZ5  PH_2A77VAS3  \n",
       "8330         NaN  negative                NaN    PT_GKH2YJTV  PH_B68G824Y  \n",
       "8331         NaN  negative                NaN    PT_5YNTF06A  PH_AHY3TNC0  \n",
       "8332         NaN  negative                NaN    PT_WSJ8H57D  PH_ZVEETSYC  \n",
       "8333         NaN  negative                NaN    PT_WJ2F5KBT  PH_WQA0VP5X  \n",
       "8334         NaN  negative                NaN    PT_E14YEG3F  PH_A246ZWNG  \n",
       "8335         NaN  negative                NaN    PT_G5GKEW54  PH_WNDB15KT  \n",
       "\n",
       "[8336 rows x 9 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/Users/singhn4/Desktop/phenotype.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uuid</th>\n",
       "      <th>created_at</th>\n",
       "      <th>modified_at</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>hpo_id</th>\n",
       "      <th>observed</th>\n",
       "      <th>age_at_event_days</th>\n",
       "      <th>participant_id</th>\n",
       "      <th>kf_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>f77d6506-70d8-4075-b4ff-e54fff9a452e</td>\n",
       "      <td>2018-02-27 09:35:00.03638</td>\n",
       "      <td>2018-02-27 09:35:00.036388</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_8JWM84CR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5cdea204-b001-4469-9934-5dfd8ce64f01</td>\n",
       "      <td>2018-02-27 09:35:00.038478</td>\n",
       "      <td>2018-02-27 09:35:00.038484</td>\n",
       "      <td>AIRWAY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_4WAKYZ20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>61c9f931-b47a-4849-bd40-09fc0eb614e6</td>\n",
       "      <td>2018-02-27 09:35:00.039441</td>\n",
       "      <td>2018-02-27 09:35:00.039446</td>\n",
       "      <td>CHEST_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_Z5KN5SN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>51cdd902-d508-48a0-a1c1-77a91d5e57b0</td>\n",
       "      <td>2018-02-27 09:35:00.040381</td>\n",
       "      <td>2018-02-27 09:35:00.040386</td>\n",
       "      <td>CLUB_FOOT</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_H17JQKYG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>a1690ee3-3565-4d2c-8f04-6d903ceda423</td>\n",
       "      <td>2018-02-27 09:35:00.041276</td>\n",
       "      <td>2018-02-27 09:35:00.041281</td>\n",
       "      <td>DERMATOLOGICAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_PFVG3BAZ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>b9f1735e-9de3-41d3-a4f1-3cfc16796ec2</td>\n",
       "      <td>2018-02-27 09:35:00.042194</td>\n",
       "      <td>2018-02-27 09:35:00.042199</td>\n",
       "      <td>DYSMORPHIC_FACIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_BY35ZYB5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ec15cd29-5631-4c38-8795-0bd68ec905cc</td>\n",
       "      <td>2018-02-27 09:35:00.043137</td>\n",
       "      <td>2018-02-27 09:35:00.043176</td>\n",
       "      <td>EAR_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_V672HJ6C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6b241509-9a2a-4bb9-bfbe-0c0b6844579a</td>\n",
       "      <td>2018-02-27 09:35:00.043983</td>\n",
       "      <td>2018-02-27 09:35:00.043989</td>\n",
       "      <td>ENDOCRINOLOGIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_JBK8BD5J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>b7c951dd-989a-4e1a-b721-d55d22024b1d</td>\n",
       "      <td>2018-02-27 09:35:00.044848</td>\n",
       "      <td>2018-02-27 09:35:00.044854</td>\n",
       "      <td>EYE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_E4YDYAEJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>14b79e40-a409-4c4c-a4ee-09da7f67aca7</td>\n",
       "      <td>2018-02-27 09:35:00.045662</td>\n",
       "      <td>2018-02-27 09:35:00.045667</td>\n",
       "      <td>FEET_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_BCRWHNSK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>fb168208-b5bb-45c2-8da7-8ea509f770e6</td>\n",
       "      <td>2018-02-27 09:35:00.046524</td>\n",
       "      <td>2018-02-27 09:35:00.046529</td>\n",
       "      <td>GENERAL_SKELETAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_2HBS7DFV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>81455ec6-1636-47e8-b015-99e3f77d8dde</td>\n",
       "      <td>2018-02-27 09:35:00.047324</td>\n",
       "      <td>2018-02-27 09:35:00.047329</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_8HMK4B5E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>3d00c9ab-8350-4ed7-81d7-97992cd88253</td>\n",
       "      <td>2018-02-27 09:35:00.048117</td>\n",
       "      <td>2018-02-27 09:35:00.048121</td>\n",
       "      <td>GENITOURINARY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_WCR9MGYP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>09340bec-d7bd-4f8d-bf5b-4230d8b072f5</td>\n",
       "      <td>2018-02-27 09:35:00.04887</td>\n",
       "      <td>2018-02-27 09:35:00.048875</td>\n",
       "      <td>HAND_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>positive</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_209WTDM8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>9a798967-4593-4dbc-bcf1-1b9573651469</td>\n",
       "      <td>2018-02-27 09:35:00.049611</td>\n",
       "      <td>2018-02-27 09:35:00.049616</td>\n",
       "      <td>HAND_POLYDACTYLY</td>\n",
       "      <td>HP:0001161</td>\n",
       "      <td>positive</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_47E0BVBD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>46b91b44-7f36-4018-8c75-416ef5bcb5e7</td>\n",
       "      <td>2018-02-27 09:35:00.050342</td>\n",
       "      <td>2018-02-27 09:35:00.050347</td>\n",
       "      <td>HEAD_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_SQ5PN54S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>602d9808-266a-47ec-8bb1-13c11f0f65a8</td>\n",
       "      <td>2018-02-27 09:35:00.051045</td>\n",
       "      <td>2018-02-27 09:35:00.05105</td>\n",
       "      <td>HEMATOLOGIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_Z1FYH8TT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6197d021-5f5c-4e98-a588-dc9efd04ad74</td>\n",
       "      <td>2018-02-27 09:35:00.051811</td>\n",
       "      <td>2018-02-27 09:35:00.051816</td>\n",
       "      <td>HEPATIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_VAKJZKD4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>26771f5e-0854-4847-bf22-14c7a6b7f56c</td>\n",
       "      <td>2018-02-27 09:35:00.052606</td>\n",
       "      <td>2018-02-27 09:35:00.05261</td>\n",
       "      <td>IMMUNOLOGIC_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_E8D9V59V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>c5d79c10-b183-45ff-aecf-68f4d3b8cfe1</td>\n",
       "      <td>2018-02-27 09:35:00.053359</td>\n",
       "      <td>2018-02-27 09:35:00.053363</td>\n",
       "      <td>MOUTH_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_NDQ0KD82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>53fff4fb-8ccb-4e9e-95d9-22c797095422</td>\n",
       "      <td>2018-02-27 09:35:00.054133</td>\n",
       "      <td>2018-02-27 09:35:00.054137</td>\n",
       "      <td>NECK_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_JBHMRFVV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>91108812-ba98-4630-84e6-a6772aebef49</td>\n",
       "      <td>2018-02-27 09:35:00.054941</td>\n",
       "      <td>2018-02-27 09:35:00.054946</td>\n",
       "      <td>NEUROLOGICAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_MGR948NV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>7114c05d-ce92-4d2d-8f25-bc2739d2ca24</td>\n",
       "      <td>2018-02-27 09:35:00.055658</td>\n",
       "      <td>2018-02-27 09:35:00.055663</td>\n",
       "      <td>NOSE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_B1M29E8S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>f032d7e8-1ae5-48b4-8737-d9cb503e48ea</td>\n",
       "      <td>2018-02-27 09:35:00.056404</td>\n",
       "      <td>2018-02-27 09:35:00.056409</td>\n",
       "      <td>PALATE_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_VHHFH6JN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>46c4c1f0-1766-48e7-a4d6-4c56c3dcc30f</td>\n",
       "      <td>2018-02-27 09:35:00.057155</td>\n",
       "      <td>2018-02-27 09:35:00.057159</td>\n",
       "      <td>PULMONARY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_08QV8GQY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>0677688c-6106-43ec-804a-244ad4e8d5e1</td>\n",
       "      <td>2018-02-27 09:35:00.057969</td>\n",
       "      <td>2018-02-27 09:35:00.057973</td>\n",
       "      <td>RENAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_J12SB1R5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>7bd0d0ec-dc7e-448f-b8ce-a0792a0ba941</td>\n",
       "      <td>2018-02-27 09:35:00.05874</td>\n",
       "      <td>2018-02-27 09:35:00.058744</td>\n",
       "      <td>THYMUS_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_YJEZ38PS</td>\n",
       "      <td>PH_N260Y9FG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>f6b6b95d-ba10-4474-962b-829907772552</td>\n",
       "      <td>2018-02-27 09:35:00.059443</td>\n",
       "      <td>2018-02-27 09:35:00.059448</td>\n",
       "      <td>ABDOMINAL_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_9YVMV8HV</td>\n",
       "      <td>PH_G7VDY692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>45884f70-f8a5-4bf0-921e-30314e17f5be</td>\n",
       "      <td>2018-02-27 09:35:00.060206</td>\n",
       "      <td>2018-02-27 09:35:00.06021</td>\n",
       "      <td>AIRWAY_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_9YVMV8HV</td>\n",
       "      <td>PH_VD250RZE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>0825ed0e-b593-468d-beb6-9750c6d1dc6c</td>\n",
       "      <td>2018-02-27 09:35:00.060895</td>\n",
       "      <td>2018-02-27 09:35:00.0609</td>\n",
       "      <td>CHEST_ABNORMALITIES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_9YVMV8HV</td>\n",
       "      <td>PH_VPHNFZ9Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8306</th>\n",
       "      <td>9a1dd373-21e5-4544-84fd-650b615042ac</td>\n",
       "      <td>2018-02-27 09:35:05.863413</td>\n",
       "      <td>2018-02-27 09:35:05.863418</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_SKK2XWHX</td>\n",
       "      <td>PH_87W0X369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8307</th>\n",
       "      <td>3523e046-29f7-4dbd-aeb9-0beb7c3aaece</td>\n",
       "      <td>2018-02-27 09:35:05.864054</td>\n",
       "      <td>2018-02-27 09:35:05.864059</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_3R09E42A</td>\n",
       "      <td>PH_HVV7MJZ1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8308</th>\n",
       "      <td>b583f43e-477d-4019-95cb-cd27e19220e7</td>\n",
       "      <td>2018-02-27 09:35:05.864671</td>\n",
       "      <td>2018-02-27 09:35:05.864675</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_SZSVF28E</td>\n",
       "      <td>PH_437PSFDX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8309</th>\n",
       "      <td>29548995-418f-4a44-a1df-1df54ab9cad5</td>\n",
       "      <td>2018-02-27 09:35:05.865283</td>\n",
       "      <td>2018-02-27 09:35:05.865287</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_DZF6WZTB</td>\n",
       "      <td>PH_W8X0Q8Q1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8310</th>\n",
       "      <td>ffb0caaa-149e-4f08-a882-c25f84d591ad</td>\n",
       "      <td>2018-02-27 09:35:05.865914</td>\n",
       "      <td>2018-02-27 09:35:05.865921</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_9JMZSWZM</td>\n",
       "      <td>PH_A0XF8JPS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8311</th>\n",
       "      <td>4eefea6b-3cb7-473b-b7ca-61368103ae53</td>\n",
       "      <td>2018-02-27 09:35:05.866713</td>\n",
       "      <td>2018-02-27 09:35:05.866719</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_N4XY02RM</td>\n",
       "      <td>PH_4RM7QPNH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8312</th>\n",
       "      <td>036ef30c-3c4a-41f0-93f6-0395918c3d48</td>\n",
       "      <td>2018-02-27 09:35:05.867373</td>\n",
       "      <td>2018-02-27 09:35:05.867378</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_CZVKMYJ5</td>\n",
       "      <td>PH_WNSTES3P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8313</th>\n",
       "      <td>a22e4b42-5447-4564-bc0a-185e3020e2df</td>\n",
       "      <td>2018-02-27 09:35:05.867984</td>\n",
       "      <td>2018-02-27 09:35:05.867989</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_ZKVVK7YB</td>\n",
       "      <td>PH_YEJBZ3FY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8314</th>\n",
       "      <td>d3bf6a0b-7cc9-4bea-a81f-28e4e39f6b41</td>\n",
       "      <td>2018-02-27 09:35:05.868581</td>\n",
       "      <td>2018-02-27 09:35:05.868585</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_NT2QGE62</td>\n",
       "      <td>PH_YHN3SKAY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8315</th>\n",
       "      <td>7de9874b-f952-4bdd-aa70-d5d493921148</td>\n",
       "      <td>2018-02-27 09:35:05.869171</td>\n",
       "      <td>2018-02-27 09:35:05.869175</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_NDNGQ5A9</td>\n",
       "      <td>PH_JGHS1JQ4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8316</th>\n",
       "      <td>6b769d1a-362f-44ac-aac7-11e4ca87ab78</td>\n",
       "      <td>2018-02-27 09:35:05.869758</td>\n",
       "      <td>2018-02-27 09:35:05.869762</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_GV191660</td>\n",
       "      <td>PH_54XJ9AWH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8317</th>\n",
       "      <td>52956a71-1f83-4877-b2e0-48076451fec5</td>\n",
       "      <td>2018-02-27 09:35:05.870441</td>\n",
       "      <td>2018-02-27 09:35:05.870445</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_1718C9HN</td>\n",
       "      <td>PH_WW9GG285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8318</th>\n",
       "      <td>9ba23215-2638-41c3-88ab-240d06e623b0</td>\n",
       "      <td>2018-02-27 09:35:05.871029</td>\n",
       "      <td>2018-02-27 09:35:05.871033</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_PDAY4P1E</td>\n",
       "      <td>PH_0JHW0EGR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8319</th>\n",
       "      <td>c35fcf3d-e74f-4ec0-accf-1e1b5dfb0c5d</td>\n",
       "      <td>2018-02-27 09:35:05.871615</td>\n",
       "      <td>2018-02-27 09:35:05.871619</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_CCR3SXD1</td>\n",
       "      <td>PH_623J09Q4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8320</th>\n",
       "      <td>fdc400be-95a5-42a1-9a71-273c5a0597e0</td>\n",
       "      <td>2018-02-27 09:35:05.872198</td>\n",
       "      <td>2018-02-27 09:35:05.872202</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_FEYRKGW8</td>\n",
       "      <td>PH_WHS8NR0P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8321</th>\n",
       "      <td>ad151aec-07fc-47f0-b7a9-2892fc54967a</td>\n",
       "      <td>2018-02-27 09:35:05.872809</td>\n",
       "      <td>2018-02-27 09:35:05.872813</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_VMW3ESD6</td>\n",
       "      <td>PH_BNEBK2VY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8322</th>\n",
       "      <td>789f40bd-464f-4321-881c-969365a614e5</td>\n",
       "      <td>2018-02-27 09:35:05.873414</td>\n",
       "      <td>2018-02-27 09:35:05.873419</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_XW4D7HBG</td>\n",
       "      <td>PH_SX2BSD03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8323</th>\n",
       "      <td>31e23a7d-f040-4134-af7a-ea3b917e0e1a</td>\n",
       "      <td>2018-02-27 09:35:05.874037</td>\n",
       "      <td>2018-02-27 09:35:05.874041</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_3P4K5J8N</td>\n",
       "      <td>PH_TCT6PQVX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8324</th>\n",
       "      <td>4db4f340-8b07-4147-a12b-2ee9a61b2424</td>\n",
       "      <td>2018-02-27 09:35:05.874627</td>\n",
       "      <td>2018-02-27 09:35:05.874631</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_W3W87FY9</td>\n",
       "      <td>PH_SP6PYSA6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8325</th>\n",
       "      <td>d1cd9026-dec2-43d9-9457-c9fe95bcd464</td>\n",
       "      <td>2018-02-27 09:35:05.875229</td>\n",
       "      <td>2018-02-27 09:35:05.875234</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_9E6GW4YG</td>\n",
       "      <td>PH_GX1C3MAJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8326</th>\n",
       "      <td>5e855ef3-d585-48a2-bb3d-2c1e741dfc5f</td>\n",
       "      <td>2018-02-27 09:35:05.875827</td>\n",
       "      <td>2018-02-27 09:35:05.875832</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_BVCGQ7D8</td>\n",
       "      <td>PH_74HE7FT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8327</th>\n",
       "      <td>f0429bd0-9adc-4169-9ec0-b6e4bfec145e</td>\n",
       "      <td>2018-02-27 09:35:05.876416</td>\n",
       "      <td>2018-02-27 09:35:05.87642</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_16R6B2T7</td>\n",
       "      <td>PH_BV9QVWFV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8328</th>\n",
       "      <td>d34511a6-052f-4ecd-9bca-36c9ba6b15e7</td>\n",
       "      <td>2018-02-27 09:35:05.877007</td>\n",
       "      <td>2018-02-27 09:35:05.877011</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_26NG1DS8</td>\n",
       "      <td>PH_3V3W64CH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8329</th>\n",
       "      <td>2b42ab77-4f24-4424-9c32-f52c7f41f3e6</td>\n",
       "      <td>2018-02-27 09:35:05.877587</td>\n",
       "      <td>2018-02-27 09:35:05.877591</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_0BGHDRZ5</td>\n",
       "      <td>PH_2A77VAS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8330</th>\n",
       "      <td>61f7a594-4e22-4755-929b-cc32d25207ad</td>\n",
       "      <td>2018-02-27 09:35:05.878171</td>\n",
       "      <td>2018-02-27 09:35:05.878175</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_GKH2YJTV</td>\n",
       "      <td>PH_B68G824Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8331</th>\n",
       "      <td>6d1c2b97-13bb-4753-97b9-cfdf29425e35</td>\n",
       "      <td>2018-02-27 09:35:05.878762</td>\n",
       "      <td>2018-02-27 09:35:05.878766</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_5YNTF06A</td>\n",
       "      <td>PH_AHY3TNC0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8332</th>\n",
       "      <td>b9d33cf4-9fb3-43df-85f8-f887da88617c</td>\n",
       "      <td>2018-02-27 09:35:05.87934</td>\n",
       "      <td>2018-02-27 09:35:05.879345</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_WSJ8H57D</td>\n",
       "      <td>PH_ZVEETSYC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8333</th>\n",
       "      <td>b7df02a0-f5a4-4a06-8d0c-f90cf4aac5ef</td>\n",
       "      <td>2018-02-27 09:35:05.880003</td>\n",
       "      <td>2018-02-27 09:35:05.880008</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_WJ2F5KBT</td>\n",
       "      <td>PH_WQA0VP5X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8334</th>\n",
       "      <td>c000b4de-2d18-4484-af5d-4a551fa37011</td>\n",
       "      <td>2018-02-27 09:35:05.880601</td>\n",
       "      <td>2018-02-27 09:35:05.880605</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_E14YEG3F</td>\n",
       "      <td>PH_A246ZWNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8335</th>\n",
       "      <td>5357c1db-6224-4a99-b2ef-4674419b4563</td>\n",
       "      <td>2018-02-27 09:35:05.88124</td>\n",
       "      <td>2018-02-27 09:35:05.881246</td>\n",
       "      <td>GENETIC_SYNDROMES_PRESENT</td>\n",
       "      <td>None</td>\n",
       "      <td>negative</td>\n",
       "      <td>None</td>\n",
       "      <td>PT_G5GKEW54</td>\n",
       "      <td>PH_WNDB15KT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8336 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      uuid                  created_at  \\\n",
       "0     f77d6506-70d8-4075-b4ff-e54fff9a452e   2018-02-27 09:35:00.03638   \n",
       "1     5cdea204-b001-4469-9934-5dfd8ce64f01  2018-02-27 09:35:00.038478   \n",
       "2     61c9f931-b47a-4849-bd40-09fc0eb614e6  2018-02-27 09:35:00.039441   \n",
       "3     51cdd902-d508-48a0-a1c1-77a91d5e57b0  2018-02-27 09:35:00.040381   \n",
       "4     a1690ee3-3565-4d2c-8f04-6d903ceda423  2018-02-27 09:35:00.041276   \n",
       "5     b9f1735e-9de3-41d3-a4f1-3cfc16796ec2  2018-02-27 09:35:00.042194   \n",
       "6     ec15cd29-5631-4c38-8795-0bd68ec905cc  2018-02-27 09:35:00.043137   \n",
       "7     6b241509-9a2a-4bb9-bfbe-0c0b6844579a  2018-02-27 09:35:00.043983   \n",
       "8     b7c951dd-989a-4e1a-b721-d55d22024b1d  2018-02-27 09:35:00.044848   \n",
       "9     14b79e40-a409-4c4c-a4ee-09da7f67aca7  2018-02-27 09:35:00.045662   \n",
       "10    fb168208-b5bb-45c2-8da7-8ea509f770e6  2018-02-27 09:35:00.046524   \n",
       "11    81455ec6-1636-47e8-b015-99e3f77d8dde  2018-02-27 09:35:00.047324   \n",
       "12    3d00c9ab-8350-4ed7-81d7-97992cd88253  2018-02-27 09:35:00.048117   \n",
       "13    09340bec-d7bd-4f8d-bf5b-4230d8b072f5   2018-02-27 09:35:00.04887   \n",
       "14    9a798967-4593-4dbc-bcf1-1b9573651469  2018-02-27 09:35:00.049611   \n",
       "15    46b91b44-7f36-4018-8c75-416ef5bcb5e7  2018-02-27 09:35:00.050342   \n",
       "16    602d9808-266a-47ec-8bb1-13c11f0f65a8  2018-02-27 09:35:00.051045   \n",
       "17    6197d021-5f5c-4e98-a588-dc9efd04ad74  2018-02-27 09:35:00.051811   \n",
       "18    26771f5e-0854-4847-bf22-14c7a6b7f56c  2018-02-27 09:35:00.052606   \n",
       "19    c5d79c10-b183-45ff-aecf-68f4d3b8cfe1  2018-02-27 09:35:00.053359   \n",
       "20    53fff4fb-8ccb-4e9e-95d9-22c797095422  2018-02-27 09:35:00.054133   \n",
       "21    91108812-ba98-4630-84e6-a6772aebef49  2018-02-27 09:35:00.054941   \n",
       "22    7114c05d-ce92-4d2d-8f25-bc2739d2ca24  2018-02-27 09:35:00.055658   \n",
       "23    f032d7e8-1ae5-48b4-8737-d9cb503e48ea  2018-02-27 09:35:00.056404   \n",
       "24    46c4c1f0-1766-48e7-a4d6-4c56c3dcc30f  2018-02-27 09:35:00.057155   \n",
       "25    0677688c-6106-43ec-804a-244ad4e8d5e1  2018-02-27 09:35:00.057969   \n",
       "26    7bd0d0ec-dc7e-448f-b8ce-a0792a0ba941   2018-02-27 09:35:00.05874   \n",
       "27    f6b6b95d-ba10-4474-962b-829907772552  2018-02-27 09:35:00.059443   \n",
       "28    45884f70-f8a5-4bf0-921e-30314e17f5be  2018-02-27 09:35:00.060206   \n",
       "29    0825ed0e-b593-468d-beb6-9750c6d1dc6c  2018-02-27 09:35:00.060895   \n",
       "...                                    ...                         ...   \n",
       "8306  9a1dd373-21e5-4544-84fd-650b615042ac  2018-02-27 09:35:05.863413   \n",
       "8307  3523e046-29f7-4dbd-aeb9-0beb7c3aaece  2018-02-27 09:35:05.864054   \n",
       "8308  b583f43e-477d-4019-95cb-cd27e19220e7  2018-02-27 09:35:05.864671   \n",
       "8309  29548995-418f-4a44-a1df-1df54ab9cad5  2018-02-27 09:35:05.865283   \n",
       "8310  ffb0caaa-149e-4f08-a882-c25f84d591ad  2018-02-27 09:35:05.865914   \n",
       "8311  4eefea6b-3cb7-473b-b7ca-61368103ae53  2018-02-27 09:35:05.866713   \n",
       "8312  036ef30c-3c4a-41f0-93f6-0395918c3d48  2018-02-27 09:35:05.867373   \n",
       "8313  a22e4b42-5447-4564-bc0a-185e3020e2df  2018-02-27 09:35:05.867984   \n",
       "8314  d3bf6a0b-7cc9-4bea-a81f-28e4e39f6b41  2018-02-27 09:35:05.868581   \n",
       "8315  7de9874b-f952-4bdd-aa70-d5d493921148  2018-02-27 09:35:05.869171   \n",
       "8316  6b769d1a-362f-44ac-aac7-11e4ca87ab78  2018-02-27 09:35:05.869758   \n",
       "8317  52956a71-1f83-4877-b2e0-48076451fec5  2018-02-27 09:35:05.870441   \n",
       "8318  9ba23215-2638-41c3-88ab-240d06e623b0  2018-02-27 09:35:05.871029   \n",
       "8319  c35fcf3d-e74f-4ec0-accf-1e1b5dfb0c5d  2018-02-27 09:35:05.871615   \n",
       "8320  fdc400be-95a5-42a1-9a71-273c5a0597e0  2018-02-27 09:35:05.872198   \n",
       "8321  ad151aec-07fc-47f0-b7a9-2892fc54967a  2018-02-27 09:35:05.872809   \n",
       "8322  789f40bd-464f-4321-881c-969365a614e5  2018-02-27 09:35:05.873414   \n",
       "8323  31e23a7d-f040-4134-af7a-ea3b917e0e1a  2018-02-27 09:35:05.874037   \n",
       "8324  4db4f340-8b07-4147-a12b-2ee9a61b2424  2018-02-27 09:35:05.874627   \n",
       "8325  d1cd9026-dec2-43d9-9457-c9fe95bcd464  2018-02-27 09:35:05.875229   \n",
       "8326  5e855ef3-d585-48a2-bb3d-2c1e741dfc5f  2018-02-27 09:35:05.875827   \n",
       "8327  f0429bd0-9adc-4169-9ec0-b6e4bfec145e  2018-02-27 09:35:05.876416   \n",
       "8328  d34511a6-052f-4ecd-9bca-36c9ba6b15e7  2018-02-27 09:35:05.877007   \n",
       "8329  2b42ab77-4f24-4424-9c32-f52c7f41f3e6  2018-02-27 09:35:05.877587   \n",
       "8330  61f7a594-4e22-4755-929b-cc32d25207ad  2018-02-27 09:35:05.878171   \n",
       "8331  6d1c2b97-13bb-4753-97b9-cfdf29425e35  2018-02-27 09:35:05.878762   \n",
       "8332  b9d33cf4-9fb3-43df-85f8-f887da88617c   2018-02-27 09:35:05.87934   \n",
       "8333  b7df02a0-f5a4-4a06-8d0c-f90cf4aac5ef  2018-02-27 09:35:05.880003   \n",
       "8334  c000b4de-2d18-4484-af5d-4a551fa37011  2018-02-27 09:35:05.880601   \n",
       "8335  5357c1db-6224-4a99-b2ef-4674419b4563   2018-02-27 09:35:05.88124   \n",
       "\n",
       "                     modified_at                               phenotype  \\\n",
       "0     2018-02-27 09:35:00.036388         ABDOMINAL_ABNORMALITIES_PRESENT   \n",
       "1     2018-02-27 09:35:00.038484            AIRWAY_ABNORMALITIES_PRESENT   \n",
       "2     2018-02-27 09:35:00.039446             CHEST_ABNORMALITIES_PRESENT   \n",
       "3     2018-02-27 09:35:00.040386                               CLUB_FOOT   \n",
       "4     2018-02-27 09:35:00.041281    DERMATOLOGICAL_ABNORMALITIES_PRESENT   \n",
       "5     2018-02-27 09:35:00.042199               DYSMORPHIC_FACIES_PRESENT   \n",
       "6     2018-02-27 09:35:00.043176               EAR_ABNORMALITIES_PRESENT   \n",
       "7     2018-02-27 09:35:00.043989    ENDOCRINOLOGIC_ABNORMALITIES_PRESENT   \n",
       "8     2018-02-27 09:35:00.044854               EYE_ABNORMALITIES_PRESENT   \n",
       "9     2018-02-27 09:35:00.045667              FEET_ABNORMALITIES_PRESENT   \n",
       "10    2018-02-27 09:35:00.046529  GENERAL_SKELETAL_ABNORMALITIES_PRESENT   \n",
       "11    2018-02-27 09:35:00.047329               GENETIC_SYNDROMES_PRESENT   \n",
       "12    2018-02-27 09:35:00.048121     GENITOURINARY_ABNORMALITIES_PRESENT   \n",
       "13    2018-02-27 09:35:00.048875              HAND_ABNORMALITIES_PRESENT   \n",
       "14    2018-02-27 09:35:00.049616                        HAND_POLYDACTYLY   \n",
       "15    2018-02-27 09:35:00.050347              HEAD_ABNORMALITIES_PRESENT   \n",
       "16     2018-02-27 09:35:00.05105       HEMATOLOGIC_ABNORMALITIES_PRESENT   \n",
       "17    2018-02-27 09:35:00.051816           HEPATIC_ABNORMALITIES_PRESENT   \n",
       "18     2018-02-27 09:35:00.05261       IMMUNOLOGIC_ABNORMALITIES_PRESENT   \n",
       "19    2018-02-27 09:35:00.053363             MOUTH_ABNORMALITIES_PRESENT   \n",
       "20    2018-02-27 09:35:00.054137              NECK_ABNORMALITIES_PRESENT   \n",
       "21    2018-02-27 09:35:00.054946      NEUROLOGICAL_ABNORMALITIES_PRESENT   \n",
       "22    2018-02-27 09:35:00.055663              NOSE_ABNORMALITIES_PRESENT   \n",
       "23    2018-02-27 09:35:00.056409            PALATE_ABNORMALITIES_PRESENT   \n",
       "24    2018-02-27 09:35:00.057159         PULMONARY_ABNORMALITIES_PRESENT   \n",
       "25    2018-02-27 09:35:00.057973             RENAL_ABNORMALITIES_PRESENT   \n",
       "26    2018-02-27 09:35:00.058744            THYMUS_ABNORMALITIES_PRESENT   \n",
       "27    2018-02-27 09:35:00.059448         ABDOMINAL_ABNORMALITIES_PRESENT   \n",
       "28     2018-02-27 09:35:00.06021            AIRWAY_ABNORMALITIES_PRESENT   \n",
       "29      2018-02-27 09:35:00.0609             CHEST_ABNORMALITIES_PRESENT   \n",
       "...                          ...                                     ...   \n",
       "8306  2018-02-27 09:35:05.863418               GENETIC_SYNDROMES_PRESENT   \n",
       "8307  2018-02-27 09:35:05.864059               GENETIC_SYNDROMES_PRESENT   \n",
       "8308  2018-02-27 09:35:05.864675               GENETIC_SYNDROMES_PRESENT   \n",
       "8309  2018-02-27 09:35:05.865287               GENETIC_SYNDROMES_PRESENT   \n",
       "8310  2018-02-27 09:35:05.865921               GENETIC_SYNDROMES_PRESENT   \n",
       "8311  2018-02-27 09:35:05.866719               GENETIC_SYNDROMES_PRESENT   \n",
       "8312  2018-02-27 09:35:05.867378               GENETIC_SYNDROMES_PRESENT   \n",
       "8313  2018-02-27 09:35:05.867989               GENETIC_SYNDROMES_PRESENT   \n",
       "8314  2018-02-27 09:35:05.868585               GENETIC_SYNDROMES_PRESENT   \n",
       "8315  2018-02-27 09:35:05.869175               GENETIC_SYNDROMES_PRESENT   \n",
       "8316  2018-02-27 09:35:05.869762               GENETIC_SYNDROMES_PRESENT   \n",
       "8317  2018-02-27 09:35:05.870445               GENETIC_SYNDROMES_PRESENT   \n",
       "8318  2018-02-27 09:35:05.871033               GENETIC_SYNDROMES_PRESENT   \n",
       "8319  2018-02-27 09:35:05.871619               GENETIC_SYNDROMES_PRESENT   \n",
       "8320  2018-02-27 09:35:05.872202               GENETIC_SYNDROMES_PRESENT   \n",
       "8321  2018-02-27 09:35:05.872813               GENETIC_SYNDROMES_PRESENT   \n",
       "8322  2018-02-27 09:35:05.873419               GENETIC_SYNDROMES_PRESENT   \n",
       "8323  2018-02-27 09:35:05.874041               GENETIC_SYNDROMES_PRESENT   \n",
       "8324  2018-02-27 09:35:05.874631               GENETIC_SYNDROMES_PRESENT   \n",
       "8325  2018-02-27 09:35:05.875234               GENETIC_SYNDROMES_PRESENT   \n",
       "8326  2018-02-27 09:35:05.875832               GENETIC_SYNDROMES_PRESENT   \n",
       "8327   2018-02-27 09:35:05.87642               GENETIC_SYNDROMES_PRESENT   \n",
       "8328  2018-02-27 09:35:05.877011               GENETIC_SYNDROMES_PRESENT   \n",
       "8329  2018-02-27 09:35:05.877591               GENETIC_SYNDROMES_PRESENT   \n",
       "8330  2018-02-27 09:35:05.878175               GENETIC_SYNDROMES_PRESENT   \n",
       "8331  2018-02-27 09:35:05.878766               GENETIC_SYNDROMES_PRESENT   \n",
       "8332  2018-02-27 09:35:05.879345               GENETIC_SYNDROMES_PRESENT   \n",
       "8333  2018-02-27 09:35:05.880008               GENETIC_SYNDROMES_PRESENT   \n",
       "8334  2018-02-27 09:35:05.880605               GENETIC_SYNDROMES_PRESENT   \n",
       "8335  2018-02-27 09:35:05.881246               GENETIC_SYNDROMES_PRESENT   \n",
       "\n",
       "          hpo_id  observed age_at_event_days participant_id        kf_id  \n",
       "0           None  negative              None    PT_YJEZ38PS  PH_8JWM84CR  \n",
       "1           None  negative              None    PT_YJEZ38PS  PH_4WAKYZ20  \n",
       "2           None  negative              None    PT_YJEZ38PS  PH_Z5KN5SN2  \n",
       "3           None  positive              None    PT_YJEZ38PS  PH_H17JQKYG  \n",
       "4           None  negative              None    PT_YJEZ38PS  PH_PFVG3BAZ  \n",
       "5           None  negative              None    PT_YJEZ38PS  PH_BY35ZYB5  \n",
       "6           None  negative              None    PT_YJEZ38PS  PH_V672HJ6C  \n",
       "7           None  negative              None    PT_YJEZ38PS  PH_JBK8BD5J  \n",
       "8           None  negative              None    PT_YJEZ38PS  PH_E4YDYAEJ  \n",
       "9           None  positive              None    PT_YJEZ38PS  PH_BCRWHNSK  \n",
       "10          None  negative              None    PT_YJEZ38PS  PH_2HBS7DFV  \n",
       "11          None  negative              None    PT_YJEZ38PS  PH_8HMK4B5E  \n",
       "12          None  negative              None    PT_YJEZ38PS  PH_WCR9MGYP  \n",
       "13          None  positive              None    PT_YJEZ38PS  PH_209WTDM8  \n",
       "14    HP:0001161  positive              None    PT_YJEZ38PS  PH_47E0BVBD  \n",
       "15          None  negative              None    PT_YJEZ38PS  PH_SQ5PN54S  \n",
       "16          None  negative              None    PT_YJEZ38PS  PH_Z1FYH8TT  \n",
       "17          None  negative              None    PT_YJEZ38PS  PH_VAKJZKD4  \n",
       "18          None  negative              None    PT_YJEZ38PS  PH_E8D9V59V  \n",
       "19          None  negative              None    PT_YJEZ38PS  PH_NDQ0KD82  \n",
       "20          None  negative              None    PT_YJEZ38PS  PH_JBHMRFVV  \n",
       "21          None  negative              None    PT_YJEZ38PS  PH_MGR948NV  \n",
       "22          None  negative              None    PT_YJEZ38PS  PH_B1M29E8S  \n",
       "23          None  negative              None    PT_YJEZ38PS  PH_VHHFH6JN  \n",
       "24          None  negative              None    PT_YJEZ38PS  PH_08QV8GQY  \n",
       "25          None  negative              None    PT_YJEZ38PS  PH_J12SB1R5  \n",
       "26          None  negative              None    PT_YJEZ38PS  PH_N260Y9FG  \n",
       "27          None  negative              None    PT_9YVMV8HV  PH_G7VDY692  \n",
       "28          None  negative              None    PT_9YVMV8HV  PH_VD250RZE  \n",
       "29          None  negative              None    PT_9YVMV8HV  PH_VPHNFZ9Q  \n",
       "...          ...       ...               ...            ...          ...  \n",
       "8306        None  negative              None    PT_SKK2XWHX  PH_87W0X369  \n",
       "8307        None  negative              None    PT_3R09E42A  PH_HVV7MJZ1  \n",
       "8308        None  negative              None    PT_SZSVF28E  PH_437PSFDX  \n",
       "8309        None  negative              None    PT_DZF6WZTB  PH_W8X0Q8Q1  \n",
       "8310        None  negative              None    PT_9JMZSWZM  PH_A0XF8JPS  \n",
       "8311        None  negative              None    PT_N4XY02RM  PH_4RM7QPNH  \n",
       "8312        None  negative              None    PT_CZVKMYJ5  PH_WNSTES3P  \n",
       "8313        None  negative              None    PT_ZKVVK7YB  PH_YEJBZ3FY  \n",
       "8314        None  negative              None    PT_NT2QGE62  PH_YHN3SKAY  \n",
       "8315        None  negative              None    PT_NDNGQ5A9  PH_JGHS1JQ4  \n",
       "8316        None  negative              None    PT_GV191660  PH_54XJ9AWH  \n",
       "8317        None  negative              None    PT_1718C9HN  PH_WW9GG285  \n",
       "8318        None  negative              None    PT_PDAY4P1E  PH_0JHW0EGR  \n",
       "8319        None  negative              None    PT_CCR3SXD1  PH_623J09Q4  \n",
       "8320        None  negative              None    PT_FEYRKGW8  PH_WHS8NR0P  \n",
       "8321        None  negative              None    PT_VMW3ESD6  PH_BNEBK2VY  \n",
       "8322        None  negative              None    PT_XW4D7HBG  PH_SX2BSD03  \n",
       "8323        None  negative              None    PT_3P4K5J8N  PH_TCT6PQVX  \n",
       "8324        None  negative              None    PT_W3W87FY9  PH_SP6PYSA6  \n",
       "8325        None  negative              None    PT_9E6GW4YG  PH_GX1C3MAJ  \n",
       "8326        None  negative              None    PT_BVCGQ7D8  PH_74HE7FT2  \n",
       "8327        None  negative              None    PT_16R6B2T7  PH_BV9QVWFV  \n",
       "8328        None  negative              None    PT_26NG1DS8  PH_3V3W64CH  \n",
       "8329        None  negative              None    PT_0BGHDRZ5  PH_2A77VAS3  \n",
       "8330        None  negative              None    PT_GKH2YJTV  PH_B68G824Y  \n",
       "8331        None  negative              None    PT_5YNTF06A  PH_AHY3TNC0  \n",
       "8332        None  negative              None    PT_WSJ8H57D  PH_ZVEETSYC  \n",
       "8333        None  negative              None    PT_WJ2F5KBT  PH_WQA0VP5X  \n",
       "8334        None  negative              None    PT_E14YEG3F  PH_A246ZWNG  \n",
       "8335        None  negative              None    PT_G5GKEW54  PH_WNDB15KT  \n",
       "\n",
       "[8336 rows x 9 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.where((pd.notnull(df)), None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
